{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "86c88da5-4793-43df-a099-ab477cce2cac",
   "metadata": {},
   "source": [
    "# Genearate Data\n",
    "Generate all data needed to run analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0be072a-c289-436d-83e6-90dc2bff85cf",
   "metadata": {},
   "source": [
    "### Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "49bd78dd-82b0-4930-9e30-db578c2ef177",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import qiime2 as q2\n",
    "from qiime2.plugins import taxa\n",
    "\n",
    "from python_scripts.import_split_biom import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f25effc8-530a-4fe6-bd78-b564b9468616",
   "metadata": {},
   "source": [
    "### Datatypes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83dc865c-d46a-42cd-91f1-57d5b4f03c0e",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### EAC ICGC\n",
    "\n",
    "\"Pan-Cancer Analysis of Whole Genomes.\" Nature, vol. 578, no. 7793, 2020, pp. 82-93,  https://doi.org/10.1038/s41586-020-1969-6. Accessed 17 Jan. 2024.\n",
    "(Qiita: 14857)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42d3a9e2-a281-4b4f-8fce-24d2e01d3fcb",
   "metadata": {},
   "source": [
    "Note: As a default now, removes the samples that are also in Ross-Innes (Ginny_list)  \n",
    "Other params (might use):  remove_reag, keep_hu  \n",
    "Other params (likely won't use):ZEBRA: zebra='EAC_ICGC_old', zebra_threshold=0.013, REMOVED sketky looking microbes: remove_bad_gotu=True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "301d6a97-8f7d-4e17-b773-6e6fcb131dbf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>project</th>\n",
       "      <th>filename</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>donor_tumour_staging_system_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis_supplemental</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14857.14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deceased</td>\n",
       "      <td>...</td>\n",
       "      <td>65.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>66.0</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>712.0</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>no</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:1b</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14857.14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alive</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>68.0</td>\n",
       "      <td>stable</td>\n",
       "      <td>401.0</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>TNM6</td>\n",
       "      <td>UICC:3</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14857.14521.04ceb5f141916f7fc091a5e0aa51ada3</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>04ceb5f141916f7fc091a5e0aa51ada3.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50406</td>\n",
       "      <td>SA528949</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alive</td>\n",
       "      <td>...</td>\n",
       "      <td>74.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>76.0</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>767.0</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>no</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:1b</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    sample_name  project  \\\n",
       "1  14857.14521.0337efea9b59b88f804ea384c356377c  ESAD-UK   \n",
       "2  14857.14521.04a446c6a660694dafe43ccaf1db7162  ESAD-UK   \n",
       "3  14857.14521.04ceb5f141916f7fc091a5e0aa51ada3  ESAD-UK   \n",
       "\n",
       "                                        filename donor_sex project_code  \\\n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam      male      ESAD-UK   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam      male      ESAD-UK   \n",
       "3  04ceb5f141916f7fc091a5e0aa51ada3.filtered.bam      male      ESAD-UK   \n",
       "\n",
       "  icgc_donor_id icgc_sample_id prior_malignancy donor_relapse_type  \\\n",
       "1       DO50389       SA528947          unknown                NaN   \n",
       "2       DO50412       SA528884          unknown                NaN   \n",
       "3       DO50406       SA528949          unknown                NaN   \n",
       "\n",
       "  donor_vital_status  ... donor_age_at_enrollment  study_donor_involved_in  \\\n",
       "1           deceased  ...                    65.0                    PCAWG   \n",
       "2              alive  ...                     NaN                    PCAWG   \n",
       "3              alive  ...                    74.0                    PCAWG   \n",
       "\n",
       "  donor_age_at_last_followup  disease_status_last_followup  \\\n",
       "1                       66.0        no evidence of disease   \n",
       "2                       68.0                        stable   \n",
       "3                       76.0        no evidence of disease   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_tumour_stage_at_diagnosis  \\\n",
       "1                            712.0                           T2N0M0   \n",
       "2                            401.0                           T3N1M0   \n",
       "3                            767.0                           T2N0M0   \n",
       "\n",
       "  cancer_history_first_degree_relative  \\\n",
       "1                                   no   \n",
       "2                              unknown   \n",
       "3                                   no   \n",
       "\n",
       "   donor_tumour_staging_system_at_diagnosis  \\\n",
       "1                                      TNM7   \n",
       "2                                      TNM6   \n",
       "3                                      TNM7   \n",
       "\n",
       "  donor_tumour_stage_at_diagnosis_supplemental  qiita_study_id  \n",
       "1                                      UICC:1b           14857  \n",
       "2                                       UICC:3           14857  \n",
       "3                                      UICC:1b           14857  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_ICGC_species_WOL_scrubbed  =  83\n",
      "DF len 3453\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:236: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Default\n",
    "eac_14857_data_split(os.getcwd() +'/qiita_downloads/qiita14857_EAC_ICGC/162938_species_WoLr1_scrubbed/feature-table.biom',\n",
    "                'EAC_ICGC_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "618c6164-aff2-44f1-9e22-e9a8463cf8c1",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Normal/Healthy & GERD Esoph Brushings  \n",
    "Deshpande, N. P., Riordan, S. M., Castaño-Rodríguez, N., Wilkins, M. R. & Kaakoush, N. O. Signatures within the esophageal microbiome are associated with host genetics, age, and disease. Microbiome vol. 6 (2018) https://doi.org/10.1186/s40168-018-0611-4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "763ce1de-fe4c-4514-a27d-edf624eb28d8",
   "metadata": {},
   "source": [
    "Other params (might use):  remove_reag, keep_hu  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5386f24e-43d2-48f5-8be7-0e212052e620",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Normal / Healthy Esophgous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "af2a1551-8f5a-4256-a657-5fec421cb777",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_species_WOL_scrubbed  =  50\n",
      "DF len 2594\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:236: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Default\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_species_WOL_scrubbed/feature-table.biom',\n",
    "                       'normal_species_WOL_scrubbed',)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6f428a7-8d58-4b0a-8faa-bfab0942a8dd",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### GERD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "123fed91-a898-4b2c-956d-497bf9931e11",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_species_WOL_scrubbed  =  29\n",
      "DF len 2594\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:236: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Default\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_species_WOL_scrubbed/feature-table.biom',\n",
    "                       'GERD_species_WOL_scrubbed',)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "230a33d7-437a-4f89-b456-54593cca3d48",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Barrett's Esophgous Progressors/Non Progressors\n",
    "Paulson, T. G. et al. Somatic whole genome dynamics of precancer in Barrett’s esophagus reveals features associated with disease progression. Nature Communications vol. 13 (2022) https://doi.org/10.1038/s41467-022-29767-7"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6dea2d92-af99-4b6e-a54d-20a980f4c671",
   "metadata": {},
   "source": [
    "Other params (might use):  remove_reag, keep_hu, exact_pairs = True"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea5c4296-def2-4229-b2ae-27670bfb85be",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "349b20c1-8cd2-4433-9d73-754b45cb2b5a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               53.23           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_species_WOL_scrubbed  =  80\n",
      "DF len 1578\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:236: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Default\n",
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_T1_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aef4c92a-0a61-458e-81a0-929b46fb1315",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### [Currently not in paper] Barrett's Esophgous Non Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0fc2388a-20bb-4e54-ae62-52d51aef237e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2                NP               57.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_species_WOL_scrubbed  =  78\n",
      "DF len 1578\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:236: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Default\n",
    "be_data_split(['NP'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_T2_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "29fa4ebc-f730-4c2d-81b9-0971436ad38b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2                NP               57.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_genome_RS210  =  74\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2                NP               57.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_species_WOL  =  78\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1068.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>59</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>59-24989N-165S-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23644</td>\n",
       "      <td>NP</td>\n",
       "      <td>69.0</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M        17         T2   \n",
       "1  14598.PCGA.1067.BE.02  Asian_PacificIslander      M        55         T2   \n",
       "2  14598.PCGA.1068.BE.03      White_nonHispanic      M        59         T2   \n",
       "\n",
       "  ismainstudy        pairednormal distancefromgej orginal_sampleid  \\\n",
       "0           1  17-24987N-105W-BLD               4            23567   \n",
       "1           1  55-25056N-204U-BLD               1            23614   \n",
       "2           1  59-24989N-165S-BLD               3            23644   \n",
       "\n",
       "  progressionstatus patientageatsample qiita_study_id  \\\n",
       "0                NP              87.07          14598   \n",
       "1                NP              57.07          14598   \n",
       "2                NP               69.0          14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_exact_zebra_genome_WOL  =  40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2                NP               57.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_zebra_genome_WOL  =  78\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_NP_T2_genome_RS210')\n",
    "\n",
    "be_data_split(['NP'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_NP_T2_species_WOL')\n",
    "\n",
    "be_data_split(['NP'], ['T2'], \n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_T2_exact_zebra_genome_WOL',\n",
    "             zebra = 'BE_WOL',\n",
    "             exact_pairs = True)\n",
    "\n",
    "be_data_split(['NP'], ['T2'], \n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_T2_zebra_genome_WOL',\n",
    "             zebra = 'BE_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8037a84d-adc1-4bf9-b6d3-2d837afbf1ef",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "fd49d98b-58e2-4faf-a826-d8cf0abc6ba4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               56.25           14598   \n",
       "1              Prog               56.25           14598   \n",
       "2              Prog               64.11           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T1_Prog  \n",
       "1                      T1_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_species_WOL_scrubbed  =  79\n",
      "DF len 1578\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:236: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Default\n",
    "be_data_split(['Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_Prog_T1_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "532903b1-2dcc-408f-90c5-709a875b7646",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ec4a5e0f-936d-4d98-8065-567be3acfb31",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               65.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_species_WOL_scrubbed  =  79\n",
      "DF len 1578\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:236: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Default\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_Prog_T2_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4de4bd81-f239-4139-be7f-4cf8cf58fdb0",
   "metadata": {
    "tags": []
   },
   "source": [
    "### BE, EAC Tissue\n",
    "Ross-Innes, C. S. et al. Whole-genome sequencing provides new insights into the clonal architecture of Barrett’s esophagus and esophageal adenocarcinoma. Nature Genetics vol. 47 1038–1046 (2015) https://doi.org/10.1038/ng.3357. This also has normal tissue but we are not using it. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "377f02fe-c21a-47aa-b38f-e4d42e29b1d5",
   "metadata": {},
   "source": [
    "Note: As a default now, only keeps the exact 23 samples that are acutally in Ross-Innes publication (Ginny_list)  \n",
    "Other params (might use):  remove_reag, keep_hu  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "faf44e66-26a9-4bbd-b773-97dad3f1cade",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### BE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4409317b-3357-4d7c-a837-3c4717664f9c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "      <th>EGA_ID</th>\n",
       "      <th>EGAIdentifier</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000530875</td>\n",
       "      <td>LP6007521-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000586502</td>\n",
       "      <td>LP6005335-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000525202</td>\n",
       "      <td>LP6007410-DNA_A01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  ...  % aligned pairs  \\\n",
       "0    IB             Blood                     4.0  ...            92.57   \n",
       "1  IIIC  Normal esophagus                     3.0  ...            93.50   \n",
       "2   IIB             Blood                     NaN  ...            93.01   \n",
       "\n",
       "  median insert size  % mapQ=0  error rate HiQ  genome coverage mapQ>3  \\\n",
       "0              339.0      5.30           0.245                   55.76   \n",
       "1              300.0      4.77           0.231                   67.45   \n",
       "2              310.0      5.04           0.243                   57.20   \n",
       "\n",
       "   Callable genome  totalNumberSnps  Sample purity           EGA_ID  \\\n",
       "0            0.975              NaN            0.8  EGAF00000530875   \n",
       "1            0.976              NaN            0.6  EGAF00000586502   \n",
       "2            0.975              NaN            0.3  EGAF00000525202   \n",
       "\n",
       "       EGAIdentifier  \n",
       "0  LP6007521-DNA_A01  \n",
       "1  LP6005335-DNA_A01  \n",
       "2  LP6007410-DNA_A01  \n",
       "\n",
       "[3 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_species_WOL_scrubbed  =  23\n",
      "DF len 2605\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:236: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Default\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired_scrubbed/feature-table.biom',\n",
    "                         'BE_Ross_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7242c10-b39d-4545-86d9-3b2ec15bb22b",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### EAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5c3dc5a8-9a9c-4959-bb4a-43850788fccc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "      <th>EGA_ID</th>\n",
       "      <th>EGAIdentifier</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000530893</td>\n",
       "      <td>LP6007520-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000586547</td>\n",
       "      <td>LP6005334-DNA_A03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "      <td>EGAF00000525525</td>\n",
       "      <td>LP6007409-DNA_A01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  ...  % aligned pairs  \\\n",
       "0    IB             Blood                     4.0  ...            93.72   \n",
       "1  IIIC  Normal esophagus                     3.0  ...            93.70   \n",
       "2   IIB             Blood                     NaN  ...            93.33   \n",
       "\n",
       "  median insert size  % mapQ=0  error rate HiQ  genome coverage mapQ>3  \\\n",
       "0              315.0      4.57           0.229                   58.65   \n",
       "1              306.0      5.03           0.224                   58.86   \n",
       "2              299.0      4.91           0.235                   64.32   \n",
       "\n",
       "   Callable genome  totalNumberSnps  Sample purity           EGA_ID  \\\n",
       "0            0.975              NaN            0.5  EGAF00000530893   \n",
       "1            0.972              NaN            0.7  EGAF00000586547   \n",
       "2            0.976              NaN            0.4  EGAF00000525525   \n",
       "\n",
       "       EGAIdentifier  \n",
       "0  LP6007520-DNA_A01  \n",
       "1  LP6005334-DNA_A03  \n",
       "2  LP6007409-DNA_A01  \n",
       "\n",
       "[3 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_species_WOL_scrubbed  =  23\n",
      "DF len 2605\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:236: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Default\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired_scrubbed/feature-table.biom',\n",
    "                         'EAC_Ross_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e6d5b2c-3936-4d51-94c3-c60ce4ab38ce",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### [Supplemental?] From Sam: Barrett's Esophgous Progressors/Non Progressors: using MetaPhlan4\n",
    "Paulson, T. G. et al. Somatic whole genome dynamics of precancer in Barrett’s esophagus reveals features associated with disease progression. Nature Communications vol. 13 (2022) https://doi.org/10.1038/s41467-022-29767-7"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "677326ff-6940-41df-962b-ef473411db3d",
   "metadata": {},
   "source": [
    "Note:  \n",
    "Classic MetaPhlan4 : /Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv (remove the 'True' if you want this)\n",
    "Counts MetaPhlan4: /Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dfeb2eb4-eaf7-4448-8175-a994626ff617",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "247770e3-12fe-4889-9ebd-2f7f4232a498",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               53.23           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_species_MP_counts  =  53\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['NP'], ['T1'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_NP_T1_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cc98a2e-80d4-4c39-82ff-03fc036daa8f",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d02eac75-3ef0-409d-90ae-df96522e04e5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               87.07           14598  \n",
       "1                NP               87.07           14598  \n",
       "2                NP               57.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_species_MP  =  58\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               87.07           14598  \n",
       "1                NP               87.07           14598  \n",
       "2                NP               57.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_species_MP_counts  =  58\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['NP'], ['T2'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_NP_T2_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a53bda85-177b-4514-ac06-5a8fa06927bd",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c131cff3-8586-4f4c-a4f8-7c48d5fc7c65",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               56.25           14598  \n",
       "1              Prog               56.25           14598  \n",
       "2              Prog               64.11           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_species_MP_counts  =  40\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['Prog'], ['T1'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_Prog_T1_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "063ea34d-285c-45a4-8816-6f24039051fa",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5f3932df-afb6-4fa1-af69-ddcd164ec99e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               65.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_species_MP_counts  =  51\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['Prog'], ['T2'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_Prog_T2_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a504932-45d4-4a34-8d75-9a45a329b447",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "comad_EAC_figures",
   "language": "python",
   "name": "comad_eac_figures"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
