{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "86c88da5-4793-43df-a099-ab477cce2cac",
   "metadata": {},
   "source": [
    "# Genearate Data\n",
    "Generate all data needed to run analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0be072a-c289-436d-83e6-90dc2bff85cf",
   "metadata": {},
   "source": [
    "### Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "49bd78dd-82b0-4930-9e30-db578c2ef177",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import qiime2 as q2\n",
    "from qiime2.plugins import taxa\n",
    "\n",
    "from python_scripts.import_split_biom import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f25effc8-530a-4fe6-bd78-b564b9468616",
   "metadata": {},
   "source": [
    "### Datatypes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db9670b5-42d1-4088-9221-96a7569a585a",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### EAC \n",
    "From ICGC, not exact paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "db6ce605-5d5e-41a0-8f41-cf496e36b1e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_species_WOL  =  98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>filename</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>project</th>\n",
       "      <th>project_code</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>submitted_donor_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14521.006cdcd42d55e07a2fd1ad2d31dd1a87</td>\n",
       "      <td>unknown</td>\n",
       "      <td>relapse</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>299.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>distant recurrence/metastasis</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam</td>\n",
       "      <td>DO50381</td>\n",
       "      <td>SA528854</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-RS-031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>no</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>65.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>712.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-SH-071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>unknown</td>\n",
       "      <td>stable</td>\n",
       "      <td>67.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>401.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>alive</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-ZZ-019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              sample_name  \\\n",
       "0  14521.006cdcd42d55e07a2fd1ad2d31dd1a87   \n",
       "1  14521.0337efea9b59b88f804ea384c356377c   \n",
       "2  14521.04a446c6a660694dafe43ccaf1db7162   \n",
       "\n",
       "  cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "0                              unknown                      relapse   \n",
       "1                                   no       no evidence of disease   \n",
       "2                              unknown                       stable   \n",
       "\n",
       "   donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "0                    58.0                     58.0   \n",
       "1                    65.0                     65.0   \n",
       "2                    67.0                      NaN   \n",
       "\n",
       "   donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "0                        58.0                 C15.5   \n",
       "1                        66.0                 C15.5   \n",
       "2                        68.0                 C15.5   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_relapse_interval  \\\n",
       "0                            299.0                   123.0   \n",
       "1                            712.0                     NaN   \n",
       "2                            401.0                     NaN   \n",
       "\n",
       "              donor_relapse_type  ... donor_vital_status  \\\n",
       "0  distant recurrence/metastasis  ...           deceased   \n",
       "1                            NaN  ...           deceased   \n",
       "2                            NaN  ...              alive   \n",
       "\n",
       "                                        filename icgc_donor_id icgc_sample_id  \\\n",
       "0  006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam       DO50381       SA528854   \n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam       DO50389       SA528947   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam       DO50412       SA528884   \n",
       "\n",
       "  prior_malignancy  project project_code qiita_study_id  \\\n",
       "0          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "1          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "2          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "\n",
       "  study_donor_involved_in submitted_donor_id  \n",
       "0                   PCAWG      OCCAMS-RS-031  \n",
       "1                   PCAWG      OCCAMS-SH-071  \n",
       "2                   PCAWG      OCCAMS-ZZ-019  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_genome_WOL  =  98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>filename</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>project</th>\n",
       "      <th>project_code</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>submitted_donor_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14521.006cdcd42d55e07a2fd1ad2d31dd1a87</td>\n",
       "      <td>unknown</td>\n",
       "      <td>relapse</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>299.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>distant recurrence/metastasis</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam</td>\n",
       "      <td>DO50381</td>\n",
       "      <td>SA528854</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-RS-031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>no</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>65.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>712.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-SH-071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>unknown</td>\n",
       "      <td>stable</td>\n",
       "      <td>67.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>401.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>alive</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-ZZ-019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              sample_name  \\\n",
       "0  14521.006cdcd42d55e07a2fd1ad2d31dd1a87   \n",
       "1  14521.0337efea9b59b88f804ea384c356377c   \n",
       "2  14521.04a446c6a660694dafe43ccaf1db7162   \n",
       "\n",
       "  cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "0                              unknown                      relapse   \n",
       "1                                   no       no evidence of disease   \n",
       "2                              unknown                       stable   \n",
       "\n",
       "   donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "0                    58.0                     58.0   \n",
       "1                    65.0                     65.0   \n",
       "2                    67.0                      NaN   \n",
       "\n",
       "   donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "0                        58.0                 C15.5   \n",
       "1                        66.0                 C15.5   \n",
       "2                        68.0                 C15.5   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_relapse_interval  \\\n",
       "0                            299.0                   123.0   \n",
       "1                            712.0                     NaN   \n",
       "2                            401.0                     NaN   \n",
       "\n",
       "              donor_relapse_type  ... donor_vital_status  \\\n",
       "0  distant recurrence/metastasis  ...           deceased   \n",
       "1                            NaN  ...           deceased   \n",
       "2                            NaN  ...              alive   \n",
       "\n",
       "                                        filename icgc_donor_id icgc_sample_id  \\\n",
       "0  006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam       DO50381       SA528854   \n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam       DO50389       SA528947   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam       DO50412       SA528884   \n",
       "\n",
       "  prior_malignancy  project project_code qiita_study_id  \\\n",
       "0          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "1          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "2          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "\n",
       "  study_donor_involved_in submitted_donor_id  \n",
       "0                   PCAWG      OCCAMS-RS-031  \n",
       "1                   PCAWG      OCCAMS-SH-071  \n",
       "2                   PCAWG      OCCAMS-ZZ-019  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147030_species_WOL.biom',\n",
    "                'EAC_species_WOL')\n",
    "\n",
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147032_genome_WOL.biom',\n",
    "                'EAC_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "618c6164-aff2-44f1-9e22-e9a8463cf8c1",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Normal/Healthy & GERD Esoph Brushings  \n",
    "Deshpande, N. P., Riordan, S. M., Castaño-Rodríguez, N., Wilkins, M. R. & Kaakoush, N. O. Signatures within the esophageal microbiome are associated with host genetics, age, and disease. Microbiome vol. 6 (2018) https://doi.org/10.1186/s40168-018-0611-4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c90fcf1f-f6bd-4325-b451-35e5603cfe5b",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Convert to various types form Qiime2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "669b0dc9-4f6a-4d8b-b03f-c638426c9acf",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Table\n",
    "table = q2.Artifact.import_data('FeatureTable[Frequency]', os.getcwd() + '/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom')\n",
    "\n",
    "#WOL Taxonomy\n",
    "taxdf = pd.read_csv('/Users/cguccion/Dropbox/Storage/HelpfulLabDocs/taxonomy_trees/WOL/lineages.txt', sep='\\t', index_col=0, header=None)\n",
    "taxdf.columns = ['Taxon']\n",
    "taxdf.index.name = 'Feature ID'\n",
    "taxonomy = q2.Artifact.import_data('FeatureData[Taxonomy]', taxdf)\n",
    "\n",
    "collapsed_table = taxa.methods.collapse(table=table, taxonomy=taxonomy, level=7).collapsed_table\n",
    "\n",
    "#display(collapsed_table.view(pd.DataFrame).T)\n",
    "Artifact.export_data(collapsed_table, 'species_WOL')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5386f24e-43d2-48f5-8be7-0e212052e620",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Normal / Healthy Esophgous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "57a7b7f5-6ec6-4acd-bd7b-c4c625b3f0f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_genome_RS210  =  50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_species_WOL  =  50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_genome_WOL  =  50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169129_perGenome_RS210.biom',\n",
    "                       'normal_genome_RS210')\n",
    "\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/species_WOL/feature-table.biom',\n",
    "                       'normal_species_WOL')\n",
    "\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom',\n",
    "                       'normal_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6f428a7-8d58-4b0a-8faa-bfab0942a8dd",
   "metadata": {},
   "source": [
    "##### GERD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3f7efc1f-0613-4276-a652-27eecc095c60",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_genome_RS210  =  29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_species_WOL  =  29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_genome_WOL  =  29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169129_perGenome_RS210.biom',\n",
    "                       'GERD_genome_RS210')\n",
    "\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/species_WOL/feature-table.biom',\n",
    "                       'GERD_species_WOL')\n",
    "\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() + '/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom',\n",
    "                       'GERD_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e6d5b2c-3936-4d51-94c3-c60ce4ab38ce",
   "metadata": {
    "tags": []
   },
   "source": [
    "### From Sam: Barrett's Esophgous Progressors/Non Progressors: using MetaPhlan4\n",
    "Paulson, T. G. et al. Somatic whole genome dynamics of precancer in Barrett’s esophagus reveals features associated with disease progression. Nature Communications vol. 13 (2022) https://doi.org/10.1038/s41467-022-29767-7"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90118ef0-c827-48f1-b4a0-8467e99759df",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a8a9880a-bf07-43ec-8caa-19ec988fa8ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               87.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_species_MP  =  202\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               87.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_species_MP_counts  =  202\n"
     ]
    }
   ],
   "source": [
    "#Relative Abudnace\n",
    "Sam_BE_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "72fe3d28-81a2-474b-880e-400e9aba02a9",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "58ff07c9-4c06-416e-9afc-979fcc299299",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               87.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_species_MP  =  111\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               87.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_species_MP_counts  =  111\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['NP',], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_NP_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['NP'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_NP_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dfeb2eb4-eaf7-4448-8175-a994626ff617",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "247770e3-12fe-4889-9ebd-2f7f4232a498",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               53.23           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_species_MP  =  53\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               53.23           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_species_MP_counts  =  53\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['NP',], ['T1'], \n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_NP_T1_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['NP'], ['T1'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_NP_T1_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cc98a2e-80d4-4c39-82ff-03fc036daa8f",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d02eac75-3ef0-409d-90ae-df96522e04e5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               87.07           14598  \n",
       "1                NP               87.07           14598  \n",
       "2                NP               57.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_species_MP  =  58\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               87.07           14598  \n",
       "1                NP               87.07           14598  \n",
       "2                NP               57.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_species_MP_counts  =  58\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['NP',], ['T2'], \n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_NP_T2_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['NP'], ['T2'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_NP_T2_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae337c1f-f4aa-4686-b3fc-ecab1010e916",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1356a3c6-ad57-4886-8295-215590e0046b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               56.25           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_species_MP  =  91\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               56.25           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_species_MP_counts  =  91\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['Prog',], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_Prog_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_Prog_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a53bda85-177b-4514-ac06-5a8fa06927bd",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c131cff3-8586-4f4c-a4f8-7c48d5fc7c65",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               56.25           14598  \n",
       "1              Prog               56.25           14598  \n",
       "2              Prog               64.11           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_species_MP  =  40\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               56.25           14598  \n",
       "1              Prog               56.25           14598  \n",
       "2              Prog               64.11           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_species_MP_counts  =  40\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['Prog',], ['T1'], \n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_Prog_T1_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['Prog'], ['T1'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_Prog_T1_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "063ea34d-285c-45a4-8816-6f24039051fa",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5f3932df-afb6-4fa1-af69-ddcd164ec99e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               65.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_species_MP  =  51\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               65.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_species_MP_counts  =  51\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['Prog',], ['T2'], \n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_Prog_T2_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['Prog'], ['T2'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_Prog_T2_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "230a33d7-437a-4f89-b456-54593cca3d48",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Barrett's Esophgous Progressors/Non Progressors\n",
    "Paulson, T. G. et al. Somatic whole genome dynamics of precancer in Barrett’s esophagus reveals features associated with disease progression. Nature Communications vol. 13 (2022) https://doi.org/10.1038/s41467-022-29767-7"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af7c8631-d14b-45e6-96ae-df61afaa05df",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "972a186b-de3d-4ba6-9620-0c5cbe579e7f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_genome_RS210  =  302\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_species_WOL  =  316\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_genome_WOL  =  316\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_genome_RS210')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_species_WOL')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3ecf352-905f-44c5-8778-9c98cc2abc8b",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "36bb1096-c2a1-4d8d-93f1-bd8baa17f53f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_genome_RS210  =  150\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_species_WOL  =  158\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_NP_genome_RS210')\n",
    "\n",
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_NP_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea5c4296-def2-4229-b2ae-27670bfb85be",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7c956478-7b19-4de9-b969-2e0c8df8fb5e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               53.23           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_genome_RS210  =  76\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               53.23           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_species_WOL  =  80\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_NP_T1_genome_RS210')\n",
    "\n",
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_NP_T1_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aef4c92a-0a61-458e-81a0-929b46fb1315",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "29fa4ebc-f730-4c2d-81b9-0971436ad38b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               87.07           14598  \n",
       "1                NP               87.07           14598  \n",
       "2                NP               57.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_genome_RS210  =  74\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               87.07           14598  \n",
       "1                NP               87.07           14598  \n",
       "2                NP               57.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_species_WOL  =  78\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_NP_T2_genome_RS210')\n",
    "\n",
    "be_data_split(['NP'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_NP_T2_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2327758-9e16-465d-b3c3-a864a8d76dc6",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "01050185-76d7-44d0-9093-02e371e74830",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_genome_RS210  =  152\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_species_WOL  =  158\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['Prog'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_Prog_genome_RS210')\n",
    "\n",
    "be_data_split(['Prog'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_Prog_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8037a84d-adc1-4bf9-b6d3-2d837afbf1ef",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d9a3dfa3-8d7d-40ac-8e9d-79293ce8c321",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               56.25           14598  \n",
       "1              Prog               56.25           14598  \n",
       "2              Prog               64.11           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_genome_RS210  =  76\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               56.25           14598  \n",
       "1              Prog               56.25           14598  \n",
       "2              Prog               64.11           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_species_WOL  =  79\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_Prog_T1_genome_RS210')\n",
    "\n",
    "be_data_split(['Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_Prog_T1_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "532903b1-2dcc-408f-90c5-709a875b7646",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "cef0e3fa-fb27-4b5c-908f-9c00fae00760",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               65.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_genome_RS210  =  76\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               65.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_species_WOL  =  79\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               65.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_genome_WOL  =  79\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_Prog_T2_genome_RS210')\n",
    "\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_Prog_T2_species_WOL')\n",
    "\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_Prog_T2_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47c91944-f659-4117-8a8c-b4493c920159",
   "metadata": {},
   "source": [
    "##### Barrett's Esophgous T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "19463279-9ce3-4958-a19f-1312813c6e98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1065.BE.03      Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1   42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T1_genome_RS210  =  152\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1065.BE.03      Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1   42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T1_species_WOL  =  159\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1065.BE.03      Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1   42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T1_genome_WOL  =  159\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP','Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_T1_genome_RS210')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_T1_species_WOL')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_T1_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "209583ec-8ee9-4f90-b5aa-4828a0ddc8d3",
   "metadata": {},
   "source": [
    "##### Barrett's Esophgous T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "03e25eea-2b82-48a0-a7c1-66ec6813094d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04  White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1065.BE.01      Other_Unknown      M         42         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1   42-24954N-53L-GST                5             24283   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2              Prog               57.65           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T2_genome_RS210  =  150\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04  White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1065.BE.01      Other_Unknown      M         42         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1   42-24954N-53L-GST                5             24283   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2              Prog               57.65           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T2_species_WOL  =  157\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04  White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1065.BE.01      Other_Unknown      M         42         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1   42-24954N-53L-GST                5             24283   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2              Prog               57.65           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T2_genome_WOL  =  157\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP','Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_T2_genome_RS210')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_T2_species_WOL')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_T2_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4de4bd81-f239-4139-be7f-4cf8cf58fdb0",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Normal, BE, EAC Tissue \n",
    "Ross-Innes, C. S. et al. Whole-genome sequencing provides new insights into the clonal architecture of Barrett’s esophagus and esophageal adenocarcinoma. Nature Genetics vol. 47 1038–1046 (2015) https://doi.org/10.1038/ng.3357"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93ea269c-b34c-4929-8f25-aab51638bd0a",
   "metadata": {},
   "source": [
    "##### Normal, BE, EAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b30e7f41-f4f6-4af1-afed-acb2f6ded801",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in ALL_Ross_combo_species_WOL  =  88\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Normal', 'Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_combo_R1_R2_paired_unpaired.biom',\n",
    "                         'ALL_Ross_combo_species_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9ae58a46-601f-47ee-a37b-ca45feca700a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in ALL_Ross_paired_species_WOL  =  88\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in ALL_Ross_paired_genome_WOL  =  88\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Normal', 'Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'ALL_Ross_paired_species_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['Normal', 'Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'ALL_Ross_paired_genome_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "750caafd-6f91-4db2-be75-00615ddcbb2d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in ALL_Ross_unpaired_species_WOL  =  88\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Normal', 'Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_unpaired.biom',\n",
    "                         'ALL_Ross_unpaired_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "faf44e66-26a9-4bbd-b773-97dad3f1cade",
   "metadata": {},
   "source": [
    "##### BE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "326381c7-499e-4722-bd04-09f663075f42",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_combo_species_WOL  =  35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_combo_R1_R2_paired_unpaired.biom',\n",
    "                         'BE_Ross_combo_species_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5e62bcf1-57a4-4be9-862b-d5927f906bc3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_paired_species_WOL  =  35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_paired_genome_WOL  =  35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'BE_Ross_paired_species_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'BE_Ross_paired_genome_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "df2bbcea-9caf-4927-9a66-49dcf4a79068",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_unpaired_species_WOL  =  35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_unpaired.biom',\n",
    "                         'BE_Ross_unpaired_species_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "adb1ea91-0f36-4fe6-b170-09a07027ae72",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_paired_species_WOL  =  30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_paired_genome_WOL  =  30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'EAC_Ross_paired_species_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'EAC_Ross_paired_genome_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a4f79443-a5d0-4304-b20a-b1ed2d80d424",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_combo_species_WOL  =  30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split('EAC',\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_combo_R1_R2_paired_unpaired.biom',\n",
    "                         'EAC_Ross_combo_species_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c578f6fe-b9d1-4762-8bde-831c1f79985c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Normal_10_EGAF00000530908</td>\n",
       "      <td>10</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>807,792,398</td>\n",
       "      <td>92.49</td>\n",
       "      <td>296.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.245</td>\n",
       "      <td>48.64</td>\n",
       "      <td>0.968</td>\n",
       "      <td>3523708.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Normal_11_EGAF00000586533</td>\n",
       "      <td>11</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>639,793,060</td>\n",
       "      <td>92.22</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.88</td>\n",
       "      <td>0.241</td>\n",
       "      <td>37.77</td>\n",
       "      <td>0.969</td>\n",
       "      <td>3541117.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Normal_12_EGAF00000525507</td>\n",
       "      <td>12</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>559,984,669</td>\n",
       "      <td>88.64</td>\n",
       "      <td>299.0</td>\n",
       "      <td>5.09</td>\n",
       "      <td>0.257</td>\n",
       "      <td>32.15</td>\n",
       "      <td>0.964</td>\n",
       "      <td>3491591.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample_name Patient Sample_type  \\\n",
       "0  Normal_10_EGAF00000530908      10      Normal   \n",
       "1  Normal_11_EGAF00000586533      11      Normal   \n",
       "2  Normal_12_EGAF00000525507      12      Normal   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           807,792,398            92.49               296.0      5.03   \n",
       "1           639,793,060            92.22               315.0      4.88   \n",
       "2           559,984,669            88.64               299.0      5.09   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   48.64            0.968        3523708.0   \n",
       "1           0.241                   37.77            0.969        3541117.0   \n",
       "2           0.257                   32.15            0.964        3491591.0   \n",
       "\n",
       "   Sample purity  \n",
       "0            NaN  \n",
       "1            NaN  \n",
       "2            NaN  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in Norm_Ross_combo_species_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split('Normal',\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_combo_R1_R2_paired_unpaired.biom',\n",
    "                         'Norm_Ross_combo_species_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "594e7bef-0d76-4f1c-a598-f8288733cbde",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Normal_10_EGAF00000530908</td>\n",
       "      <td>10</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>807,792,398</td>\n",
       "      <td>92.49</td>\n",
       "      <td>296.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.245</td>\n",
       "      <td>48.64</td>\n",
       "      <td>0.968</td>\n",
       "      <td>3523708.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Normal_11_EGAF00000586533</td>\n",
       "      <td>11</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>639,793,060</td>\n",
       "      <td>92.22</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.88</td>\n",
       "      <td>0.241</td>\n",
       "      <td>37.77</td>\n",
       "      <td>0.969</td>\n",
       "      <td>3541117.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Normal_12_EGAF00000525507</td>\n",
       "      <td>12</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>559,984,669</td>\n",
       "      <td>88.64</td>\n",
       "      <td>299.0</td>\n",
       "      <td>5.09</td>\n",
       "      <td>0.257</td>\n",
       "      <td>32.15</td>\n",
       "      <td>0.964</td>\n",
       "      <td>3491591.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample_name Patient Sample_type  \\\n",
       "0  Normal_10_EGAF00000530908      10      Normal   \n",
       "1  Normal_11_EGAF00000586533      11      Normal   \n",
       "2  Normal_12_EGAF00000525507      12      Normal   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           807,792,398            92.49               296.0      5.03   \n",
       "1           639,793,060            92.22               315.0      4.88   \n",
       "2           559,984,669            88.64               299.0      5.09   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   48.64            0.968        3523708.0   \n",
       "1           0.241                   37.77            0.969        3541117.0   \n",
       "2           0.257                   32.15            0.964        3491591.0   \n",
       "\n",
       "   Sample purity  \n",
       "0            NaN  \n",
       "1            NaN  \n",
       "2            NaN  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in Norm_Ross_paired_species_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Normal'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'Norm_Ross_paired_species_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "69b549fb-dad1-4ffb-be59-e72314be1416",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in Norm_BE_Ross_paired_species_WOL  =  58\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Normal', 'Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'Norm_BE_Ross_paired_species_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c0f7dd8e-1592-4f3b-bbe9-008a23c0f65b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_EAC_Ross_paired_species_WOL  =  65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'BE_EAC_Ross_paired_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6502089f-bc50-4e48-8bba-5d427003b747",
   "metadata": {},
   "source": [
    "#### EAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "393f423b-8e66-4c07-a92f-7119585e4501",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_species_WOL  =  98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eac_data_split(os.getcwd() + '/qiita_downloads/qiita14521_EAC_ICGC/147032_perGenome_WOL.biom',\n",
    "                         'EAC_species_WOL')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "comad_EAC_figures",
   "language": "python",
   "name": "comad_eac_figures"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
