{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "86c88da5-4793-43df-a099-ab477cce2cac",
   "metadata": {},
   "source": [
    "# Genearate Data\n",
    "Generate all data needed to run analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0be072a-c289-436d-83e6-90dc2bff85cf",
   "metadata": {},
   "source": [
    "### Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "49bd78dd-82b0-4930-9e30-db578c2ef177",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import qiime2 as q2\n",
    "from qiime2.plugins import taxa\n",
    "\n",
    "from python_scripts.import_split_biom import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f25effc8-530a-4fe6-bd78-b564b9468616",
   "metadata": {},
   "source": [
    "### Datatypes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83dc865c-d46a-42cd-91f1-57d5b4f03c0e",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### EAC ICGC\n",
    "\n",
    "\"Pan-Cancer Analysis of Whole Genomes.\" Nature, vol. 578, no. 7793, 2020, pp. 82-93,  https://doi.org/10.1038/s41586-020-1969-6. Accessed 17 Jan. 2024.\n",
    "\n",
    "(Qiita: 14857)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa3f214f-02e9-4038-823f-57a7bc4c5791",
   "metadata": {
    "tags": []
   },
   "source": [
    "###### Used in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a13a8e9e-c75c-49e0-aa98-2f317bd506b9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>project</th>\n",
       "      <th>filename</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>donor_tumour_staging_system_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis_supplemental</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14857.14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deceased</td>\n",
       "      <td>...</td>\n",
       "      <td>65.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>66.0</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>712.0</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>no</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:1b</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14857.14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alive</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>68.0</td>\n",
       "      <td>stable</td>\n",
       "      <td>401.0</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>TNM6</td>\n",
       "      <td>UICC:3</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14857.14521.04ceb5f141916f7fc091a5e0aa51ada3</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>04ceb5f141916f7fc091a5e0aa51ada3.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50406</td>\n",
       "      <td>SA528949</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alive</td>\n",
       "      <td>...</td>\n",
       "      <td>74.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>76.0</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>767.0</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>no</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:1b</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    sample_name  project  \\\n",
       "1  14857.14521.0337efea9b59b88f804ea384c356377c  ESAD-UK   \n",
       "2  14857.14521.04a446c6a660694dafe43ccaf1db7162  ESAD-UK   \n",
       "3  14857.14521.04ceb5f141916f7fc091a5e0aa51ada3  ESAD-UK   \n",
       "\n",
       "                                        filename donor_sex project_code  \\\n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam      male      ESAD-UK   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam      male      ESAD-UK   \n",
       "3  04ceb5f141916f7fc091a5e0aa51ada3.filtered.bam      male      ESAD-UK   \n",
       "\n",
       "  icgc_donor_id icgc_sample_id prior_malignancy donor_relapse_type  \\\n",
       "1       DO50389       SA528947          unknown                NaN   \n",
       "2       DO50412       SA528884          unknown                NaN   \n",
       "3       DO50406       SA528949          unknown                NaN   \n",
       "\n",
       "  donor_vital_status  ... donor_age_at_enrollment  study_donor_involved_in  \\\n",
       "1           deceased  ...                    65.0                    PCAWG   \n",
       "2              alive  ...                     NaN                    PCAWG   \n",
       "3              alive  ...                    74.0                    PCAWG   \n",
       "\n",
       "  donor_age_at_last_followup  disease_status_last_followup  \\\n",
       "1                       66.0        no evidence of disease   \n",
       "2                       68.0                        stable   \n",
       "3                       76.0        no evidence of disease   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_tumour_stage_at_diagnosis  \\\n",
       "1                            712.0                           T2N0M0   \n",
       "2                            401.0                           T3N1M0   \n",
       "3                            767.0                           T2N0M0   \n",
       "\n",
       "  cancer_history_first_degree_relative  \\\n",
       "1                                   no   \n",
       "2                              unknown   \n",
       "3                                   no   \n",
       "\n",
       "   donor_tumour_staging_system_at_diagnosis  \\\n",
       "1                                      TNM7   \n",
       "2                                      TNM6   \n",
       "3                                      TNM7   \n",
       "\n",
       "  donor_tumour_stage_at_diagnosis_supplemental  qiita_study_id  \n",
       "1                                      UICC:1b           14857  \n",
       "2                                       UICC:3           14857  \n",
       "3                                      UICC:1b           14857  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_ICGC_ginnyR_species_WOL_scrubbed  =  83\n",
      "<artifact: FeatureTable[Frequency] uuid: f93e08ee-a9f5-488c-8afd-d0d8a32ab2b9>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:133: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eac_14857_data_split(os.getcwd() +'/qiita_downloads/qiita14857_EAC_ICGC/162938_species_WoLr1_scrubbed/feature-table.biom',\n",
    "                'EAC_ICGC_ginnyR_species_WOL_scrubbed', Ginny_list=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "45b99c23-3c83-4507-a3f4-db23a0888488",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>project</th>\n",
       "      <th>filename</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>donor_tumour_staging_system_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis_supplemental</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14857.14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deceased</td>\n",
       "      <td>...</td>\n",
       "      <td>65.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>66.0</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>712.0</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>no</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:1b</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14857.14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alive</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>68.0</td>\n",
       "      <td>stable</td>\n",
       "      <td>401.0</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>TNM6</td>\n",
       "      <td>UICC:3</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14857.14521.04ceb5f141916f7fc091a5e0aa51ada3</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>04ceb5f141916f7fc091a5e0aa51ada3.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50406</td>\n",
       "      <td>SA528949</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alive</td>\n",
       "      <td>...</td>\n",
       "      <td>74.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>76.0</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>767.0</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>no</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:1b</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    sample_name  project  \\\n",
       "1  14857.14521.0337efea9b59b88f804ea384c356377c  ESAD-UK   \n",
       "2  14857.14521.04a446c6a660694dafe43ccaf1db7162  ESAD-UK   \n",
       "3  14857.14521.04ceb5f141916f7fc091a5e0aa51ada3  ESAD-UK   \n",
       "\n",
       "                                        filename donor_sex project_code  \\\n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam      male      ESAD-UK   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam      male      ESAD-UK   \n",
       "3  04ceb5f141916f7fc091a5e0aa51ada3.filtered.bam      male      ESAD-UK   \n",
       "\n",
       "  icgc_donor_id icgc_sample_id prior_malignancy donor_relapse_type  \\\n",
       "1       DO50389       SA528947          unknown                NaN   \n",
       "2       DO50412       SA528884          unknown                NaN   \n",
       "3       DO50406       SA528949          unknown                NaN   \n",
       "\n",
       "  donor_vital_status  ... donor_age_at_enrollment  study_donor_involved_in  \\\n",
       "1           deceased  ...                    65.0                    PCAWG   \n",
       "2              alive  ...                     NaN                    PCAWG   \n",
       "3              alive  ...                    74.0                    PCAWG   \n",
       "\n",
       "  donor_age_at_last_followup  disease_status_last_followup  \\\n",
       "1                       66.0        no evidence of disease   \n",
       "2                       68.0                        stable   \n",
       "3                       76.0        no evidence of disease   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_tumour_stage_at_diagnosis  \\\n",
       "1                            712.0                           T2N0M0   \n",
       "2                            401.0                           T3N1M0   \n",
       "3                            767.0                           T2N0M0   \n",
       "\n",
       "  cancer_history_first_degree_relative  \\\n",
       "1                                   no   \n",
       "2                              unknown   \n",
       "3                                   no   \n",
       "\n",
       "   donor_tumour_staging_system_at_diagnosis  \\\n",
       "1                                      TNM7   \n",
       "2                                      TNM6   \n",
       "3                                      TNM7   \n",
       "\n",
       "  donor_tumour_stage_at_diagnosis_supplemental  qiita_study_id  \n",
       "1                                      UICC:1b           14857  \n",
       "2                                       UICC:3           14857  \n",
       "3                                      UICC:1b           14857  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_ICGC_ginnyR_species_WOL_scrubbed_rmBad  =  83\n",
      "<artifact: FeatureTable[Frequency] uuid: 3d723418-2280-4477-971f-eb9aa9034a25>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:133: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eac_14857_data_split(os.getcwd() +'/qiita_downloads/qiita14857_EAC_ICGC/162938_species_WoLr1_scrubbed/feature-table.biom',\n",
    "                'EAC_ICGC_ginnyR_species_WOL_scrubbed_rmBad', Ginny_list=True, remove_bad_microbes=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "81c9408a-c87a-4165-b797-317a148cd370",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>project</th>\n",
       "      <th>filename</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>donor_tumour_staging_system_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis_supplemental</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14857.14521.006cdcd42d55e07a2fd1ad2d31dd1a87</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50381</td>\n",
       "      <td>SA528854</td>\n",
       "      <td>unknown</td>\n",
       "      <td>distant recurrence/metastasis</td>\n",
       "      <td>deceased</td>\n",
       "      <td>...</td>\n",
       "      <td>58.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>58.0</td>\n",
       "      <td>relapse</td>\n",
       "      <td>299.0</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:3a</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14857.14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deceased</td>\n",
       "      <td>...</td>\n",
       "      <td>65.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>66.0</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>712.0</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>no</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:1b</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14857.14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alive</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>68.0</td>\n",
       "      <td>stable</td>\n",
       "      <td>401.0</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>TNM6</td>\n",
       "      <td>UICC:3</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    sample_name  project  \\\n",
       "0  14857.14521.006cdcd42d55e07a2fd1ad2d31dd1a87  ESAD-UK   \n",
       "1  14857.14521.0337efea9b59b88f804ea384c356377c  ESAD-UK   \n",
       "2  14857.14521.04a446c6a660694dafe43ccaf1db7162  ESAD-UK   \n",
       "\n",
       "                                        filename donor_sex project_code  \\\n",
       "0  006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam      male      ESAD-UK   \n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam      male      ESAD-UK   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam      male      ESAD-UK   \n",
       "\n",
       "  icgc_donor_id icgc_sample_id prior_malignancy  \\\n",
       "0       DO50381       SA528854          unknown   \n",
       "1       DO50389       SA528947          unknown   \n",
       "2       DO50412       SA528884          unknown   \n",
       "\n",
       "              donor_relapse_type donor_vital_status  ...  \\\n",
       "0  distant recurrence/metastasis           deceased  ...   \n",
       "1                            NaN           deceased  ...   \n",
       "2                            NaN              alive  ...   \n",
       "\n",
       "  donor_age_at_enrollment  study_donor_involved_in donor_age_at_last_followup  \\\n",
       "0                    58.0                    PCAWG                       58.0   \n",
       "1                    65.0                    PCAWG                       66.0   \n",
       "2                     NaN                    PCAWG                       68.0   \n",
       "\n",
       "   disease_status_last_followup  donor_interval_of_last_followup  \\\n",
       "0                       relapse                            299.0   \n",
       "1        no evidence of disease                            712.0   \n",
       "2                        stable                            401.0   \n",
       "\n",
       "   donor_tumour_stage_at_diagnosis cancer_history_first_degree_relative  \\\n",
       "0                           T3N1M0                              unknown   \n",
       "1                           T2N0M0                                   no   \n",
       "2                           T3N1M0                              unknown   \n",
       "\n",
       "   donor_tumour_staging_system_at_diagnosis  \\\n",
       "0                                      TNM7   \n",
       "1                                      TNM7   \n",
       "2                                      TNM6   \n",
       "\n",
       "  donor_tumour_stage_at_diagnosis_supplemental  qiita_study_id  \n",
       "0                                      UICC:3a           14857  \n",
       "1                                      UICC:1b           14857  \n",
       "2                                       UICC:3           14857  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_ICGC_wo-RI_species_WOL_scrubbed_rmBad  =  97\n",
      "<artifact: FeatureTable[Frequency] uuid: c5d588cb-fe0d-49b7-b529-3f7484f23196>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#NOTE: This was used prior to May 2024 -- but a little tricky in how things were removed so going to clean it up now\n",
    "#since we have to re-do with exact microbes from Ginny\n",
    "eac_14857_data_split(os.getcwd() +'/qiita_downloads/qiita14857_EAC_ICGC/162938_species_WoLr1_scrubbed/feature-table.biom',\n",
    "                'EAC_ICGC_wo-RI_species_WOL_scrubbed_rmBad',\n",
    "                     remove_bad_microbes=True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a7d4ed91-a986-4057-9d94-7467f143a883",
   "metadata": {
    "tags": []
   },
   "source": [
    "###### Not in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d54d992d-a116-4f41-ba55-9fd5e44044ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "eac_14857_data_split(os.getcwd() +'/qiita_downloads/qiita14857_EAC_ICGC/162938_species_WoLr1_scrubbed/feature-table.biom',\n",
    "                'EAC_ICGC_wo-RI_species_WOL_scrubbed')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "79c1aa92-b518-41fe-8a7f-dde6ca061ccf",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>project</th>\n",
       "      <th>filename</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>donor_tumour_staging_system_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis_supplemental</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14857.14521.006cdcd42d55e07a2fd1ad2d31dd1a87</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50381</td>\n",
       "      <td>SA528854</td>\n",
       "      <td>unknown</td>\n",
       "      <td>distant recurrence/metastasis</td>\n",
       "      <td>deceased</td>\n",
       "      <td>...</td>\n",
       "      <td>58.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>58.0</td>\n",
       "      <td>relapse</td>\n",
       "      <td>299.0</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:3a</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14857.14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deceased</td>\n",
       "      <td>...</td>\n",
       "      <td>65.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>66.0</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>712.0</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>no</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:1b</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14857.14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alive</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>68.0</td>\n",
       "      <td>stable</td>\n",
       "      <td>401.0</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>TNM6</td>\n",
       "      <td>UICC:3</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    sample_name  project  \\\n",
       "0  14857.14521.006cdcd42d55e07a2fd1ad2d31dd1a87  ESAD-UK   \n",
       "1  14857.14521.0337efea9b59b88f804ea384c356377c  ESAD-UK   \n",
       "2  14857.14521.04a446c6a660694dafe43ccaf1db7162  ESAD-UK   \n",
       "\n",
       "                                        filename donor_sex project_code  \\\n",
       "0  006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam      male      ESAD-UK   \n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam      male      ESAD-UK   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam      male      ESAD-UK   \n",
       "\n",
       "  icgc_donor_id icgc_sample_id prior_malignancy  \\\n",
       "0       DO50381       SA528854          unknown   \n",
       "1       DO50389       SA528947          unknown   \n",
       "2       DO50412       SA528884          unknown   \n",
       "\n",
       "              donor_relapse_type donor_vital_status  ...  \\\n",
       "0  distant recurrence/metastasis           deceased  ...   \n",
       "1                            NaN           deceased  ...   \n",
       "2                            NaN              alive  ...   \n",
       "\n",
       "  donor_age_at_enrollment  study_donor_involved_in donor_age_at_last_followup  \\\n",
       "0                    58.0                    PCAWG                       58.0   \n",
       "1                    65.0                    PCAWG                       66.0   \n",
       "2                     NaN                    PCAWG                       68.0   \n",
       "\n",
       "   disease_status_last_followup  donor_interval_of_last_followup  \\\n",
       "0                       relapse                            299.0   \n",
       "1        no evidence of disease                            712.0   \n",
       "2                        stable                            401.0   \n",
       "\n",
       "   donor_tumour_stage_at_diagnosis cancer_history_first_degree_relative  \\\n",
       "0                           T3N1M0                              unknown   \n",
       "1                           T2N0M0                                   no   \n",
       "2                           T3N1M0                              unknown   \n",
       "\n",
       "   donor_tumour_staging_system_at_diagnosis  \\\n",
       "0                                      TNM7   \n",
       "1                                      TNM7   \n",
       "2                                      TNM6   \n",
       "\n",
       "  donor_tumour_stage_at_diagnosis_supplemental  qiita_study_id  \n",
       "0                                      UICC:3a           14857  \n",
       "1                                      UICC:1b           14857  \n",
       "2                                       UICC:3           14857  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_ICGC_species_WOL_scrubbed  =  97\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eac_14857_data_split(os.getcwd() +'/qiita_downloads/qiita14857_EAC_ICGC/162938_species_WoLr1_scrubbed/feature-table.biom',\n",
    "                'EAC_ICGC_species_WOL_scrubbed')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "60f156e8-e1b4-4daa-9f67-382fcc9e7b5b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>project</th>\n",
       "      <th>filename</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>donor_tumour_staging_system_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis_supplemental</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14857.14521.006cdcd42d55e07a2fd1ad2d31dd1a87</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50381</td>\n",
       "      <td>SA528854</td>\n",
       "      <td>unknown</td>\n",
       "      <td>distant recurrence/metastasis</td>\n",
       "      <td>deceased</td>\n",
       "      <td>...</td>\n",
       "      <td>58.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>58.0</td>\n",
       "      <td>relapse</td>\n",
       "      <td>299.0</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:3a</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14857.14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deceased</td>\n",
       "      <td>...</td>\n",
       "      <td>65.0</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>66.0</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>712.0</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>no</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>UICC:1b</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14857.14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>male</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alive</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>68.0</td>\n",
       "      <td>stable</td>\n",
       "      <td>401.0</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>TNM6</td>\n",
       "      <td>UICC:3</td>\n",
       "      <td>14857</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    sample_name  project  \\\n",
       "0  14857.14521.006cdcd42d55e07a2fd1ad2d31dd1a87  ESAD-UK   \n",
       "1  14857.14521.0337efea9b59b88f804ea384c356377c  ESAD-UK   \n",
       "2  14857.14521.04a446c6a660694dafe43ccaf1db7162  ESAD-UK   \n",
       "\n",
       "                                        filename donor_sex project_code  \\\n",
       "0  006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam      male      ESAD-UK   \n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam      male      ESAD-UK   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam      male      ESAD-UK   \n",
       "\n",
       "  icgc_donor_id icgc_sample_id prior_malignancy  \\\n",
       "0       DO50381       SA528854          unknown   \n",
       "1       DO50389       SA528947          unknown   \n",
       "2       DO50412       SA528884          unknown   \n",
       "\n",
       "              donor_relapse_type donor_vital_status  ...  \\\n",
       "0  distant recurrence/metastasis           deceased  ...   \n",
       "1                            NaN           deceased  ...   \n",
       "2                            NaN              alive  ...   \n",
       "\n",
       "  donor_age_at_enrollment  study_donor_involved_in donor_age_at_last_followup  \\\n",
       "0                    58.0                    PCAWG                       58.0   \n",
       "1                    65.0                    PCAWG                       66.0   \n",
       "2                     NaN                    PCAWG                       68.0   \n",
       "\n",
       "   disease_status_last_followup  donor_interval_of_last_followup  \\\n",
       "0                       relapse                            299.0   \n",
       "1        no evidence of disease                            712.0   \n",
       "2                        stable                            401.0   \n",
       "\n",
       "   donor_tumour_stage_at_diagnosis cancer_history_first_degree_relative  \\\n",
       "0                           T3N1M0                              unknown   \n",
       "1                           T2N0M0                                   no   \n",
       "2                           T3N1M0                              unknown   \n",
       "\n",
       "   donor_tumour_staging_system_at_diagnosis  \\\n",
       "0                                      TNM7   \n",
       "1                                      TNM7   \n",
       "2                                      TNM6   \n",
       "\n",
       "  donor_tumour_stage_at_diagnosis_supplemental  qiita_study_id  \n",
       "0                                      UICC:3a           14857  \n",
       "1                                      UICC:1b           14857  \n",
       "2                                       UICC:3           14857  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "NameError",
     "evalue": "name 'zebra' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43meac_14857_data_split\u001b[49m\u001b[43m(\u001b[49m\u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mgetcwd\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m/qiita_downloads/qiita14857_EAC_ICGC/162938_none_WoLr1_scrubbed/feature-table.biom\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m      2\u001b[0m \u001b[43m                \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mEAC_ICGC_genome_WOL_scrubbed\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:233\u001b[0m, in \u001b[0;36meac_14857_data_split\u001b[0;34m(biom, fn, trim)\u001b[0m\n\u001b[1;32m    230\u001b[0m display(meta[:\u001b[38;5;241m3\u001b[39m])\n\u001b[1;32m    232\u001b[0m \u001b[38;5;66;03m#Create all files based on biom/meta_custom\u001b[39;00m\n\u001b[0;32m--> 233\u001b[0m data_split_helper(biom, meta, fn, zebra\u001b[38;5;241m=\u001b[39m\u001b[43mzebra\u001b[49m)\n\u001b[1;32m    235\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m()\n",
      "\u001b[0;31mNameError\u001b[0m: name 'zebra' is not defined"
     ]
    }
   ],
   "source": [
    "eac_14857_data_split(os.getcwd() +'/qiita_downloads/qiita14857_EAC_ICGC/162938_none_WoLr1_scrubbed/feature-table.biom',\n",
    "                'EAC_ICGC_genome_WOL_scrubbed')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "618c6164-aff2-44f1-9e22-e9a8463cf8c1",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Normal/Healthy & GERD Esoph Brushings  \n",
    "Deshpande, N. P., Riordan, S. M., CastaÃ±o-RodrÃ­guez, N., Wilkins, M. R. & Kaakoush, N. O. Signatures within the esophageal microbiome are associated with host genetics, age, and disease. Microbiome vol. 6 (2018) https://doi.org/10.1186/s40168-018-0611-4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c90fcf1f-f6bd-4325-b451-35e5603cfe5b",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Convert to various types form Qiime2 (only had to run once)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "669b0dc9-4f6a-4d8b-b03f-c638426c9acf",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Table\n",
    "table = q2.Artifact.import_data('FeatureTable[Frequency]', os.getcwd() + '/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom')\n",
    "\n",
    "#WOL Taxonomy\n",
    "taxdf = pd.read_csv('/Users/cguccion/Dropbox/Storage/HelpfulLabDocs/taxonomy_trees/WOL/lineages.txt', sep='\\t', index_col=0, header=None)\n",
    "taxdf.columns = ['Taxon']\n",
    "taxdf.index.name = 'Feature ID'\n",
    "taxonomy = q2.Artifact.import_data('FeatureData[Taxonomy]', taxdf)\n",
    "\n",
    "collapsed_table = taxa.methods.collapse(table=table, taxonomy=taxonomy, level=7).collapsed_table\n",
    "\n",
    "#display(collapsed_table.view(pd.DataFrame).T)\n",
    "Artifact.export_data(collapsed_table, 'species_WOL')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5386f24e-43d2-48f5-8be7-0e212052e620",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Normal / Healthy Esophgous"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4988e6a1-4daf-4b5a-912e-7118aa187e3d",
   "metadata": {},
   "source": [
    "###### Used in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d900e7b3-7b18-49ca-ae04-252b342631e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_species_WOL_scrubbed  =  50\n",
      "<artifact: FeatureTable[Frequency] uuid: f3ca9e18-c070-49fc-a3a3-6510ffc68ec7>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Wol Scrubbed\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_species_WOL_scrubbed/feature-table.biom',\n",
    "                       'normal_species_WOL_scrubbed')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "496c4c4d-a4bf-4695-aae2-ba11204cccad",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Not in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "57a7b7f5-6ec6-4acd-bd7b-c4c625b3f0f8",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_genome_WOL_scrubbed  =  50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_zebra_genome_WOL_scrubbed  =  50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_genome_RS210  =  50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_species_WOL  =  50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_genome_WOL  =  50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373134</td>\n",
       "      <td>70</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H96_S87_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H96</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373133</td>\n",
       "      <td>23</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H95_S86_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H95</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "1  14458.ERR2373134   70  YES      F     YES    Normal     H96_S87_SG   \n",
       "2  14458.ERR2373133   23  YES      F     YES    Normal     H95_S86_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0              NO                H98           14458  \n",
       "1              NO                H96           14458  \n",
       "2              NO                H95           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_zebra_genome_WOL  =  50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>14458.ERR2373058</td>\n",
       "      <td>49</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H12_S11_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H12</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>14458.ERR2373089</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H48_S42_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H48</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14458.ERR2373127</td>\n",
       "      <td>49</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H90_S80_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H90</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "45  14458.ERR2373058   49  NO      F     YES    Normal     H12_S11_SG   \n",
       "30  14458.ERR2373089   34  NO      F     YES    Normal     H48_S42_SG   \n",
       "3   14458.ERR2373127   49  NO      M     YES    Normal     H90_S80_SG   \n",
       "\n",
       "   reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "45              NO                H12           14458  \n",
       "30              NO                H48           14458  \n",
       "3               NO                H90           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_23_genome_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>14458.ERR2373106</td>\n",
       "      <td>60</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H66_S59_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H66</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>14458.ERR2373117</td>\n",
       "      <td>51</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H79_S70_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H79</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>14458.ERR2373083</td>\n",
       "      <td>31</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H41_S36_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H41</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "18  14458.ERR2373106   60  YES      F     YES    Normal     H66_S59_SG   \n",
       "12  14458.ERR2373117   51  YES      F     YES    Normal     H79_S70_SG   \n",
       "32  14458.ERR2373083   31   NO      F     YES    Normal     H41_S36_SG   \n",
       "\n",
       "   reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "18              NO                H66           14458  \n",
       "12              NO                H79           14458  \n",
       "32              NO                H41           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_23_species_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373136</td>\n",
       "      <td>55</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H98_S89_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H98</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>14458.ERR2373095</td>\n",
       "      <td>57</td>\n",
       "      <td>YES</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H55_S48_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H55</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>14458.ERR2373075</td>\n",
       "      <td>82</td>\n",
       "      <td>YES</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>H29_S28_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H29</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0   14458.ERR2373136   55  YES      F     YES    Normal     H98_S89_SG   \n",
       "24  14458.ERR2373095   57  YES      M     YES    Normal     H55_S48_SG   \n",
       "39  14458.ERR2373075   82  YES      M     YES    Normal     H29_S28_SG   \n",
       "\n",
       "   reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0               NO                H98           14458  \n",
       "24              NO                H55           14458  \n",
       "39              NO                H29           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_zebra_23_genome_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Wol Scrubbed\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL_scrubbed/feature-table.biom',\n",
    "                       'normal_genome_WOL_scrubbed')\n",
    "\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL_scrubbed/feature-table.biom',\n",
    "                       'normal_zebra_genome_WOL_scrubbed',\n",
    "                      zebra='norm_GERD_WOL')\n",
    "\n",
    "#\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169129_perGenome_RS210.biom',\n",
    "                       'normal_genome_RS210')\n",
    "\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/species_WOL/feature-table.biom',\n",
    "                       'normal_species_WOL')\n",
    "\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom',\n",
    "                       'normal_genome_WOL')\n",
    "\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom',\n",
    "                       'normal_zebra_genome_WOL',\n",
    "                       zebra='norm_GERD_WOL')\n",
    "\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom',\n",
    "                       'normal_23_genome_WOL',\n",
    "                      trim =23)\n",
    "\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/species_WOL/feature-table.biom',\n",
    "                       'normal_23_species_WOL',\n",
    "                      trim = 23)\n",
    "\n",
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom',\n",
    "                       'normal_zebra_23_genome_WOL',\n",
    "                       zebra='norm_GERD_WOL',\n",
    "                      trim = 23)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6f428a7-8d58-4b0a-8faa-bfab0942a8dd",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### GERD"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46d70abe-7c33-4496-a77c-1f961256b9f5",
   "metadata": {},
   "source": [
    "###### Used in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "58b5e847-a677-48c4-b399-e995095578e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_species_WOL_scrubbed  =  29\n",
      "<artifact: FeatureTable[Frequency] uuid: 023fbb04-764c-45af-85ad-fb081504e2d9>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Wol scrubbed\n",
    "\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_species_WOL_scrubbed/feature-table.biom',\n",
    "                       'GERD_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2bbb95db-a229-428d-8dbf-ac68c59a85ab",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Not in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3f7efc1f-0613-4276-a652-27eecc095c60",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_genome_WOL_scrubbed  =  29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_zebra_genome_WOL_scrubbed  =  29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_genome_RS210  =  29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_species_WOL  =  29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_genome_WOL  =  29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14458.ERR2373137</td>\n",
       "      <td>53</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H99_S90_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H99</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14458.ERR2373132</td>\n",
       "      <td>34</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H94_S85_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H94</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_name  age ppi gender shotgun diagnosis og_sample_name  \\\n",
       "0  14458.ERR2373137   53  NO      M     YES      GERD     H99_S90_SG   \n",
       "1  14458.ERR2373135   36  NO      M     YES      GERD     H97_S88_SG   \n",
       "2  14458.ERR2373132   34  NO      F     YES      GERD     H94_S85_SG   \n",
       "\n",
       "  reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "0             YES                H99           14458  \n",
       "1             YES                H97           14458  \n",
       "2             YES                H94           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_zebra_genome_WOL  =  29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>14458.ERR2373059</td>\n",
       "      <td>67</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H13_S12_SG</td>\n",
       "      <td>NO</td>\n",
       "      <td>H13</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14458.ERR2373130</td>\n",
       "      <td>42</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H92_S83_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H92</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>14458.ERR2373057</td>\n",
       "      <td>39</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H11_S10_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H11</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "25  14458.ERR2373059   67   NO      F     YES      GERD     H13_S12_SG   \n",
       "4   14458.ERR2373130   42  YES      F     YES      GERD     H92_S83_SG   \n",
       "26  14458.ERR2373057   39   NO      F     YES      GERD     H11_S10_SG   \n",
       "\n",
       "   reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "25              NO                H13           14458  \n",
       "4              YES                H92           14458  \n",
       "26             YES                H11           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_23_species_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>og_sample_name</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14458.ERR2373102</td>\n",
       "      <td>44</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H62_S55_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H62</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14458.ERR2373135</td>\n",
       "      <td>36</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H97_S88_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H97</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14458.ERR2373130</td>\n",
       "      <td>42</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>H92_S83_SG</td>\n",
       "      <td>YES</td>\n",
       "      <td>H92</td>\n",
       "      <td>14458</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample_name  age  ppi gender shotgun diagnosis og_sample_name  \\\n",
       "13  14458.ERR2373102   44   NO      F     YES      GERD     H62_S55_SG   \n",
       "1   14458.ERR2373135   36   NO      M     YES      GERD     H97_S88_SG   \n",
       "4   14458.ERR2373130   42  YES      F     YES      GERD     H92_S83_SG   \n",
       "\n",
       "   reflux_symptoms metadata_sample_id  qiita_study_id  \n",
       "13             YES                H62           14458  \n",
       "1              YES                H97           14458  \n",
       "4              YES                H92           14458  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_zebra_23_genome_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Wol Scrubbed\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL_scrubbed/feature-table.biom',\n",
    "                       'GERD_genome_WOL_scrubbed')\n",
    "\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL_scrubbed/feature-table.biom',\n",
    "                       'GERD_zebra_genome_WOL_scrubbed',\n",
    "                       zebra='norm_GERD_WOL')\n",
    "                       \n",
    "#\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/169129_perGenome_RS210.biom',\n",
    "                       'GERD_genome_RS210')\n",
    "\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/species_WOL/feature-table.biom',\n",
    "                       'GERD_species_WOL')\n",
    "\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() + '/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom',\n",
    "                       'GERD_genome_WOL')\n",
    "\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() + '/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom',\n",
    "                       'GERD_zebra_genome_WOL',\n",
    "                       zebra='norm_GERD_WOL')\n",
    "\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14458_NormalGerdEsoph/species_WOL/feature-table.biom',\n",
    "                       'GERD_23_species_WOL',\n",
    "                      trim =23)\n",
    "\n",
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() + '/qiita_downloads/qiita14458_NormalGerdEsoph/169125_none_WOL.biom',\n",
    "                       'GERD_zebra_23_genome_WOL',\n",
    "                       zebra='norm_GERD_WOL',\n",
    "                      trim = 23)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a613f92-deb7-48e9-b988-13c8a7680006",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### (not in paper) 16S DATA!!! (More info in Kit testing folder of comad ... done on 10/24/23)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "765b0fbb-c325-406d-af9f-9abc3c555f52",
   "metadata": {},
   "source": [
    "##### Normal - 16S"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2a8d1534-1037-4c07-bfdc-31454c5f81b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14325.H3</td>\n",
       "      <td>53.0</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NO</td>\n",
       "      <td>14325</td>\n",
       "      <td>H3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14325.H5</td>\n",
       "      <td>58.0</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NO</td>\n",
       "      <td>14325</td>\n",
       "      <td>H5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14325.H6</td>\n",
       "      <td>56.0</td>\n",
       "      <td>YES</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NO</td>\n",
       "      <td>14325</td>\n",
       "      <td>H6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_name   age  ppi gender shotgun diagnosis reflux_symptoms  \\\n",
       "0    14325.H3  53.0  YES      F     YES    Normal              NO   \n",
       "1    14325.H5  58.0   NO      F     YES    Normal              NO   \n",
       "2    14325.H6  56.0  YES      F     YES    Normal              NO   \n",
       "\n",
       "   qiita_study_id metadata_sample_id  \n",
       "0           14325                 H3  \n",
       "1           14325                 H5  \n",
       "2           14325                 H6  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in normal_16S  =  50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gerd_normal_data_split('Normal',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14325_16S_NormalGerdEsoph/139540_reference-hit_prep12316.biom',\n",
    "                       'normal_16S',\n",
    "                      shotgun=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4970acf4-d3b6-4652-93f0-ccb42bd32842",
   "metadata": {},
   "source": [
    "##### GERD - 16S"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "10499d84-c331-4c39-bbb2-d1785c06f877",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>age</th>\n",
       "      <th>ppi</th>\n",
       "      <th>gender</th>\n",
       "      <th>shotgun</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>reflux_symptoms</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>metadata_sample_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14325.H2</td>\n",
       "      <td>62.0</td>\n",
       "      <td>NO</td>\n",
       "      <td>F</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>YES</td>\n",
       "      <td>14325</td>\n",
       "      <td>H2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14325.H4</td>\n",
       "      <td>76.0</td>\n",
       "      <td>YES</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>YES</td>\n",
       "      <td>14325</td>\n",
       "      <td>H4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14325.H7</td>\n",
       "      <td>76.0</td>\n",
       "      <td>NO</td>\n",
       "      <td>M</td>\n",
       "      <td>YES</td>\n",
       "      <td>GERD</td>\n",
       "      <td>YES</td>\n",
       "      <td>14325</td>\n",
       "      <td>H7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_name   age  ppi gender shotgun diagnosis reflux_symptoms  \\\n",
       "0    14325.H2  62.0   NO      F     YES      GERD             YES   \n",
       "1    14325.H4  76.0  YES      M     YES      GERD             YES   \n",
       "2    14325.H7  76.0   NO      M     YES      GERD             YES   \n",
       "\n",
       "   qiita_study_id metadata_sample_id  \n",
       "0           14325                 H2  \n",
       "1           14325                 H4  \n",
       "2           14325                 H7  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in GERD_16S  =  29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gerd_normal_data_split('GERD',\n",
    "                       os.getcwd() +'/qiita_downloads/qiita14325_16S_NormalGerdEsoph/139540_reference-hit_prep12316.biom',\n",
    "                       'GERD_16S',\n",
    "                      shotgun=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "230a33d7-437a-4f89-b456-54593cca3d48",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Barrett's Esophgous Progressors/Non Progressors\n",
    "Paulson, T. G. et al. Somatic whole genome dynamics of precancer in Barrettâ€™s esophagus reveals features associated with disease progression. Nature Communications vol. 13 (2022) https://doi.org/10.1038/s41467-022-29767-7"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af7c8631-d14b-45e6-96ae-df61afaa05df",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### (Not in paper) Barrett's Esophgous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "972a186b-de3d-4ba6-9620-0c5cbe579e7f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_genome_RS210  =  302\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_species_WOL  =  316\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_genome_WOL  =  316\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_zebra_genome_WOL  =  316\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in all_BE_genome_WOL  =  336\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_genome_RS210')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_species_WOL')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_genome_WOL')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_zebra_genome_WOL',\n",
    "             zebra = 'BE_WOL')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1', 'T2','LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'all_BE_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3ecf352-905f-44c5-8778-9c98cc2abc8b",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "238a6ebb-c834-4023-b21a-a6b32d8c7266",
   "metadata": {
    "tags": []
   },
   "source": [
    "###### Used in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "08ae62c4-8174-41e5-9365-e0ced40c24aa",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               53.23           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_species_WOL_scrubbed  =  80\n",
      "<artifact: FeatureTable[Frequency] uuid: 3139877c-5e5a-463c-a9cd-f0cbf2655e2c>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL scrubbed -TP1\n",
    "\n",
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_T1_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "88fe01bd-fd5a-4fbb-9d08-e4a0fca3fb65",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_wLF_species_WOL_scrubbed  =  178\n",
      "<artifact: FeatureTable[Frequency] uuid: 5e01b2b6-6a62-4ebd-902b-7b084a091965>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL scrubbed\n",
    "\n",
    "be_data_split(['NP'], ['T1', 'T2', 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_wLF_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e923a07-4b2c-40a9-a7ac-e620dc91ad82",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Used in supp ?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43652f3b-b770-47ed-8a3b-fd98297e4fea",
   "metadata": {},
   "outputs": [],
   "source": [
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "def454aa-9c9e-4fc0-9592-abb20d8a78ab",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Not in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "36bb1096-c2a1-4d8d-93f1-bd8baa17f53f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_genome_WOL_scrubbed  =  158\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_wLF_zebra_genome_WOL_scrubbed  =  178\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_wLF_genome_WOL_scrubbed  =  178\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               53.23           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_genome_WOL_scrubbed  =  80\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               53.23           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_zebra_genome_WOL_scrubbed  =  80\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2                NP               57.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_genome_WOL_scrubbed  =  78\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_genome_RS210  =  150\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_species_WOL  =  158\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               87.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_zebra_genome_WOL  =  158\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M        17         T1   \n",
       "1  14598.PCGA.1064.BE.03      White_nonHispanic      M        17         T2   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M        55         T1   \n",
       "\n",
       "  ismainstudy        pairednormal distancefromgej orginal_sampleid  \\\n",
       "0           1  17-24987N-105W-BLD               3            23561   \n",
       "1           1  17-24987N-105W-BLD               4            23567   \n",
       "2           1  55-25056N-204U-BLD               3            23611   \n",
       "\n",
       "  progressionstatus patientageatsample qiita_study_id  \\\n",
       "0                NP              78.83          14598   \n",
       "1                NP              87.07          14598   \n",
       "2                NP              53.23          14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T2_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_exact_genome_WOL  =  80\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M        17         T1   \n",
       "1  14598.PCGA.1064.BE.03      White_nonHispanic      M        17         T2   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M        55         T1   \n",
       "\n",
       "  ismainstudy        pairednormal distancefromgej orginal_sampleid  \\\n",
       "0           1  17-24987N-105W-BLD               3            23561   \n",
       "1           1  17-24987N-105W-BLD               4            23567   \n",
       "2           1  55-25056N-204U-BLD               3            23611   \n",
       "\n",
       "  progressionstatus patientageatsample qiita_study_id  \\\n",
       "0                NP              78.83          14598   \n",
       "1                NP              87.07          14598   \n",
       "2                NP              53.23          14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T2_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_exact_species_WOL  =  80\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M        17         T1   \n",
       "1  14598.PCGA.1064.BE.03      White_nonHispanic      M        17         T2   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M        55         T1   \n",
       "\n",
       "  ismainstudy        pairednormal distancefromgej orginal_sampleid  \\\n",
       "0           1  17-24987N-105W-BLD               3            23561   \n",
       "1           1  17-24987N-105W-BLD               4            23567   \n",
       "2           1  55-25056N-204U-BLD               3            23611   \n",
       "\n",
       "  progressionstatus patientageatsample qiita_study_id  \\\n",
       "0                NP              78.83          14598   \n",
       "1                NP              87.07          14598   \n",
       "2                NP              53.23          14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T2_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_zebra_exact_genome_WOL  =  80\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>14598.PCGA.1087.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>385</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>385-24955N-148R-GST</td>\n",
       "      <td>2</td>\n",
       "      <td>24819</td>\n",
       "      <td>NP</td>\n",
       "      <td>68.34</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>14598.PCGA.1123.BE.03</td>\n",
       "      <td>AmericanIndian_AlaskanNative</td>\n",
       "      <td>M</td>\n",
       "      <td>806</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>806-25035N-53S-BLD</td>\n",
       "      <td>2</td>\n",
       "      <td>24157</td>\n",
       "      <td>NP</td>\n",
       "      <td>60.13</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>14598.PCGA.1095.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>478</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>478-25071N-116V-BLD</td>\n",
       "      <td>2</td>\n",
       "      <td>23938</td>\n",
       "      <td>NP</td>\n",
       "      <td>71.24</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               sample_name                          race gender  patientid  \\\n",
       "50   14598.PCGA.1087.BE.04             White_nonHispanic      M        385   \n",
       "112  14598.PCGA.1123.BE.03  AmericanIndian_AlaskanNative      M        806   \n",
       "55   14598.PCGA.1095.BE.01             White_nonHispanic      M        478   \n",
       "\n",
       "    sampletype  ismainstudy         pairednormal  distancefromgej  \\\n",
       "50          T2            1  385-24955N-148R-GST                2   \n",
       "112         T2            1   806-25035N-53S-BLD                2   \n",
       "55          T1            1  478-25071N-116V-BLD                2   \n",
       "\n",
       "     orginal_sampleid progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "50              24819                NP               68.34           14598   \n",
       "112             24157                NP               60.13           14598   \n",
       "55              23938                NP               71.24           14598   \n",
       "\n",
       "    sampletype_progressionstatus  \n",
       "50                         T2_NP  \n",
       "112                        T2_NP  \n",
       "55                         T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_zebra_23_genome_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL scrubbed\n",
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_genome_WOL_scrubbed')\n",
    "\n",
    "be_data_split(['NP'], ['T1', 'T2', 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_wLF_zebra_genome_WOL_scrubbed',\n",
    "             zebra = 'BE_WOL')\n",
    "\n",
    "be_data_split(['NP'], ['T1', 'T2', 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_wLF_genome_WOL_scrubbed')\n",
    "\n",
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_T1_genome_WOL_scrubbed')\n",
    "\n",
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_T1_zebra_genome_WOL_scrubbed',\n",
    "             zebra = 'BE_WOL')\n",
    "\n",
    "be_data_split(['NP'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_T2_genome_WOL_scrubbed')\n",
    "\n",
    "#\n",
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_NP_genome_RS210')\n",
    "\n",
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_NP_species_WOL')\n",
    "\n",
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_zebra_genome_WOL',\n",
    "             zebra = 'BE_WOL')\n",
    "\n",
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_exact_genome_WOL',\n",
    "             exact_pairs = True)\n",
    "\n",
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_NP_exact_species_WOL',\n",
    "             exact_pairs = True)\n",
    "\n",
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_zebra_exact_genome_WOL',\n",
    "             zebra = 'BE_WOL',\n",
    "             exact_pairs = True)\n",
    "\n",
    "be_data_split(['NP'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_zebra_23_genome_WOL',\n",
    "             zebra = 'BE_WOL',\n",
    "             trim = 23)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea5c4296-def2-4229-b2ae-27670bfb85be",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### (Supp ?) Barrett's Esophgous Non Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a16cba6e-2fb2-4702-9ac4-44977b635e61",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               53.23           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_species_WOL_scrubbed  =  80\n",
      "<artifact: FeatureTable[Frequency] uuid: 2dd1c7db-8044-4682-8954-5db78d0cf5f9>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Used in supp.\n",
    "\n",
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_T1_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7c956478-7b19-4de9-b969-2e0c8df8fb5e",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               53.23           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_genome_RS210  =  76\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               53.23           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_species_WOL  =  80\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1068.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>59</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>59-24989N-165S-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23638</td>\n",
       "      <td>NP</td>\n",
       "      <td>67.14</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M        17         T1   \n",
       "1  14598.PCGA.1067.BE.01  Asian_PacificIslander      M        55         T1   \n",
       "2  14598.PCGA.1068.BE.01      White_nonHispanic      M        59         T1   \n",
       "\n",
       "  ismainstudy        pairednormal distancefromgej orginal_sampleid  \\\n",
       "0           1  17-24987N-105W-BLD               3            23561   \n",
       "1           1  55-25056N-204U-BLD               3            23611   \n",
       "2           1  59-24989N-165S-BLD               3            23638   \n",
       "\n",
       "  progressionstatus patientageatsample qiita_study_id  \\\n",
       "0                NP              78.83          14598   \n",
       "1                NP              53.23          14598   \n",
       "2                NP              67.14          14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_exact_species_WOL  =  40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1068.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>59</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>59-24989N-165S-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23638</td>\n",
       "      <td>NP</td>\n",
       "      <td>67.14</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M        17         T1   \n",
       "1  14598.PCGA.1067.BE.01  Asian_PacificIslander      M        55         T1   \n",
       "2  14598.PCGA.1068.BE.01      White_nonHispanic      M        59         T1   \n",
       "\n",
       "  ismainstudy        pairednormal distancefromgej orginal_sampleid  \\\n",
       "0           1  17-24987N-105W-BLD               3            23561   \n",
       "1           1  55-25056N-204U-BLD               3            23611   \n",
       "2           1  59-24989N-165S-BLD               3            23638   \n",
       "\n",
       "  progressionstatus patientageatsample qiita_study_id  \\\n",
       "0                NP              78.83          14598   \n",
       "1                NP              53.23          14598   \n",
       "2                NP              67.14          14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_exact_zebra_genome_WOL  =  40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>14598.PCGA.1078.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>222</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>222-25062N-179T-BLD</td>\n",
       "      <td>9</td>\n",
       "      <td>24376</td>\n",
       "      <td>NP</td>\n",
       "      <td>75.01</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>14598.PCGA.1096.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>483</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>483-24961N-246S-GST</td>\n",
       "      <td>6</td>\n",
       "      <td>23824</td>\n",
       "      <td>NP</td>\n",
       "      <td>68.52</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>14598.PCGA.1098.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>515</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>515-25073N-18V-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23683</td>\n",
       "      <td>NP</td>\n",
       "      <td>50.28</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              sample_name               race gender patientid sampletype  \\\n",
       "8   14598.PCGA.1078.BE.03  White_nonHispanic      M       222         T1   \n",
       "15  14598.PCGA.1096.BE.01  White_nonHispanic      M       483         T1   \n",
       "16  14598.PCGA.1098.BE.01  White_nonHispanic      M       515         T1   \n",
       "\n",
       "   ismainstudy         pairednormal distancefromgej orginal_sampleid  \\\n",
       "8            1  222-25062N-179T-BLD               9            24376   \n",
       "15           1  483-24961N-246S-GST               6            23824   \n",
       "16           1   515-25073N-18V-BLD               3            23683   \n",
       "\n",
       "   progressionstatus patientageatsample qiita_study_id  \\\n",
       "8                 NP              75.01          14598   \n",
       "15                NP              68.52          14598   \n",
       "16                NP              50.28          14598   \n",
       "\n",
       "   sampletype_progressionstatus  \n",
       "8                         T1_NP  \n",
       "15                        T1_NP  \n",
       "16                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_exact_23_species_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>14598.PCGA.1119.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>729</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>729-25033N-237R-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23752</td>\n",
       "      <td>NP</td>\n",
       "      <td>65.51</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>14598.PCGA.1130.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>900</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>900-25089N-32P-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>23926</td>\n",
       "      <td>NP</td>\n",
       "      <td>46.33</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>14598.PCGA.1111.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>635</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>635-25080N-95V-BLD</td>\n",
       "      <td>6</td>\n",
       "      <td>24037</td>\n",
       "      <td>NP</td>\n",
       "      <td>54.82</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              sample_name               race gender patientid sampletype  \\\n",
       "27  14598.PCGA.1119.BE.01  White_nonHispanic      M       729         T1   \n",
       "31  14598.PCGA.1130.BE.01  White_nonHispanic      M       900         T1   \n",
       "24  14598.PCGA.1111.BE.01  White_nonHispanic      M       635         T1   \n",
       "\n",
       "   ismainstudy         pairednormal distancefromgej orginal_sampleid  \\\n",
       "27           1  729-25033N-237R-BLD               1            23752   \n",
       "31           1   900-25089N-32P-GST               5            23926   \n",
       "24           1   635-25080N-95V-BLD               6            24037   \n",
       "\n",
       "   progressionstatus patientageatsample qiita_study_id  \\\n",
       "27                NP              65.51          14598   \n",
       "31                NP              46.33          14598   \n",
       "24                NP              54.82          14598   \n",
       "\n",
       "   sampletype_progressionstatus  \n",
       "27                        T1_NP  \n",
       "31                        T1_NP  \n",
       "24                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_exact_zebra_23_genome_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2                NP               53.23           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                        T1_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_zebra_genome_WOL  =  80\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_NP_T1_genome_RS210')\n",
    "\n",
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_NP_T1_species_WOL')\n",
    "\n",
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_NP_T1_exact_species_WOL',\n",
    "             exact_pairs = True)\n",
    "\n",
    "be_data_split(['NP'], ['T1'], \n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_T1_exact_zebra_genome_WOL',\n",
    "             zebra = 'BE_WOL',\n",
    "             exact_pairs = True)\n",
    "\n",
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_NP_T1_exact_23_species_WOL',\n",
    "             exact_pairs = True,\n",
    "              trim = 23)\n",
    "\n",
    "be_data_split(['NP'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_T1_exact_zebra_23_genome_WOL',\n",
    "             exact_pairs = True,\n",
    "              zebra = 'BE_WOL',\n",
    "              trim = 23)\n",
    "\n",
    "be_data_split(['NP'], ['T1'], \n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_T1_zebra_genome_WOL',\n",
    "             zebra = 'BE_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aef4c92a-0a61-458e-81a0-929b46fb1315",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### (Supp ?) Barrett's Esophgous Non Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0fc2388a-20bb-4e54-ae62-52d51aef237e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2                NP               57.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_wLT_species_WOL_scrubbed  =  98\n",
      "<artifact: FeatureTable[Frequency] uuid: 8c4cabab-1aad-457f-b22b-55adbde8a71d>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Used in supp.\n",
    "\n",
    "be_data_split(['NP'], ['T2', 'LongFollowUp'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_NP_T2_wLT_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "29fa4ebc-f730-4c2d-81b9-0971436ad38b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2                NP               57.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_genome_RS210  =  74\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2                NP               57.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_species_WOL  =  78\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1068.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>59</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>59-24989N-165S-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23644</td>\n",
       "      <td>NP</td>\n",
       "      <td>69.0</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M        17         T2   \n",
       "1  14598.PCGA.1067.BE.02  Asian_PacificIslander      M        55         T2   \n",
       "2  14598.PCGA.1068.BE.03      White_nonHispanic      M        59         T2   \n",
       "\n",
       "  ismainstudy        pairednormal distancefromgej orginal_sampleid  \\\n",
       "0           1  17-24987N-105W-BLD               4            23567   \n",
       "1           1  55-25056N-204U-BLD               1            23614   \n",
       "2           1  59-24989N-165S-BLD               3            23644   \n",
       "\n",
       "  progressionstatus patientageatsample qiita_study_id  \\\n",
       "0                NP              87.07          14598   \n",
       "1                NP              57.07          14598   \n",
       "2                NP               69.0          14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_exact_zebra_genome_WOL  =  40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2                NP               57.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                        T2_NP  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_zebra_genome_WOL  =  78\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_NP_T2_genome_RS210')\n",
    "\n",
    "be_data_split(['NP'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_NP_T2_species_WOL')\n",
    "\n",
    "be_data_split(['NP'], ['T2'], \n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_T2_exact_zebra_genome_WOL',\n",
    "             zebra = 'BE_WOL',\n",
    "             exact_pairs = True)\n",
    "\n",
    "be_data_split(['NP'], ['T2'], \n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_NP_T2_zebra_genome_WOL',\n",
    "             zebra = 'BE_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e2327758-9e16-465d-b3c3-a864a8d76dc6",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### (Not used in paper) Barrett's Esophgous Progressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "01050185-76d7-44d0-9093-02e371e74830",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_genome_RS210  =  152\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_species_WOL  =  158\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_zebra_genome_WOL  =  158\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['Prog'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_Prog_genome_RS210')\n",
    "\n",
    "be_data_split(['Prog'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_Prog_species_WOL')\n",
    "\n",
    "be_data_split(['Prog'], ['T1', 'T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_Prog_zebra_genome_WOL',\n",
    "             zebra = 'BE_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8037a84d-adc1-4bf9-b6d3-2d837afbf1ef",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea36bda7-cdf6-4f82-bc60-9b97dfbbf939",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Used in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d599bee1-1cb7-4177-b5ae-d14b86f97720",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               56.25           14598   \n",
       "1              Prog               56.25           14598   \n",
       "2              Prog               64.11           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T1_Prog  \n",
       "1                      T1_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_species_WOL_scrubbed  =  79\n",
      "<artifact: FeatureTable[Frequency] uuid: 9c594e17-9ee5-4aea-b753-b8d5c3a5ceb9>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL scrubbed\n",
    "\n",
    "be_data_split(['Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_Prog_T1_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d4d18e87-f688-4683-9da3-04e8b23fb75d",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Not used in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d9a3dfa3-8d7d-40ac-8e9d-79293ce8c321",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               56.25           14598   \n",
       "1              Prog               56.25           14598   \n",
       "2              Prog               64.11           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T1_Prog  \n",
       "1                      T1_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_genome_WOL_scrubbed  =  79\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               56.25           14598   \n",
       "1              Prog               56.25           14598   \n",
       "2              Prog               64.11           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T1_Prog  \n",
       "1                      T1_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_zebra_genome_WOL_scrubbed  =  79\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               56.25           14598   \n",
       "1              Prog               56.25           14598   \n",
       "2              Prog               64.11           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T1_Prog  \n",
       "1                      T1_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_genome_RS210  =  76\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               56.25           14598   \n",
       "1              Prog               56.25           14598   \n",
       "2              Prog               64.11           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T1_Prog  \n",
       "1                      T1_Prog  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_species_WOL  =  79\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL scrubbed\n",
    "be_data_split(['Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_Prog_T1_genome_WOL_scrubbed')\n",
    "\n",
    "be_data_split(['Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_Prog_T1_zebra_genome_WOL_scrubbed',\n",
    "              zebra = 'BE_WOL')\n",
    "\n",
    "#\n",
    "be_data_split(['Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_Prog_T1_genome_RS210')\n",
    "\n",
    "be_data_split(['Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_Prog_T1_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "532903b1-2dcc-408f-90c5-709a875b7646",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6995b4be-6a12-4ab1-b2aa-5c09dc9832ae",
   "metadata": {
    "tags": []
   },
   "source": [
    "###### Used in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3f0e0fd9-62c9-4494-8731-a2bab0344604",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               65.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_species_WOL_scrubbed  =  79\n",
      "<artifact: FeatureTable[Frequency] uuid: a9ee04bc-9036-4143-8b21-30b9c5b6cb4d>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL scrubbed\n",
    "\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_Prog_T2_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2905cee7-4449-40a0-af1c-4d945a30eeec",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Not used in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "cef0e3fa-fb27-4b5c-908f-9c00fae00760",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               65.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_genome_WOL_scrubbed  =  79\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               65.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_genome_RS210  =  76\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               65.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_species_WOL  =  79\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0              Prog               57.65           14598   \n",
       "1              Prog               57.65           14598   \n",
       "2              Prog               65.07           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_genome_WOL  =  79\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1069.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>74</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>74-25057N-80U-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23677</td>\n",
       "      <td>Prog</td>\n",
       "      <td>52.21</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01      Other_Unknown      M        42         T2   \n",
       "1  14598.PCGA.1066.BE.02      Other_Unknown      F        43         T2   \n",
       "2  14598.PCGA.1069.BE.02  White_nonHispanic      M        74         T2   \n",
       "\n",
       "  ismainstudy       pairednormal distancefromgej orginal_sampleid  \\\n",
       "0           1  42-24954N-53L-GST               5            24283   \n",
       "1           1  43-24956N-29P-GST               4            23593   \n",
       "2           1  74-25057N-80U-BLD               4            23677   \n",
       "\n",
       "  progressionstatus patientageatsample qiita_study_id  \\\n",
       "0              Prog              57.65          14598   \n",
       "1              Prog              65.07          14598   \n",
       "2              Prog              52.21          14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_exact_species_WOL  =  40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1069.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>74</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>74-25057N-80U-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23677</td>\n",
       "      <td>Prog</td>\n",
       "      <td>52.21</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01      Other_Unknown      M        42         T2   \n",
       "1  14598.PCGA.1066.BE.02      Other_Unknown      F        43         T2   \n",
       "2  14598.PCGA.1069.BE.02  White_nonHispanic      M        74         T2   \n",
       "\n",
       "  ismainstudy       pairednormal distancefromgej orginal_sampleid  \\\n",
       "0           1  42-24954N-53L-GST               5            24283   \n",
       "1           1  43-24956N-29P-GST               4            23593   \n",
       "2           1  74-25057N-80U-BLD               4            23677   \n",
       "\n",
       "  progressionstatus patientageatsample qiita_study_id  \\\n",
       "0              Prog              57.65          14598   \n",
       "1              Prog              65.07          14598   \n",
       "2              Prog              52.21          14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                      T2_Prog  \n",
       "1                      T2_Prog  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_exact_zebra_genome_WOL  =  40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>14598.PCGA.1088.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>387</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>387-24974N-309J-GST</td>\n",
       "      <td>0</td>\n",
       "      <td>23662</td>\n",
       "      <td>Prog</td>\n",
       "      <td>70.26</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>14598.PCGA.1124.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>833</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>833-25085N-211U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>24016</td>\n",
       "      <td>Prog</td>\n",
       "      <td>80.53</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>14598.PCGA.1091.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>403</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>403-24957N-153N-GST</td>\n",
       "      <td>6</td>\n",
       "      <td>24383</td>\n",
       "      <td>Prog</td>\n",
       "      <td>77.81</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              sample_name               race gender patientid sampletype  \\\n",
       "11  14598.PCGA.1088.BE.03  White_nonHispanic      M       387         T2   \n",
       "31  14598.PCGA.1124.BE.02  White_nonHispanic      M       833         T2   \n",
       "14  14598.PCGA.1091.BE.01  White_nonHispanic      M       403         T2   \n",
       "\n",
       "   ismainstudy         pairednormal distancefromgej orginal_sampleid  \\\n",
       "11           1  387-24974N-309J-GST               0            23662   \n",
       "31           1  833-25085N-211U-BLD               3            24016   \n",
       "14           1  403-24957N-153N-GST               6            24383   \n",
       "\n",
       "   progressionstatus patientageatsample qiita_study_id  \\\n",
       "11              Prog              70.26          14598   \n",
       "31              Prog              80.53          14598   \n",
       "14              Prog              77.81          14598   \n",
       "\n",
       "   sampletype_progressionstatus  \n",
       "11                      T2_Prog  \n",
       "31                      T2_Prog  \n",
       "14                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_exact_23_species_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>14598.PCGA.1094.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>450</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>450-25070N-120R-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23920</td>\n",
       "      <td>Prog</td>\n",
       "      <td>46.46</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>14598.PCGA.1136.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>F</td>\n",
       "      <td>954</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>954-25048N-259T-BLD</td>\n",
       "      <td>2</td>\n",
       "      <td>24214</td>\n",
       "      <td>Prog</td>\n",
       "      <td>77.46</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>14598.PCGA.1082.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>295</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>295-25066N-72T-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24175</td>\n",
       "      <td>Prog</td>\n",
       "      <td>69.51</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              sample_name               race gender patientid sampletype  \\\n",
       "16  14598.PCGA.1094.BE.02  White_nonHispanic      M       450         T2   \n",
       "36  14598.PCGA.1136.BE.03  White_nonHispanic      F       954         T2   \n",
       "9   14598.PCGA.1082.BE.03  White_nonHispanic      M       295         T2   \n",
       "\n",
       "   ismainstudy         pairednormal distancefromgej orginal_sampleid  \\\n",
       "16           1  450-25070N-120R-BLD               1            23920   \n",
       "36           1  954-25048N-259T-BLD               2            24214   \n",
       "9            1   295-25066N-72T-BLD               1            24175   \n",
       "\n",
       "   progressionstatus patientageatsample qiita_study_id  \\\n",
       "16              Prog              46.46          14598   \n",
       "36              Prog              77.46          14598   \n",
       "9               Prog              69.51          14598   \n",
       "\n",
       "   sampletype_progressionstatus  \n",
       "16                      T2_Prog  \n",
       "36                      T2_Prog  \n",
       "9                       T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_exact_zebra_23_genome_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL scrubbed\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_Prog_T2_genome_WOL_scrubbed')\n",
    "\n",
    "#\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_Prog_T2_genome_RS210')\n",
    "\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_Prog_T2_species_WOL')\n",
    "\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_Prog_T2_genome_WOL')\n",
    "\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_Prog_T2_exact_species_WOL',\n",
    "             exact_pairs = True)\n",
    "\n",
    "be_data_split(['Prog'], ['T2'], \n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_Prog_T2_exact_zebra_genome_WOL',\n",
    "             zebra = 'BE_WOL',\n",
    "             exact_pairs = True)\n",
    "\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_Prog_T2_exact_23_species_WOL',\n",
    "             exact_pairs = True,\n",
    "              trim = 23)\n",
    "\n",
    "be_data_split(['Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_Prog_T2_exact_zebra_23_genome_WOL',\n",
    "             exact_pairs = True,\n",
    "              zebra = 'BE_WOL',\n",
    "              trim = 23)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47c91944-f659-4117-8a8c-b4493c920159",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### (Might be used in paper) Barrett's Esophgous T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1ce7cb42-a039-4d82-a2b4-2efa0c722021",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1065.BE.03      Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1   42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T1_species_WOL_scrubbed  =  159\n",
      "<artifact: FeatureTable[Frequency] uuid: 80f05428-f347-44b9-ac0b-b12b8ff66c23>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:133: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Date: 5/23/2024; Goal: To create a TP1 NCO vs. TP1 CO to run birdman\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_species_WOL_scrubbed/feature-table.biom',\n",
    "              'BE_T1_species_WOL_scrubbed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "19463279-9ce3-4958-a19f-1312813c6e98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1065.BE.03      Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1   42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T1_genome_RS210  =  152\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1065.BE.03      Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1   42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T1_species_WOL  =  159\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "      <td>T1_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1065.BE.03      Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1   42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               78.83           14598   \n",
       "1                NP               78.83           14598   \n",
       "2              Prog               56.25           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T1_NP  \n",
       "1                        T1_NP  \n",
       "2                      T1_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T1_genome_WOL  =  159\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP','Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_T1_genome_RS210')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_T1_species_WOL')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T1'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_T1_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "209583ec-8ee9-4f90-b5aa-4828a0ddc8d3",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### (Not used in paper) Barrett's Esophgous T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "03e25eea-2b82-48a0-a7c1-66ec6813094d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04  White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1065.BE.01      Other_Unknown      M         42         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1   42-24954N-53L-GST                5             24283   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2              Prog               57.65           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T2_genome_RS210  =  150\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04  White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1065.BE.01      Other_Unknown      M         42         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1   42-24954N-53L-GST                5             24283   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2              Prog               57.65           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T2_species_WOL  =  157\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>sampletype_progressionstatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_NP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "      <td>T2_Prog</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04  White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1065.BE.01      Other_Unknown      M         42         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1   42-24954N-53L-GST                5             24283   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \\\n",
       "0                NP               87.07           14598   \n",
       "1                NP               87.07           14598   \n",
       "2              Prog               57.65           14598   \n",
       "\n",
       "  sampletype_progressionstatus  \n",
       "0                        T2_NP  \n",
       "1                        T2_NP  \n",
       "2                      T2_Prog  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_T2_genome_WOL  =  157\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "be_data_split(['NP','Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/169098_perGenome_RS210.biom',\n",
    "              'BE_T2_genome_RS210')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149428_species_WOL.biom',\n",
    "              'BE_T2_species_WOL')\n",
    "\n",
    "be_data_split(['NP', 'Prog'], ['T2'],\n",
    "              os.getcwd() + '/qiita_downloads/qiita14598_BE_Esoph/149431_genome_WOL.biom',\n",
    "              'BE_T2_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4de4bd81-f239-4139-be7f-4cf8cf58fdb0",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Normal, BE, EAC Tissue \n",
    "Ross-Innes, C. S. et al. Whole-genome sequencing provides new insights into the clonal architecture of Barrettâ€™s esophagus and esophageal adenocarcinoma. Nature Genetics vol. 47 1038â€“1046 (2015) https://doi.org/10.1038/ng.3357"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93ea269c-b34c-4929-8f25-aab51638bd0a",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### (Not used in paper) Normal, BE, EAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b30e7f41-f4f6-4af1-afed-acb2f6ded801",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in ALL_Ross_combo_species_WOL  =  88\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Normal', 'Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_combo_R1_R2_paired_unpaired.biom',\n",
    "                         'ALL_Ross_combo_species_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9ae58a46-601f-47ee-a37b-ca45feca700a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in ALL_Ross_paired_species_WOL  =  88\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in ALL_Ross_paired_genome_WOL  =  88\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Normal', 'Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'ALL_Ross_paired_species_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['Normal', 'Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'ALL_Ross_paired_genome_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "750caafd-6f91-4db2-be75-00615ddcbb2d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in ALL_Ross_unpaired_species_WOL  =  88\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Normal', 'Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_unpaired.biom',\n",
    "                         'ALL_Ross_unpaired_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "faf44e66-26a9-4bbd-b773-97dad3f1cade",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### BE"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "50f9c6c6-0bed-47e7-934a-a793d8cc9f14",
   "metadata": {
    "tags": []
   },
   "source": [
    "###### Used in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a19f2fcd-c8b8-4934-9b08-edb4ef184656",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "      <th>EGA_ID</th>\n",
       "      <th>EGAIdentifier</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000530875</td>\n",
       "      <td>LP6007521-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000586502</td>\n",
       "      <td>LP6005335-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000525202</td>\n",
       "      <td>LP6007410-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Barretts_13_EGAF00000525459</td>\n",
       "      <td>13</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>F</td>\n",
       "      <td>61</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>15.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.02</td>\n",
       "      <td>298.0</td>\n",
       "      <td>5.19</td>\n",
       "      <td>0.222</td>\n",
       "      <td>54.36</td>\n",
       "      <td>0.973</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000525459</td>\n",
       "      <td>LP6005691-DNA_F01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Barretts_14_EGAF00000560129</td>\n",
       "      <td>14</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>M</td>\n",
       "      <td>68</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.48</td>\n",
       "      <td>317.0</td>\n",
       "      <td>4.70</td>\n",
       "      <td>0.253</td>\n",
       "      <td>61.75</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000560129</td>\n",
       "      <td>LP6005501-DNA_E01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Barretts_14_EGAF00000814878</td>\n",
       "      <td>14</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>M</td>\n",
       "      <td>68</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.48</td>\n",
       "      <td>317.0</td>\n",
       "      <td>4.70</td>\n",
       "      <td>0.253</td>\n",
       "      <td>61.75</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000814878</td>\n",
       "      <td>LP6008023-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Barretts_15_EGAF00000525456</td>\n",
       "      <td>15</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N3M1</td>\n",
       "      <td>IV</td>\n",
       "      <td>Blood</td>\n",
       "      <td>9.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.07</td>\n",
       "      <td>323.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.237</td>\n",
       "      <td>67.85</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000525456</td>\n",
       "      <td>LP6005691-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Barretts_15_EGAF00000814867</td>\n",
       "      <td>15</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N3M1</td>\n",
       "      <td>IV</td>\n",
       "      <td>Blood</td>\n",
       "      <td>9.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.07</td>\n",
       "      <td>323.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.237</td>\n",
       "      <td>67.85</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000814867</td>\n",
       "      <td>LP6008023-DNA_E01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Barretts_15_EGAF00000814879</td>\n",
       "      <td>15</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N3M1</td>\n",
       "      <td>IV</td>\n",
       "      <td>Blood</td>\n",
       "      <td>9.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.07</td>\n",
       "      <td>323.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.237</td>\n",
       "      <td>67.85</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000814879</td>\n",
       "      <td>LP6008023-DNA_F01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Barretts_16_EGAF00000525464</td>\n",
       "      <td>16</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>92.38</td>\n",
       "      <td>292.0</td>\n",
       "      <td>4.79</td>\n",
       "      <td>0.234</td>\n",
       "      <td>57.70</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000525464</td>\n",
       "      <td>LP6005691-DNA_A02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Barretts_17_EGAF00000525463</td>\n",
       "      <td>17</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>82</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.36</td>\n",
       "      <td>319.0</td>\n",
       "      <td>4.79</td>\n",
       "      <td>0.236</td>\n",
       "      <td>71.36</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000525463</td>\n",
       "      <td>LP6005691-DNA_D01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Barretts_17_EGAF00000814869</td>\n",
       "      <td>17</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>82</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.36</td>\n",
       "      <td>319.0</td>\n",
       "      <td>4.79</td>\n",
       "      <td>0.236</td>\n",
       "      <td>71.36</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000814869</td>\n",
       "      <td>LP6008023-DNA_A02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Barretts_17_EGAF00000814882</td>\n",
       "      <td>17</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>82</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.36</td>\n",
       "      <td>319.0</td>\n",
       "      <td>4.79</td>\n",
       "      <td>0.236</td>\n",
       "      <td>71.36</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000814882</td>\n",
       "      <td>LP6008023-DNA_B02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Barretts_17_EGAF00000814883</td>\n",
       "      <td>17</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>82</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.36</td>\n",
       "      <td>319.0</td>\n",
       "      <td>4.79</td>\n",
       "      <td>0.236</td>\n",
       "      <td>71.36</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000814883</td>\n",
       "      <td>LP6008023-DNA_C02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Barretts_17_EGAF00000814884</td>\n",
       "      <td>17</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>82</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.36</td>\n",
       "      <td>319.0</td>\n",
       "      <td>4.79</td>\n",
       "      <td>0.236</td>\n",
       "      <td>71.36</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000814884</td>\n",
       "      <td>LP6008023-DNA_D02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Barretts_18_EGAF00000560130</td>\n",
       "      <td>18</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>M</td>\n",
       "      <td>61</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>10.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.88</td>\n",
       "      <td>282.0</td>\n",
       "      <td>4.71</td>\n",
       "      <td>0.240</td>\n",
       "      <td>60.85</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000560130</td>\n",
       "      <td>LP6005501-DNA_F01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Barretts_19_EGAF00000530863</td>\n",
       "      <td>19</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.65</td>\n",
       "      <td>440.0</td>\n",
       "      <td>4.32</td>\n",
       "      <td>0.242</td>\n",
       "      <td>73.95</td>\n",
       "      <td>0.979</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000530863</td>\n",
       "      <td>LP2000110-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Barretts_1_EGAF00000501981</td>\n",
       "      <td>1</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>62</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.44</td>\n",
       "      <td>325.0</td>\n",
       "      <td>4.25</td>\n",
       "      <td>0.227</td>\n",
       "      <td>73.05</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000501981</td>\n",
       "      <td>LP6007592</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Barretts_20_EGAF00000560125</td>\n",
       "      <td>20</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Edinburgh Royal Infirmary, Edinburgh</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.31</td>\n",
       "      <td>324.0</td>\n",
       "      <td>4.11</td>\n",
       "      <td>0.227</td>\n",
       "      <td>62.97</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000560125</td>\n",
       "      <td>LP6005501-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Barretts_21_EGAF00000560126</td>\n",
       "      <td>21</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>61</td>\n",
       "      <td>T1N0M0</td>\n",
       "      <td>IA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.66</td>\n",
       "      <td>330.0</td>\n",
       "      <td>4.42</td>\n",
       "      <td>0.238</td>\n",
       "      <td>63.21</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000560126</td>\n",
       "      <td>LP6005501-DNA_B01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Barretts_22_EGAF00000525458</td>\n",
       "      <td>22</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>50</td>\n",
       "      <td>T3N3M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Blood</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.42</td>\n",
       "      <td>284.0</td>\n",
       "      <td>4.35</td>\n",
       "      <td>0.226</td>\n",
       "      <td>56.27</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000525458</td>\n",
       "      <td>LP6005691-DNA_B01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Barretts_23_EGAF00000501983</td>\n",
       "      <td>23</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>64</td>\n",
       "      <td>TxNxMx</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blood</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>94.53</td>\n",
       "      <td>311.0</td>\n",
       "      <td>4.31</td>\n",
       "      <td>0.221</td>\n",
       "      <td>86.50</td>\n",
       "      <td>0.979</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000501983</td>\n",
       "      <td>LP6007598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Barretts_2_EGAF00000525461</td>\n",
       "      <td>2</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>F</td>\n",
       "      <td>55</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.86</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.43</td>\n",
       "      <td>0.226</td>\n",
       "      <td>56.30</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000525461</td>\n",
       "      <td>LP6005691-DNA_H01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Barretts_2_EGAF00000814889</td>\n",
       "      <td>2</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>F</td>\n",
       "      <td>55</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.86</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.43</td>\n",
       "      <td>0.226</td>\n",
       "      <td>56.30</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000814889</td>\n",
       "      <td>LP6008023-DNA_C03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Barretts_2_EGAF00000814890</td>\n",
       "      <td>2</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>F</td>\n",
       "      <td>55</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.86</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.43</td>\n",
       "      <td>0.226</td>\n",
       "      <td>56.30</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000814890</td>\n",
       "      <td>LP6008023-DNA_D03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Barretts_3_EGAF00000560128</td>\n",
       "      <td>3</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>61</td>\n",
       "      <td>T3N2M1</td>\n",
       "      <td>IV</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.82</td>\n",
       "      <td>325.0</td>\n",
       "      <td>4.42</td>\n",
       "      <td>0.238</td>\n",
       "      <td>60.52</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "      <td>EGAF00000560128</td>\n",
       "      <td>LP6005501-DNA_D01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Barretts_4_EGAF00000525457</td>\n",
       "      <td>4</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>58</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.48</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.96</td>\n",
       "      <td>0.233</td>\n",
       "      <td>55.51</td>\n",
       "      <td>0.974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000525457</td>\n",
       "      <td>LP6005691-DNA_E01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Barretts_4_EGAF00000814870</td>\n",
       "      <td>4</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>58</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.48</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.96</td>\n",
       "      <td>0.233</td>\n",
       "      <td>55.51</td>\n",
       "      <td>0.974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000814870</td>\n",
       "      <td>LP6008023-DNA_E02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Barretts_4_EGAF00000814871</td>\n",
       "      <td>4</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>58</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.48</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.96</td>\n",
       "      <td>0.233</td>\n",
       "      <td>55.51</td>\n",
       "      <td>0.974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000814871</td>\n",
       "      <td>LP6008023-DNA_F02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Barretts_5_EGAF00000525462</td>\n",
       "      <td>5</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Gloucestershire Royal Hospital, Gloucester</td>\n",
       "      <td>M</td>\n",
       "      <td>64</td>\n",
       "      <td>T1N0M0</td>\n",
       "      <td>IA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>92.23</td>\n",
       "      <td>294.0</td>\n",
       "      <td>4.49</td>\n",
       "      <td>0.237</td>\n",
       "      <td>54.01</td>\n",
       "      <td>0.974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000525462</td>\n",
       "      <td>LP6005691-DNA_C01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Barretts_6_EGAF00000560127</td>\n",
       "      <td>6</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Gloucestershire Royal Hospital, Gloucester</td>\n",
       "      <td>M</td>\n",
       "      <td>73</td>\n",
       "      <td>T3N1Mx</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.70</td>\n",
       "      <td>333.0</td>\n",
       "      <td>4.26</td>\n",
       "      <td>0.239</td>\n",
       "      <td>58.54</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9</td>\n",
       "      <td>EGAF00000560127</td>\n",
       "      <td>LP6005501-DNA_C01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Barretts_7_EGAF00000501982</td>\n",
       "      <td>7</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.48</td>\n",
       "      <td>308.0</td>\n",
       "      <td>4.47</td>\n",
       "      <td>0.235</td>\n",
       "      <td>71.75</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000501982</td>\n",
       "      <td>LP6007595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Barretts_8_EGAF00000525460</td>\n",
       "      <td>8</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Salford Royal Infirmary, Manchester</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.99</td>\n",
       "      <td>297.0</td>\n",
       "      <td>4.99</td>\n",
       "      <td>0.245</td>\n",
       "      <td>57.43</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2</td>\n",
       "      <td>EGAF00000525460</td>\n",
       "      <td>LP6005691-DNA_G01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Barretts_8_EGAF00000814886</td>\n",
       "      <td>8</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Salford Royal Infirmary, Manchester</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.99</td>\n",
       "      <td>297.0</td>\n",
       "      <td>4.99</td>\n",
       "      <td>0.245</td>\n",
       "      <td>57.43</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2</td>\n",
       "      <td>EGAF00000814886</td>\n",
       "      <td>LP6008023-DNA_H02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Barretts_9_EGAF00000525201</td>\n",
       "      <td>9</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T1N0M0</td>\n",
       "      <td>IA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.46</td>\n",
       "      <td>317.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.242</td>\n",
       "      <td>61.34</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000525201</td>\n",
       "      <td>LP6007405-DNA_A01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample_name Patient Sample_type  \\\n",
       "0   Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1   Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2   Barretts_12_EGAF00000525202      12    Barretts   \n",
       "3   Barretts_13_EGAF00000525459      13    Barretts   \n",
       "4   Barretts_14_EGAF00000560129      14    Barretts   \n",
       "5   Barretts_14_EGAF00000814878      14    Barretts   \n",
       "6   Barretts_15_EGAF00000525456      15    Barretts   \n",
       "7   Barretts_15_EGAF00000814867      15    Barretts   \n",
       "8   Barretts_15_EGAF00000814879      15    Barretts   \n",
       "9   Barretts_16_EGAF00000525464      16    Barretts   \n",
       "10  Barretts_17_EGAF00000525463      17    Barretts   \n",
       "11  Barretts_17_EGAF00000814869      17    Barretts   \n",
       "12  Barretts_17_EGAF00000814882      17    Barretts   \n",
       "13  Barretts_17_EGAF00000814883      17    Barretts   \n",
       "14  Barretts_17_EGAF00000814884      17    Barretts   \n",
       "15  Barretts_18_EGAF00000560130      18    Barretts   \n",
       "16  Barretts_19_EGAF00000530863      19    Barretts   \n",
       "17   Barretts_1_EGAF00000501981       1    Barretts   \n",
       "18  Barretts_20_EGAF00000560125      20    Barretts   \n",
       "19  Barretts_21_EGAF00000560126      21    Barretts   \n",
       "20  Barretts_22_EGAF00000525458      22    Barretts   \n",
       "21  Barretts_23_EGAF00000501983      23    Barretts   \n",
       "22   Barretts_2_EGAF00000525461       2    Barretts   \n",
       "23   Barretts_2_EGAF00000814889       2    Barretts   \n",
       "24   Barretts_2_EGAF00000814890       2    Barretts   \n",
       "25   Barretts_3_EGAF00000560128       3    Barretts   \n",
       "26   Barretts_4_EGAF00000525457       4    Barretts   \n",
       "27   Barretts_4_EGAF00000814870       4    Barretts   \n",
       "28   Barretts_4_EGAF00000814871       4    Barretts   \n",
       "29   Barretts_5_EGAF00000525462       5    Barretts   \n",
       "30   Barretts_6_EGAF00000560127       6    Barretts   \n",
       "31   Barretts_7_EGAF00000501982       7    Barretts   \n",
       "32   Barretts_8_EGAF00000525460       8    Barretts   \n",
       "33   Barretts_8_EGAF00000814886       8    Barretts   \n",
       "34   Barretts_9_EGAF00000525201       9    Barretts   \n",
       "\n",
       "                                        Centre Gender  \\\n",
       "0             Addenbrookes Hospital, Cambridge      F   \n",
       "1             Addenbrookes Hospital, Cambridge      M   \n",
       "2             Addenbrookes Hospital, Cambridge      M   \n",
       "3      Royal Surrey County Hospital, Guildford      F   \n",
       "4                   St Thomas Hospital, London      M   \n",
       "5                   St Thomas Hospital, London      M   \n",
       "6      Royal Surrey County Hospital, Guildford      M   \n",
       "7      Royal Surrey County Hospital, Guildford      M   \n",
       "8      Royal Surrey County Hospital, Guildford      M   \n",
       "9             Addenbrookes Hospital, Cambridge      M   \n",
       "10     Royal Surrey County Hospital, Guildford      M   \n",
       "11     Royal Surrey County Hospital, Guildford      M   \n",
       "12     Royal Surrey County Hospital, Guildford      M   \n",
       "13     Royal Surrey County Hospital, Guildford      M   \n",
       "14     Royal Surrey County Hospital, Guildford      M   \n",
       "15                  St Thomas Hospital, London      M   \n",
       "16            Addenbrookes Hospital, Cambridge      M   \n",
       "17            Addenbrookes Hospital, Cambridge      M   \n",
       "18        Edinburgh Royal Infirmary, Edinburgh      F   \n",
       "19     Royal Surrey County Hospital, Guildford      M   \n",
       "20     Royal Surrey County Hospital, Guildford      M   \n",
       "21            Addenbrookes Hospital, Cambridge      M   \n",
       "22                  St Thomas Hospital, London      F   \n",
       "23                  St Thomas Hospital, London      F   \n",
       "24                  St Thomas Hospital, London      F   \n",
       "25            Addenbrookes Hospital, Cambridge      M   \n",
       "26     Royal Surrey County Hospital, Guildford      M   \n",
       "27     Royal Surrey County Hospital, Guildford      M   \n",
       "28     Royal Surrey County Hospital, Guildford      M   \n",
       "29  Gloucestershire Royal Hospital, Gloucester      M   \n",
       "30  Gloucestershire Royal Hospital, Gloucester      M   \n",
       "31            Addenbrookes Hospital, Cambridge      M   \n",
       "32         Salford Royal Infirmary, Manchester      M   \n",
       "33         Salford Royal Infirmary, Manchester      M   \n",
       "34            Addenbrookes Hospital, Cambridge      M   \n",
       "\n",
       "    Age_at_time_of_diagnosis    TNM7 Stage   Germline tissue  \\\n",
       "0                         79  T2N0M0    IB             Blood   \n",
       "1                         67  T4N2M0  IIIC  Normal esophagus   \n",
       "2                         65  T2N1M0   IIB             Blood   \n",
       "3                         61  T2N0M0    IB             Blood   \n",
       "4                         68  T3N1M0  IIIA             Blood   \n",
       "5                         68  T3N1M0  IIIA             Blood   \n",
       "6                         75  T3N3M1    IV             Blood   \n",
       "7                         75  T3N3M1    IV             Blood   \n",
       "8                         75  T3N3M1    IV             Blood   \n",
       "9                         65  T3N1M0  IIIA             Blood   \n",
       "10                        82  T3N1M0  IIIA             Blood   \n",
       "11                        82  T3N1M0  IIIA             Blood   \n",
       "12                        82  T3N1M0  IIIA             Blood   \n",
       "13                        82  T3N1M0  IIIA             Blood   \n",
       "14                        82  T3N1M0  IIIA             Blood   \n",
       "15                        61  T3N1M0  IIIA             Blood   \n",
       "16                        75  T3N0M0   IIA             Blood   \n",
       "17                        62  T2N0M0    IB             Blood   \n",
       "18                        79  T3N1M0  IIIA             Blood   \n",
       "19                        61  T1N0M0    IA             Blood   \n",
       "20                        50  T3N3M0  IIIC             Blood   \n",
       "21                        64  TxNxMx   NaN             Blood   \n",
       "22                        55  T2N1M0   IIB             Blood   \n",
       "23                        55  T2N1M0   IIB             Blood   \n",
       "24                        55  T2N1M0   IIB             Blood   \n",
       "25                        61  T3N2M1    IV             Blood   \n",
       "26                        58  T3N1M0  IIIA             Blood   \n",
       "27                        58  T3N1M0  IIIA             Blood   \n",
       "28                        58  T3N1M0  IIIA             Blood   \n",
       "29                        64  T1N0M0    IA             Blood   \n",
       "30                        73  T3N1Mx   NaN             Blood   \n",
       "31                        75  T3N0M0   IIA             Blood   \n",
       "32                        65  T2N0M0    IB             Blood   \n",
       "33                        65  T2N0M0    IB             Blood   \n",
       "34                        67  T1N0M0    IA             Blood   \n",
       "\n",
       "    BE maximal length (cm)  ...  % aligned pairs median insert size  % mapQ=0  \\\n",
       "0                      4.0  ...            92.57              339.0      5.30   \n",
       "1                      3.0  ...            93.50              300.0      4.77   \n",
       "2                      NaN  ...            93.01              310.0      5.04   \n",
       "3                     15.0  ...            92.02              298.0      5.19   \n",
       "4                      6.0  ...            93.48              317.0      4.70   \n",
       "5                      6.0  ...            93.48              317.0      4.70   \n",
       "6                      9.0  ...            91.07              323.0      4.91   \n",
       "7                      9.0  ...            91.07              323.0      4.91   \n",
       "8                      9.0  ...            91.07              323.0      4.91   \n",
       "9                      NaN  ...            92.38              292.0      4.79   \n",
       "10                     6.0  ...            92.36              319.0      4.79   \n",
       "11                     6.0  ...            92.36              319.0      4.79   \n",
       "12                     6.0  ...            92.36              319.0      4.79   \n",
       "13                     6.0  ...            92.36              319.0      4.79   \n",
       "14                     6.0  ...            92.36              319.0      4.79   \n",
       "15                    10.0  ...            92.88              282.0      4.71   \n",
       "16                     NaN  ...            94.65              440.0      4.32   \n",
       "17                     NaN  ...            94.44              325.0      4.25   \n",
       "18                     NaN  ...            94.31              324.0      4.11   \n",
       "19                     5.0  ...            93.66              330.0      4.42   \n",
       "20                     3.0  ...            92.42              284.0      4.35   \n",
       "21                     2.0  ...            94.53              311.0      4.31   \n",
       "22                     6.0  ...            92.86              315.0      4.43   \n",
       "23                     6.0  ...            92.86              315.0      4.43   \n",
       "24                     6.0  ...            92.86              315.0      4.43   \n",
       "25                     NaN  ...            93.82              325.0      4.42   \n",
       "26                     3.0  ...            91.48              300.0      4.96   \n",
       "27                     3.0  ...            91.48              300.0      4.96   \n",
       "28                     3.0  ...            91.48              300.0      4.96   \n",
       "29                     NaN  ...            92.23              294.0      4.49   \n",
       "30                     NaN  ...            93.70              333.0      4.26   \n",
       "31                     NaN  ...            94.48              308.0      4.47   \n",
       "32                     5.0  ...            91.99              297.0      4.99   \n",
       "33                     5.0  ...            91.99              297.0      4.99   \n",
       "34                     5.0  ...            92.46              317.0      5.04   \n",
       "\n",
       "    error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0            0.245                   55.76            0.975              NaN   \n",
       "1            0.231                   67.45            0.976              NaN   \n",
       "2            0.243                   57.20            0.975              NaN   \n",
       "3            0.222                   54.36            0.973              NaN   \n",
       "4            0.253                   61.75            0.976              NaN   \n",
       "5            0.253                   61.75            0.976              NaN   \n",
       "6            0.237                   67.85            0.977              NaN   \n",
       "7            0.237                   67.85            0.977              NaN   \n",
       "8            0.237                   67.85            0.977              NaN   \n",
       "9            0.234                   57.70            0.975              NaN   \n",
       "10           0.236                   71.36            0.977              NaN   \n",
       "11           0.236                   71.36            0.977              NaN   \n",
       "12           0.236                   71.36            0.977              NaN   \n",
       "13           0.236                   71.36            0.977              NaN   \n",
       "14           0.236                   71.36            0.977              NaN   \n",
       "15           0.240                   60.85            0.976              NaN   \n",
       "16           0.242                   73.95            0.979              NaN   \n",
       "17           0.227                   73.05            0.978              NaN   \n",
       "18           0.227                   62.97            0.977              NaN   \n",
       "19           0.238                   63.21            0.976              NaN   \n",
       "20           0.226                   56.27            0.975              NaN   \n",
       "21           0.221                   86.50            0.979              NaN   \n",
       "22           0.226                   56.30            0.975              NaN   \n",
       "23           0.226                   56.30            0.975              NaN   \n",
       "24           0.226                   56.30            0.975              NaN   \n",
       "25           0.238                   60.52            0.976              NaN   \n",
       "26           0.233                   55.51            0.974              NaN   \n",
       "27           0.233                   55.51            0.974              NaN   \n",
       "28           0.233                   55.51            0.974              NaN   \n",
       "29           0.237                   54.01            0.974              NaN   \n",
       "30           0.239                   58.54            0.975              NaN   \n",
       "31           0.235                   71.75            0.978              NaN   \n",
       "32           0.245                   57.43            0.975              NaN   \n",
       "33           0.245                   57.43            0.975              NaN   \n",
       "34           0.242                   61.34            0.976              NaN   \n",
       "\n",
       "    Sample purity           EGA_ID      EGAIdentifier  \n",
       "0             0.8  EGAF00000530875  LP6007521-DNA_A01  \n",
       "1             0.6  EGAF00000586502  LP6005335-DNA_A01  \n",
       "2             0.3  EGAF00000525202  LP6007410-DNA_A01  \n",
       "3             0.6  EGAF00000525459  LP6005691-DNA_F01  \n",
       "4             0.3  EGAF00000560129  LP6005501-DNA_E01  \n",
       "5             0.3  EGAF00000814878  LP6008023-DNA_A01  \n",
       "6             0.7  EGAF00000525456  LP6005691-DNA_A01  \n",
       "7             0.7  EGAF00000814867  LP6008023-DNA_E01  \n",
       "8             0.7  EGAF00000814879  LP6008023-DNA_F01  \n",
       "9             0.8  EGAF00000525464  LP6005691-DNA_A02  \n",
       "10            0.7  EGAF00000525463  LP6005691-DNA_D01  \n",
       "11            0.7  EGAF00000814869  LP6008023-DNA_A02  \n",
       "12            0.7  EGAF00000814882  LP6008023-DNA_B02  \n",
       "13            0.7  EGAF00000814883  LP6008023-DNA_C02  \n",
       "14            0.7  EGAF00000814884  LP6008023-DNA_D02  \n",
       "15            0.3  EGAF00000560130  LP6005501-DNA_F01  \n",
       "16            0.5  EGAF00000530863  LP2000110-DNA_A01  \n",
       "17            0.6  EGAF00000501981          LP6007592  \n",
       "18            0.3  EGAF00000560125  LP6005501-DNA_A01  \n",
       "19            0.7  EGAF00000560126  LP6005501-DNA_B01  \n",
       "20            0.3  EGAF00000525458  LP6005691-DNA_B01  \n",
       "21            0.3  EGAF00000501983          LP6007598  \n",
       "22            0.3  EGAF00000525461  LP6005691-DNA_H01  \n",
       "23            0.3  EGAF00000814889  LP6008023-DNA_C03  \n",
       "24            0.3  EGAF00000814890  LP6008023-DNA_D03  \n",
       "25            0.4  EGAF00000560128  LP6005501-DNA_D01  \n",
       "26            0.7  EGAF00000525457  LP6005691-DNA_E01  \n",
       "27            0.7  EGAF00000814870  LP6008023-DNA_E02  \n",
       "28            0.7  EGAF00000814871  LP6008023-DNA_F02  \n",
       "29            0.8  EGAF00000525462  LP6005691-DNA_C01  \n",
       "30            0.9  EGAF00000560127  LP6005501-DNA_C01  \n",
       "31            0.7  EGAF00000501982          LP6007595  \n",
       "32            0.2  EGAF00000525460  LP6005691-DNA_G01  \n",
       "33            0.2  EGAF00000814886  LP6008023-DNA_H02  \n",
       "34            0.5  EGAF00000525201  LP6007405-DNA_A01  \n",
       "\n",
       "[35 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "      <th>EGA_ID</th>\n",
       "      <th>EGAIdentifier</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000530875</td>\n",
       "      <td>LP6007521-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000586502</td>\n",
       "      <td>LP6005335-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000525202</td>\n",
       "      <td>LP6007410-DNA_A01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  ...  % aligned pairs  \\\n",
       "0    IB             Blood                     4.0  ...            92.57   \n",
       "1  IIIC  Normal esophagus                     3.0  ...            93.50   \n",
       "2   IIB             Blood                     NaN  ...            93.01   \n",
       "\n",
       "  median insert size  % mapQ=0  error rate HiQ  genome coverage mapQ>3  \\\n",
       "0              339.0      5.30           0.245                   55.76   \n",
       "1              300.0      4.77           0.231                   67.45   \n",
       "2              310.0      5.04           0.243                   57.20   \n",
       "\n",
       "   Callable genome  totalNumberSnps  Sample purity           EGA_ID  \\\n",
       "0            0.975              NaN            0.8  EGAF00000530875   \n",
       "1            0.976              NaN            0.6  EGAF00000586502   \n",
       "2            0.975              NaN            0.3  EGAF00000525202   \n",
       "\n",
       "       EGAIdentifier  \n",
       "0  LP6007521-DNA_A01  \n",
       "1  LP6005335-DNA_A01  \n",
       "2  LP6007410-DNA_A01  \n",
       "\n",
       "[3 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_paired_species_WOL_scrubbed_rmBad  =  35\n",
      "<artifact: FeatureTable[Frequency] uuid: 08844b7f-5746-42e2-ba18-3062a7741207>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL - scrubbed\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired_scrubbed/feature-table.biom',\n",
    "                         'BE_Ross_paired_species_WOL_scrubbed_rmBad',\n",
    "                        remove_bad_microbes=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "19e63542-cdce-4af9-ac9a-9bb3309d559f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "      <th>EGA_ID</th>\n",
       "      <th>EGAIdentifier</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000530875</td>\n",
       "      <td>LP6007521-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000586502</td>\n",
       "      <td>LP6005335-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000525202</td>\n",
       "      <td>LP6007410-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Barretts_13_EGAF00000525459</td>\n",
       "      <td>13</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>F</td>\n",
       "      <td>61</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>15.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.02</td>\n",
       "      <td>298.0</td>\n",
       "      <td>5.19</td>\n",
       "      <td>0.222</td>\n",
       "      <td>54.36</td>\n",
       "      <td>0.973</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000525459</td>\n",
       "      <td>LP6005691-DNA_F01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Barretts_14_EGAF00000560129</td>\n",
       "      <td>14</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>M</td>\n",
       "      <td>68</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.48</td>\n",
       "      <td>317.0</td>\n",
       "      <td>4.70</td>\n",
       "      <td>0.253</td>\n",
       "      <td>61.75</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000560129</td>\n",
       "      <td>LP6005501-DNA_E01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Barretts_15_EGAF00000525456</td>\n",
       "      <td>15</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N3M1</td>\n",
       "      <td>IV</td>\n",
       "      <td>Blood</td>\n",
       "      <td>9.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.07</td>\n",
       "      <td>323.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.237</td>\n",
       "      <td>67.85</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000525456</td>\n",
       "      <td>LP6005691-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Barretts_16_EGAF00000525464</td>\n",
       "      <td>16</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>92.38</td>\n",
       "      <td>292.0</td>\n",
       "      <td>4.79</td>\n",
       "      <td>0.234</td>\n",
       "      <td>57.70</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000525464</td>\n",
       "      <td>LP6005691-DNA_A02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Barretts_17_EGAF00000525463</td>\n",
       "      <td>17</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>82</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.36</td>\n",
       "      <td>319.0</td>\n",
       "      <td>4.79</td>\n",
       "      <td>0.236</td>\n",
       "      <td>71.36</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000525463</td>\n",
       "      <td>LP6005691-DNA_D01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Barretts_18_EGAF00000560130</td>\n",
       "      <td>18</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>M</td>\n",
       "      <td>61</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>10.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.88</td>\n",
       "      <td>282.0</td>\n",
       "      <td>4.71</td>\n",
       "      <td>0.240</td>\n",
       "      <td>60.85</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000560130</td>\n",
       "      <td>LP6005501-DNA_F01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Barretts_19_EGAF00000530863</td>\n",
       "      <td>19</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.65</td>\n",
       "      <td>440.0</td>\n",
       "      <td>4.32</td>\n",
       "      <td>0.242</td>\n",
       "      <td>73.95</td>\n",
       "      <td>0.979</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000530863</td>\n",
       "      <td>LP2000110-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Barretts_1_EGAF00000501981</td>\n",
       "      <td>1</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>62</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.44</td>\n",
       "      <td>325.0</td>\n",
       "      <td>4.25</td>\n",
       "      <td>0.227</td>\n",
       "      <td>73.05</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000501981</td>\n",
       "      <td>LP6007592</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Barretts_20_EGAF00000560125</td>\n",
       "      <td>20</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Edinburgh Royal Infirmary, Edinburgh</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.31</td>\n",
       "      <td>324.0</td>\n",
       "      <td>4.11</td>\n",
       "      <td>0.227</td>\n",
       "      <td>62.97</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000560125</td>\n",
       "      <td>LP6005501-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Barretts_21_EGAF00000560126</td>\n",
       "      <td>21</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>61</td>\n",
       "      <td>T1N0M0</td>\n",
       "      <td>IA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.66</td>\n",
       "      <td>330.0</td>\n",
       "      <td>4.42</td>\n",
       "      <td>0.238</td>\n",
       "      <td>63.21</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000560126</td>\n",
       "      <td>LP6005501-DNA_B01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Barretts_22_EGAF00000525458</td>\n",
       "      <td>22</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>50</td>\n",
       "      <td>T3N3M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Blood</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.42</td>\n",
       "      <td>284.0</td>\n",
       "      <td>4.35</td>\n",
       "      <td>0.226</td>\n",
       "      <td>56.27</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000525458</td>\n",
       "      <td>LP6005691-DNA_B01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Barretts_23_EGAF00000501983</td>\n",
       "      <td>23</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>64</td>\n",
       "      <td>TxNxMx</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blood</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>94.53</td>\n",
       "      <td>311.0</td>\n",
       "      <td>4.31</td>\n",
       "      <td>0.221</td>\n",
       "      <td>86.50</td>\n",
       "      <td>0.979</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000501983</td>\n",
       "      <td>LP6007598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Barretts_2_EGAF00000525461</td>\n",
       "      <td>2</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>F</td>\n",
       "      <td>55</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.86</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.43</td>\n",
       "      <td>0.226</td>\n",
       "      <td>56.30</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000525461</td>\n",
       "      <td>LP6005691-DNA_H01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Barretts_3_EGAF00000560128</td>\n",
       "      <td>3</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>61</td>\n",
       "      <td>T3N2M1</td>\n",
       "      <td>IV</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.82</td>\n",
       "      <td>325.0</td>\n",
       "      <td>4.42</td>\n",
       "      <td>0.238</td>\n",
       "      <td>60.52</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "      <td>EGAF00000560128</td>\n",
       "      <td>LP6005501-DNA_D01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Barretts_4_EGAF00000525457</td>\n",
       "      <td>4</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>58</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.48</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.96</td>\n",
       "      <td>0.233</td>\n",
       "      <td>55.51</td>\n",
       "      <td>0.974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000525457</td>\n",
       "      <td>LP6005691-DNA_E01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Barretts_5_EGAF00000525462</td>\n",
       "      <td>5</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Gloucestershire Royal Hospital, Gloucester</td>\n",
       "      <td>M</td>\n",
       "      <td>64</td>\n",
       "      <td>T1N0M0</td>\n",
       "      <td>IA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>92.23</td>\n",
       "      <td>294.0</td>\n",
       "      <td>4.49</td>\n",
       "      <td>0.237</td>\n",
       "      <td>54.01</td>\n",
       "      <td>0.974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000525462</td>\n",
       "      <td>LP6005691-DNA_C01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Barretts_6_EGAF00000560127</td>\n",
       "      <td>6</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Gloucestershire Royal Hospital, Gloucester</td>\n",
       "      <td>M</td>\n",
       "      <td>73</td>\n",
       "      <td>T3N1Mx</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.70</td>\n",
       "      <td>333.0</td>\n",
       "      <td>4.26</td>\n",
       "      <td>0.239</td>\n",
       "      <td>58.54</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9</td>\n",
       "      <td>EGAF00000560127</td>\n",
       "      <td>LP6005501-DNA_C01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Barretts_7_EGAF00000501982</td>\n",
       "      <td>7</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.48</td>\n",
       "      <td>308.0</td>\n",
       "      <td>4.47</td>\n",
       "      <td>0.235</td>\n",
       "      <td>71.75</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000501982</td>\n",
       "      <td>LP6007595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Barretts_8_EGAF00000525460</td>\n",
       "      <td>8</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Salford Royal Infirmary, Manchester</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>91.99</td>\n",
       "      <td>297.0</td>\n",
       "      <td>4.99</td>\n",
       "      <td>0.245</td>\n",
       "      <td>57.43</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2</td>\n",
       "      <td>EGAF00000525460</td>\n",
       "      <td>LP6005691-DNA_G01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Barretts_9_EGAF00000525201</td>\n",
       "      <td>9</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T1N0M0</td>\n",
       "      <td>IA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.46</td>\n",
       "      <td>317.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.242</td>\n",
       "      <td>61.34</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000525201</td>\n",
       "      <td>LP6007405-DNA_A01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample_name Patient Sample_type  \\\n",
       "0   Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1   Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2   Barretts_12_EGAF00000525202      12    Barretts   \n",
       "3   Barretts_13_EGAF00000525459      13    Barretts   \n",
       "4   Barretts_14_EGAF00000560129      14    Barretts   \n",
       "6   Barretts_15_EGAF00000525456      15    Barretts   \n",
       "9   Barretts_16_EGAF00000525464      16    Barretts   \n",
       "10  Barretts_17_EGAF00000525463      17    Barretts   \n",
       "15  Barretts_18_EGAF00000560130      18    Barretts   \n",
       "16  Barretts_19_EGAF00000530863      19    Barretts   \n",
       "17   Barretts_1_EGAF00000501981       1    Barretts   \n",
       "18  Barretts_20_EGAF00000560125      20    Barretts   \n",
       "19  Barretts_21_EGAF00000560126      21    Barretts   \n",
       "20  Barretts_22_EGAF00000525458      22    Barretts   \n",
       "21  Barretts_23_EGAF00000501983      23    Barretts   \n",
       "22   Barretts_2_EGAF00000525461       2    Barretts   \n",
       "25   Barretts_3_EGAF00000560128       3    Barretts   \n",
       "26   Barretts_4_EGAF00000525457       4    Barretts   \n",
       "29   Barretts_5_EGAF00000525462       5    Barretts   \n",
       "30   Barretts_6_EGAF00000560127       6    Barretts   \n",
       "31   Barretts_7_EGAF00000501982       7    Barretts   \n",
       "32   Barretts_8_EGAF00000525460       8    Barretts   \n",
       "34   Barretts_9_EGAF00000525201       9    Barretts   \n",
       "\n",
       "                                        Centre Gender  \\\n",
       "0             Addenbrookes Hospital, Cambridge      F   \n",
       "1             Addenbrookes Hospital, Cambridge      M   \n",
       "2             Addenbrookes Hospital, Cambridge      M   \n",
       "3      Royal Surrey County Hospital, Guildford      F   \n",
       "4                   St Thomas Hospital, London      M   \n",
       "6      Royal Surrey County Hospital, Guildford      M   \n",
       "9             Addenbrookes Hospital, Cambridge      M   \n",
       "10     Royal Surrey County Hospital, Guildford      M   \n",
       "15                  St Thomas Hospital, London      M   \n",
       "16            Addenbrookes Hospital, Cambridge      M   \n",
       "17            Addenbrookes Hospital, Cambridge      M   \n",
       "18        Edinburgh Royal Infirmary, Edinburgh      F   \n",
       "19     Royal Surrey County Hospital, Guildford      M   \n",
       "20     Royal Surrey County Hospital, Guildford      M   \n",
       "21            Addenbrookes Hospital, Cambridge      M   \n",
       "22                  St Thomas Hospital, London      F   \n",
       "25            Addenbrookes Hospital, Cambridge      M   \n",
       "26     Royal Surrey County Hospital, Guildford      M   \n",
       "29  Gloucestershire Royal Hospital, Gloucester      M   \n",
       "30  Gloucestershire Royal Hospital, Gloucester      M   \n",
       "31            Addenbrookes Hospital, Cambridge      M   \n",
       "32         Salford Royal Infirmary, Manchester      M   \n",
       "34            Addenbrookes Hospital, Cambridge      M   \n",
       "\n",
       "    Age_at_time_of_diagnosis    TNM7 Stage   Germline tissue  \\\n",
       "0                         79  T2N0M0    IB             Blood   \n",
       "1                         67  T4N2M0  IIIC  Normal esophagus   \n",
       "2                         65  T2N1M0   IIB             Blood   \n",
       "3                         61  T2N0M0    IB             Blood   \n",
       "4                         68  T3N1M0  IIIA             Blood   \n",
       "6                         75  T3N3M1    IV             Blood   \n",
       "9                         65  T3N1M0  IIIA             Blood   \n",
       "10                        82  T3N1M0  IIIA             Blood   \n",
       "15                        61  T3N1M0  IIIA             Blood   \n",
       "16                        75  T3N0M0   IIA             Blood   \n",
       "17                        62  T2N0M0    IB             Blood   \n",
       "18                        79  T3N1M0  IIIA             Blood   \n",
       "19                        61  T1N0M0    IA             Blood   \n",
       "20                        50  T3N3M0  IIIC             Blood   \n",
       "21                        64  TxNxMx   NaN             Blood   \n",
       "22                        55  T2N1M0   IIB             Blood   \n",
       "25                        61  T3N2M1    IV             Blood   \n",
       "26                        58  T3N1M0  IIIA             Blood   \n",
       "29                        64  T1N0M0    IA             Blood   \n",
       "30                        73  T3N1Mx   NaN             Blood   \n",
       "31                        75  T3N0M0   IIA             Blood   \n",
       "32                        65  T2N0M0    IB             Blood   \n",
       "34                        67  T1N0M0    IA             Blood   \n",
       "\n",
       "    BE maximal length (cm)  ...  % aligned pairs median insert size  % mapQ=0  \\\n",
       "0                      4.0  ...            92.57              339.0      5.30   \n",
       "1                      3.0  ...            93.50              300.0      4.77   \n",
       "2                      NaN  ...            93.01              310.0      5.04   \n",
       "3                     15.0  ...            92.02              298.0      5.19   \n",
       "4                      6.0  ...            93.48              317.0      4.70   \n",
       "6                      9.0  ...            91.07              323.0      4.91   \n",
       "9                      NaN  ...            92.38              292.0      4.79   \n",
       "10                     6.0  ...            92.36              319.0      4.79   \n",
       "15                    10.0  ...            92.88              282.0      4.71   \n",
       "16                     NaN  ...            94.65              440.0      4.32   \n",
       "17                     NaN  ...            94.44              325.0      4.25   \n",
       "18                     NaN  ...            94.31              324.0      4.11   \n",
       "19                     5.0  ...            93.66              330.0      4.42   \n",
       "20                     3.0  ...            92.42              284.0      4.35   \n",
       "21                     2.0  ...            94.53              311.0      4.31   \n",
       "22                     6.0  ...            92.86              315.0      4.43   \n",
       "25                     NaN  ...            93.82              325.0      4.42   \n",
       "26                     3.0  ...            91.48              300.0      4.96   \n",
       "29                     NaN  ...            92.23              294.0      4.49   \n",
       "30                     NaN  ...            93.70              333.0      4.26   \n",
       "31                     NaN  ...            94.48              308.0      4.47   \n",
       "32                     5.0  ...            91.99              297.0      4.99   \n",
       "34                     5.0  ...            92.46              317.0      5.04   \n",
       "\n",
       "    error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0            0.245                   55.76            0.975              NaN   \n",
       "1            0.231                   67.45            0.976              NaN   \n",
       "2            0.243                   57.20            0.975              NaN   \n",
       "3            0.222                   54.36            0.973              NaN   \n",
       "4            0.253                   61.75            0.976              NaN   \n",
       "6            0.237                   67.85            0.977              NaN   \n",
       "9            0.234                   57.70            0.975              NaN   \n",
       "10           0.236                   71.36            0.977              NaN   \n",
       "15           0.240                   60.85            0.976              NaN   \n",
       "16           0.242                   73.95            0.979              NaN   \n",
       "17           0.227                   73.05            0.978              NaN   \n",
       "18           0.227                   62.97            0.977              NaN   \n",
       "19           0.238                   63.21            0.976              NaN   \n",
       "20           0.226                   56.27            0.975              NaN   \n",
       "21           0.221                   86.50            0.979              NaN   \n",
       "22           0.226                   56.30            0.975              NaN   \n",
       "25           0.238                   60.52            0.976              NaN   \n",
       "26           0.233                   55.51            0.974              NaN   \n",
       "29           0.237                   54.01            0.974              NaN   \n",
       "30           0.239                   58.54            0.975              NaN   \n",
       "31           0.235                   71.75            0.978              NaN   \n",
       "32           0.245                   57.43            0.975              NaN   \n",
       "34           0.242                   61.34            0.976              NaN   \n",
       "\n",
       "    Sample purity           EGA_ID      EGAIdentifier  \n",
       "0             0.8  EGAF00000530875  LP6007521-DNA_A01  \n",
       "1             0.6  EGAF00000586502  LP6005335-DNA_A01  \n",
       "2             0.3  EGAF00000525202  LP6007410-DNA_A01  \n",
       "3             0.6  EGAF00000525459  LP6005691-DNA_F01  \n",
       "4             0.3  EGAF00000560129  LP6005501-DNA_E01  \n",
       "6             0.7  EGAF00000525456  LP6005691-DNA_A01  \n",
       "9             0.8  EGAF00000525464  LP6005691-DNA_A02  \n",
       "10            0.7  EGAF00000525463  LP6005691-DNA_D01  \n",
       "15            0.3  EGAF00000560130  LP6005501-DNA_F01  \n",
       "16            0.5  EGAF00000530863  LP2000110-DNA_A01  \n",
       "17            0.6  EGAF00000501981          LP6007592  \n",
       "18            0.3  EGAF00000560125  LP6005501-DNA_A01  \n",
       "19            0.7  EGAF00000560126  LP6005501-DNA_B01  \n",
       "20            0.3  EGAF00000525458  LP6005691-DNA_B01  \n",
       "21            0.3  EGAF00000501983          LP6007598  \n",
       "22            0.3  EGAF00000525461  LP6005691-DNA_H01  \n",
       "25            0.4  EGAF00000560128  LP6005501-DNA_D01  \n",
       "26            0.7  EGAF00000525457  LP6005691-DNA_E01  \n",
       "29            0.8  EGAF00000525462  LP6005691-DNA_C01  \n",
       "30            0.9  EGAF00000560127  LP6005501-DNA_C01  \n",
       "31            0.7  EGAF00000501982          LP6007595  \n",
       "32            0.2  EGAF00000525460  LP6005691-DNA_G01  \n",
       "34            0.5  EGAF00000525201  LP6007405-DNA_A01  \n",
       "\n",
       "[23 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "      <th>EGA_ID</th>\n",
       "      <th>EGAIdentifier</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000530875</td>\n",
       "      <td>LP6007521-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000586502</td>\n",
       "      <td>LP6005335-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000525202</td>\n",
       "      <td>LP6007410-DNA_A01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  ...  % aligned pairs  \\\n",
       "0    IB             Blood                     4.0  ...            92.57   \n",
       "1  IIIC  Normal esophagus                     3.0  ...            93.50   \n",
       "2   IIB             Blood                     NaN  ...            93.01   \n",
       "\n",
       "  median insert size  % mapQ=0  error rate HiQ  genome coverage mapQ>3  \\\n",
       "0              339.0      5.30           0.245                   55.76   \n",
       "1              300.0      4.77           0.231                   67.45   \n",
       "2              310.0      5.04           0.243                   57.20   \n",
       "\n",
       "   Callable genome  totalNumberSnps  Sample purity           EGA_ID  \\\n",
       "0            0.975              NaN            0.8  EGAF00000530875   \n",
       "1            0.976              NaN            0.6  EGAF00000586502   \n",
       "2            0.975              NaN            0.3  EGAF00000525202   \n",
       "\n",
       "       EGAIdentifier  \n",
       "0  LP6007521-DNA_A01  \n",
       "1  LP6005335-DNA_A01  \n",
       "2  LP6007410-DNA_A01  \n",
       "\n",
       "[3 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_ginny_species_WOL_scrubbed  =  23\n",
      "<artifact: FeatureTable[Frequency] uuid: 14598780-af41-4de2-a1d3-c094f45ea852>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL - scrubbed\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired_scrubbed/feature-table.biom',\n",
    "                         'BE_Ross_ginny_species_WOL_scrubbed',\n",
    "                        Ginny_list=True)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "95766efe-d99a-427c-811c-6accaf1aec6e",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Not in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1223f030-f489-4d11-b7a7-d38f4aa41434",
   "metadata": {},
   "outputs": [],
   "source": [
    "#WOL - scrubbed\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired_scrubbed/feature-table.biom',\n",
    "                         'BE_Ross_paired_species_WOL_scrubbed')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5e62bcf1-57a4-4be9-862b-d5927f906bc3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.3</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.5</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.2</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                       79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                       67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                       65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue BE maximal length (cm) Tumor length (mm)  \\\n",
       "0    IB             Blood                    4.0              15.0   \n",
       "1  IIIC  Normal esophagus                    3.0              70.0   \n",
       "2   IIB             Blood                    NaN              29.0   \n",
       "\n",
       "  purity-filtered pairs % aligned pairs median insert size % mapQ=0  \\\n",
       "0           935,820,463           92.57              339.0      5.3   \n",
       "1         1,135,954,265            93.5              300.0     4.77   \n",
       "2           955,544,208           93.01              310.0     5.04   \n",
       "\n",
       "  error rate HiQ genome coverage mapQ>3 Callable genome totalNumberSnps  \\\n",
       "0          0.245                  55.76           0.975             NaN   \n",
       "1          0.231                  67.45           0.976             NaN   \n",
       "2          0.243                   57.2           0.975             NaN   \n",
       "\n",
       "  Sample purity  \n",
       "0           0.8  \n",
       "1           0.6  \n",
       "2           0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_exact_Ross_paired_genome_WOL_scrubbed  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_paired_species_WOL  =  35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.3</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.5</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.2</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                       79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                       67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                       65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue BE maximal length (cm) Tumor length (mm)  \\\n",
       "0    IB             Blood                    4.0              15.0   \n",
       "1  IIIC  Normal esophagus                    3.0              70.0   \n",
       "2   IIB             Blood                    NaN              29.0   \n",
       "\n",
       "  purity-filtered pairs % aligned pairs median insert size % mapQ=0  \\\n",
       "0           935,820,463           92.57              339.0      5.3   \n",
       "1         1,135,954,265            93.5              300.0     4.77   \n",
       "2           955,544,208           93.01              310.0     5.04   \n",
       "\n",
       "  error rate HiQ genome coverage mapQ>3 Callable genome totalNumberSnps  \\\n",
       "0          0.245                  55.76           0.975             NaN   \n",
       "1          0.231                  67.45           0.976             NaN   \n",
       "2          0.243                   57.2           0.975             NaN   \n",
       "\n",
       "  Sample purity  \n",
       "0           0.8  \n",
       "1           0.6  \n",
       "2           0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_exact_Ross_paired_species_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_paired_genome_WOL  =  35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_zebra_Ross_paired_genome_WOL  =  35\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.3</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.5</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.2</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                       79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                       67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                       65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue BE maximal length (cm) Tumor length (mm)  \\\n",
       "0    IB             Blood                    4.0              15.0   \n",
       "1  IIIC  Normal esophagus                    3.0              70.0   \n",
       "2   IIB             Blood                    NaN              29.0   \n",
       "\n",
       "  purity-filtered pairs % aligned pairs median insert size % mapQ=0  \\\n",
       "0           935,820,463           92.57              339.0      5.3   \n",
       "1         1,135,954,265            93.5              300.0     4.77   \n",
       "2           955,544,208           93.01              310.0     5.04   \n",
       "\n",
       "  error rate HiQ genome coverage mapQ>3 Callable genome totalNumberSnps  \\\n",
       "0          0.245                  55.76           0.975             NaN   \n",
       "1          0.231                  67.45           0.976             NaN   \n",
       "2          0.243                   57.2           0.975             NaN   \n",
       "\n",
       "  Sample purity  \n",
       "0           0.8  \n",
       "1           0.6  \n",
       "2           0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_zebra_exact_Ross_paired_genome_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_paired_species_rep200  =  35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.3</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.5</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.2</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                       79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                       67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                       65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue BE maximal length (cm) Tumor length (mm)  \\\n",
       "0    IB             Blood                    4.0              15.0   \n",
       "1  IIIC  Normal esophagus                    3.0              70.0   \n",
       "2   IIB             Blood                    NaN              29.0   \n",
       "\n",
       "  purity-filtered pairs % aligned pairs median insert size % mapQ=0  \\\n",
       "0           935,820,463           92.57              339.0      5.3   \n",
       "1         1,135,954,265            93.5              300.0     4.77   \n",
       "2           955,544,208           93.01              310.0     5.04   \n",
       "\n",
       "  error rate HiQ genome coverage mapQ>3 Callable genome totalNumberSnps  \\\n",
       "0          0.245                  55.76           0.975             NaN   \n",
       "1          0.231                  67.45           0.976             NaN   \n",
       "2          0.243                   57.2           0.975             NaN   \n",
       "\n",
       "  Sample purity  \n",
       "0           0.8  \n",
       "1           0.6  \n",
       "2           0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_exact_Ross_paired_species_rep200  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL - scrubbed\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired_scrubbed/feature-table.biom',\n",
    "                         'BE_exact_Ross_paired_genome_WOL_scrubbed',\n",
    "                        exact_pairs=True)\n",
    "\n",
    "#WOL\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'BE_Ross_paired_species_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'BE_exact_Ross_paired_species_WOL',\n",
    "                         exact_pairs=True)\n",
    "\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'BE_Ross_paired_genome_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'BE_zebra_Ross_paired_genome_WOL',\n",
    "                        zebra = 'Ross_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'BE_zebra_exact_Ross_paired_genome_WOL',\n",
    "                         exact_pairs=True,\n",
    "                        zebra = 'Ross_WOL')\n",
    "\n",
    "#Rep200\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_rep200_R1_R2_paired.biom',\n",
    "                         'BE_Ross_paired_species_rep200')\n",
    "\n",
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_rep200_R1_R2_paired.biom',\n",
    "                         'BE_exact_Ross_paired_species_rep200',\n",
    "                        exact_pairs=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "326381c7-499e-4722-bd04-09f663075f42",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_combo_species_WOL  =  35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_combo_R1_R2_paired_unpaired.biom',\n",
    "                         'BE_Ross_combo_species_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "df2bbcea-9caf-4927-9a66-49dcf4a79068",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Ross_unpaired_species_WOL  =  35\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_unpaired.biom',\n",
    "                         'BE_Ross_unpaired_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7242c10-b39d-4545-86d9-3b2ec15bb22b",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### EAC"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6720ae75-ba3c-4b51-86f1-ba062349c46e",
   "metadata": {
    "tags": []
   },
   "source": [
    "###### Used in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "66ae8eb5-5e40-4dba-b8b0-a394e064d578",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_paired_species_WOL_scrubbed_rmBad  =  30\n",
      "<artifact: FeatureTable[Frequency] uuid: 49ed5da6-ab3f-48dc-9ed7-8a6cce391086>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL - scrubbed\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired_scrubbed/feature-table.biom',\n",
    "                         'EAC_Ross_paired_species_WOL_scrubbed_rmBad',\n",
    "                        remove_bad_microbes=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5c01dca6-9c4c-45f8-a19a-9a52aff69ce8",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "      <th>EGA_ID</th>\n",
       "      <th>EGAIdentifier</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000530893</td>\n",
       "      <td>LP6007520-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000586547</td>\n",
       "      <td>LP6005334-DNA_A03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "      <td>EGAF00000525525</td>\n",
       "      <td>LP6007409-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>Tumor_13_EGAF00000525466</td>\n",
       "      <td>13</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>F</td>\n",
       "      <td>61</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>15.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.49</td>\n",
       "      <td>310.0</td>\n",
       "      <td>4.53</td>\n",
       "      <td>0.217</td>\n",
       "      <td>82.11</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000525466</td>\n",
       "      <td>LP6005690-DNA_F01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>Tumor_14_EGAF00000560172</td>\n",
       "      <td>14</td>\n",
       "      <td>EAC</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>M</td>\n",
       "      <td>68</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.87</td>\n",
       "      <td>314.0</td>\n",
       "      <td>4.63</td>\n",
       "      <td>0.254</td>\n",
       "      <td>63.20</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9</td>\n",
       "      <td>EGAF00000560172</td>\n",
       "      <td>LP6005500-DNA_E01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>Tumor_15_EGAF00000525475</td>\n",
       "      <td>15</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N3M1</td>\n",
       "      <td>IV</td>\n",
       "      <td>Blood</td>\n",
       "      <td>9.0</td>\n",
       "      <td>...</td>\n",
       "      <td>90.40</td>\n",
       "      <td>308.0</td>\n",
       "      <td>6.36</td>\n",
       "      <td>0.240</td>\n",
       "      <td>83.40</td>\n",
       "      <td>0.973</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000525475</td>\n",
       "      <td>LP6005690-DNA_C01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>Tumor_16_EGAF00000525474</td>\n",
       "      <td>16</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>92.44</td>\n",
       "      <td>308.0</td>\n",
       "      <td>5.12</td>\n",
       "      <td>0.233</td>\n",
       "      <td>65.23</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000525474</td>\n",
       "      <td>LP6005690-DNA_F03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>Tumor_17_EGAF00000525479</td>\n",
       "      <td>17</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>82</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.48</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.05</td>\n",
       "      <td>0.239</td>\n",
       "      <td>62.44</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>EGAF00000525479</td>\n",
       "      <td>LP6005690-DNA_B02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>Tumor_18_EGAF00000560173</td>\n",
       "      <td>18</td>\n",
       "      <td>EAC</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>M</td>\n",
       "      <td>61</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>10.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.91</td>\n",
       "      <td>312.0</td>\n",
       "      <td>5.46</td>\n",
       "      <td>0.706</td>\n",
       "      <td>76.41</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "      <td>EGAF00000560173</td>\n",
       "      <td>LP6005500-DNA_F01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>Tumor_19_EGAF00000530869</td>\n",
       "      <td>19</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.76</td>\n",
       "      <td>430.0</td>\n",
       "      <td>4.28</td>\n",
       "      <td>0.232</td>\n",
       "      <td>80.45</td>\n",
       "      <td>0.979</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "      <td>EGAF00000530869</td>\n",
       "      <td>LP2000104-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>Tumor_1_EGAF00000501976</td>\n",
       "      <td>1</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>62</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>95.13</td>\n",
       "      <td>297.0</td>\n",
       "      <td>4.27</td>\n",
       "      <td>0.225</td>\n",
       "      <td>75.79</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000501976</td>\n",
       "      <td>LP6007591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>Tumor_20_EGAF00000560131</td>\n",
       "      <td>20</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Edinburgh Royal Infirmary, Edinburgh</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.00</td>\n",
       "      <td>308.0</td>\n",
       "      <td>4.67</td>\n",
       "      <td>0.231</td>\n",
       "      <td>77.76</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000560131</td>\n",
       "      <td>LP6005500-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>Tumor_21_EGAF00000560132</td>\n",
       "      <td>21</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>61</td>\n",
       "      <td>T1N0M0</td>\n",
       "      <td>IA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.79</td>\n",
       "      <td>308.0</td>\n",
       "      <td>4.80</td>\n",
       "      <td>0.251</td>\n",
       "      <td>64.93</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000560132</td>\n",
       "      <td>LP6005500-DNA_B01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>Tumor_22_EGAF00000525477</td>\n",
       "      <td>22</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>50</td>\n",
       "      <td>T3N3M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Blood</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.89</td>\n",
       "      <td>308.0</td>\n",
       "      <td>4.48</td>\n",
       "      <td>0.222</td>\n",
       "      <td>63.04</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000525477</td>\n",
       "      <td>LP6005690-DNA_E01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>Tumor_23_EGAF00000501978</td>\n",
       "      <td>23</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>64</td>\n",
       "      <td>TxNxMx</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blood</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>94.48</td>\n",
       "      <td>305.0</td>\n",
       "      <td>4.18</td>\n",
       "      <td>0.216</td>\n",
       "      <td>82.51</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000501978</td>\n",
       "      <td>LP6007597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>Tumor_2_EGAF00000525468</td>\n",
       "      <td>2</td>\n",
       "      <td>EAC</td>\n",
       "      <td>St Thomas Hospital, London</td>\n",
       "      <td>F</td>\n",
       "      <td>55</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>6.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.98</td>\n",
       "      <td>303.0</td>\n",
       "      <td>5.17</td>\n",
       "      <td>0.220</td>\n",
       "      <td>89.29</td>\n",
       "      <td>0.977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000525468</td>\n",
       "      <td>LP6005690-DNA_H01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>Tumor_3_EGAF00000560133</td>\n",
       "      <td>3</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>61</td>\n",
       "      <td>T3N2M1</td>\n",
       "      <td>IV</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>92.92</td>\n",
       "      <td>312.0</td>\n",
       "      <td>5.49</td>\n",
       "      <td>0.288</td>\n",
       "      <td>60.89</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>EGAF00000560133</td>\n",
       "      <td>LP6005500-DNA_D01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>Tumor_4_EGAF00000525476</td>\n",
       "      <td>4</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Royal Surrey County Hospital, Guildford</td>\n",
       "      <td>M</td>\n",
       "      <td>58</td>\n",
       "      <td>T3N1M0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.00</td>\n",
       "      <td>313.0</td>\n",
       "      <td>4.13</td>\n",
       "      <td>0.215</td>\n",
       "      <td>76.81</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000525476</td>\n",
       "      <td>LP6005690-DNA_D01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>Tumor_5_EGAF00000525478</td>\n",
       "      <td>5</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Gloucestershire Royal Hospital, Gloucester</td>\n",
       "      <td>M</td>\n",
       "      <td>64</td>\n",
       "      <td>T1N0M0</td>\n",
       "      <td>IA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>92.11</td>\n",
       "      <td>305.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.237</td>\n",
       "      <td>59.99</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000525478</td>\n",
       "      <td>LP6005690-DNA_A02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>Tumor_6_EGAF00000560171</td>\n",
       "      <td>6</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Gloucestershire Royal Hospital, Gloucester</td>\n",
       "      <td>M</td>\n",
       "      <td>73</td>\n",
       "      <td>T3N1Mx</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.19</td>\n",
       "      <td>309.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.265</td>\n",
       "      <td>66.88</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2</td>\n",
       "      <td>EGAF00000560171</td>\n",
       "      <td>LP6005500-DNA_C01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Tumor_7_EGAF00000501977</td>\n",
       "      <td>7</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>75</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>94.94</td>\n",
       "      <td>313.0</td>\n",
       "      <td>4.38</td>\n",
       "      <td>0.225</td>\n",
       "      <td>76.14</td>\n",
       "      <td>0.978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000501977</td>\n",
       "      <td>LP6007594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>Tumor_8_EGAF00000525467</td>\n",
       "      <td>8</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Salford Royal Infirmary, Manchester</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>92.75</td>\n",
       "      <td>307.0</td>\n",
       "      <td>5.10</td>\n",
       "      <td>0.231</td>\n",
       "      <td>81.73</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9</td>\n",
       "      <td>EGAF00000525467</td>\n",
       "      <td>LP6005690-DNA_G01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>Tumor_9_EGAF00000525188</td>\n",
       "      <td>9</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T1N0M0</td>\n",
       "      <td>IA</td>\n",
       "      <td>Blood</td>\n",
       "      <td>5.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.67</td>\n",
       "      <td>306.0</td>\n",
       "      <td>4.72</td>\n",
       "      <td>0.226</td>\n",
       "      <td>66.52</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "      <td>EGAF00000525188</td>\n",
       "      <td>LP6007404-DNA_A01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample_name Patient Sample_type  \\\n",
       "70  Tumor_10_EGAF00000530893      10         EAC   \n",
       "71  Tumor_11_EGAF00000586547      11         EAC   \n",
       "72  Tumor_12_EGAF00000525525      12         EAC   \n",
       "73  Tumor_13_EGAF00000525466      13         EAC   \n",
       "74  Tumor_14_EGAF00000560172      14         EAC   \n",
       "77  Tumor_15_EGAF00000525475      15         EAC   \n",
       "80  Tumor_16_EGAF00000525474      16         EAC   \n",
       "81  Tumor_17_EGAF00000525479      17         EAC   \n",
       "82  Tumor_18_EGAF00000560173      18         EAC   \n",
       "83  Tumor_19_EGAF00000530869      19         EAC   \n",
       "84   Tumor_1_EGAF00000501976       1         EAC   \n",
       "85  Tumor_20_EGAF00000560131      20         EAC   \n",
       "86  Tumor_21_EGAF00000560132      21         EAC   \n",
       "87  Tumor_22_EGAF00000525477      22         EAC   \n",
       "88  Tumor_23_EGAF00000501978      23         EAC   \n",
       "89   Tumor_2_EGAF00000525468       2         EAC   \n",
       "90   Tumor_3_EGAF00000560133       3         EAC   \n",
       "91   Tumor_4_EGAF00000525476       4         EAC   \n",
       "93   Tumor_5_EGAF00000525478       5         EAC   \n",
       "94   Tumor_6_EGAF00000560171       6         EAC   \n",
       "95   Tumor_7_EGAF00000501977       7         EAC   \n",
       "96   Tumor_8_EGAF00000525467       8         EAC   \n",
       "99   Tumor_9_EGAF00000525188       9         EAC   \n",
       "\n",
       "                                        Centre Gender  \\\n",
       "70            Addenbrookes Hospital, Cambridge      F   \n",
       "71            Addenbrookes Hospital, Cambridge      M   \n",
       "72            Addenbrookes Hospital, Cambridge      M   \n",
       "73     Royal Surrey County Hospital, Guildford      F   \n",
       "74                  St Thomas Hospital, London      M   \n",
       "77     Royal Surrey County Hospital, Guildford      M   \n",
       "80            Addenbrookes Hospital, Cambridge      M   \n",
       "81     Royal Surrey County Hospital, Guildford      M   \n",
       "82                  St Thomas Hospital, London      M   \n",
       "83            Addenbrookes Hospital, Cambridge      M   \n",
       "84            Addenbrookes Hospital, Cambridge      M   \n",
       "85        Edinburgh Royal Infirmary, Edinburgh      F   \n",
       "86     Royal Surrey County Hospital, Guildford      M   \n",
       "87     Royal Surrey County Hospital, Guildford      M   \n",
       "88            Addenbrookes Hospital, Cambridge      M   \n",
       "89                  St Thomas Hospital, London      F   \n",
       "90            Addenbrookes Hospital, Cambridge      M   \n",
       "91     Royal Surrey County Hospital, Guildford      M   \n",
       "93  Gloucestershire Royal Hospital, Gloucester      M   \n",
       "94  Gloucestershire Royal Hospital, Gloucester      M   \n",
       "95            Addenbrookes Hospital, Cambridge      M   \n",
       "96         Salford Royal Infirmary, Manchester      M   \n",
       "99            Addenbrookes Hospital, Cambridge      M   \n",
       "\n",
       "    Age_at_time_of_diagnosis    TNM7 Stage   Germline tissue  \\\n",
       "70                        79  T2N0M0    IB             Blood   \n",
       "71                        67  T4N2M0  IIIC  Normal esophagus   \n",
       "72                        65  T2N1M0   IIB             Blood   \n",
       "73                        61  T2N0M0    IB             Blood   \n",
       "74                        68  T3N1M0  IIIA             Blood   \n",
       "77                        75  T3N3M1    IV             Blood   \n",
       "80                        65  T3N1M0  IIIA             Blood   \n",
       "81                        82  T3N1M0  IIIA             Blood   \n",
       "82                        61  T3N1M0  IIIA             Blood   \n",
       "83                        75  T3N0M0   IIA             Blood   \n",
       "84                        62  T2N0M0    IB             Blood   \n",
       "85                        79  T3N1M0  IIIA             Blood   \n",
       "86                        61  T1N0M0    IA             Blood   \n",
       "87                        50  T3N3M0  IIIC             Blood   \n",
       "88                        64  TxNxMx   NaN             Blood   \n",
       "89                        55  T2N1M0   IIB             Blood   \n",
       "90                        61  T3N2M1    IV             Blood   \n",
       "91                        58  T3N1M0  IIIA             Blood   \n",
       "93                        64  T1N0M0    IA             Blood   \n",
       "94                        73  T3N1Mx   NaN             Blood   \n",
       "95                        75  T3N0M0   IIA             Blood   \n",
       "96                        65  T2N0M0    IB             Blood   \n",
       "99                        67  T1N0M0    IA             Blood   \n",
       "\n",
       "    BE maximal length (cm)  ...  % aligned pairs median insert size  % mapQ=0  \\\n",
       "70                     4.0  ...            93.72              315.0      4.57   \n",
       "71                     3.0  ...            93.70              306.0      5.03   \n",
       "72                     NaN  ...            93.33              299.0      4.91   \n",
       "73                    15.0  ...            93.49              310.0      4.53   \n",
       "74                     6.0  ...            93.87              314.0      4.63   \n",
       "77                     9.0  ...            90.40              308.0      6.36   \n",
       "80                     NaN  ...            92.44              308.0      5.12   \n",
       "81                     6.0  ...            92.48              310.0      5.05   \n",
       "82                    10.0  ...            93.91              312.0      5.46   \n",
       "83                     NaN  ...            94.76              430.0      4.28   \n",
       "84                     NaN  ...            95.13              297.0      4.27   \n",
       "85                     NaN  ...            94.00              308.0      4.67   \n",
       "86                     5.0  ...            93.79              308.0      4.80   \n",
       "87                     3.0  ...            92.89              308.0      4.48   \n",
       "88                     2.0  ...            94.48              305.0      4.18   \n",
       "89                     6.0  ...            92.98              303.0      5.17   \n",
       "90                     NaN  ...            92.92              312.0      5.49   \n",
       "91                     3.0  ...            93.00              313.0      4.13   \n",
       "93                     NaN  ...            92.11              305.0      4.77   \n",
       "94                     NaN  ...            93.19              309.0      5.30   \n",
       "95                     NaN  ...            94.94              313.0      4.38   \n",
       "96                     5.0  ...            92.75              307.0      5.10   \n",
       "99                     5.0  ...            93.67              306.0      4.72   \n",
       "\n",
       "    error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "70           0.229                   58.65            0.975              NaN   \n",
       "71           0.224                   58.86            0.972              NaN   \n",
       "72           0.235                   64.32            0.976              NaN   \n",
       "73           0.217                   82.11            0.978              NaN   \n",
       "74           0.254                   63.20            0.976              NaN   \n",
       "77           0.240                   83.40            0.973              NaN   \n",
       "80           0.233                   65.23            0.975              NaN   \n",
       "81           0.239                   62.44            0.975              NaN   \n",
       "82           0.706                   76.41            0.978              NaN   \n",
       "83           0.232                   80.45            0.979              NaN   \n",
       "84           0.225                   75.79            0.978              NaN   \n",
       "85           0.231                   77.76            0.978              NaN   \n",
       "86           0.251                   64.93            0.976              NaN   \n",
       "87           0.222                   63.04            0.975              NaN   \n",
       "88           0.216                   82.51            0.978              NaN   \n",
       "89           0.220                   89.29            0.977              NaN   \n",
       "90           0.288                   60.89            0.975              NaN   \n",
       "91           0.215                   76.81            0.978              NaN   \n",
       "93           0.237                   59.99            0.975              NaN   \n",
       "94           0.265                   66.88            0.976              NaN   \n",
       "95           0.225                   76.14            0.978              NaN   \n",
       "96           0.231                   81.73            0.976              NaN   \n",
       "99           0.226                   66.52            0.976              NaN   \n",
       "\n",
       "    Sample purity           EGA_ID      EGAIdentifier  \n",
       "70            0.5  EGAF00000530893  LP6007520-DNA_A01  \n",
       "71            0.7  EGAF00000586547  LP6005334-DNA_A03  \n",
       "72            0.4  EGAF00000525525  LP6007409-DNA_A01  \n",
       "73            0.5  EGAF00000525466  LP6005690-DNA_F01  \n",
       "74            0.9  EGAF00000560172  LP6005500-DNA_E01  \n",
       "77            0.7  EGAF00000525475  LP6005690-DNA_C01  \n",
       "80            0.8  EGAF00000525474  LP6005690-DNA_F03  \n",
       "81            0.8  EGAF00000525479  LP6005690-DNA_B02  \n",
       "82            0.4  EGAF00000560173  LP6005500-DNA_F01  \n",
       "83            0.4  EGAF00000530869  LP2000104-DNA_A01  \n",
       "84            0.6  EGAF00000501976          LP6007591  \n",
       "85            0.5  EGAF00000560131  LP6005500-DNA_A01  \n",
       "86            0.6  EGAF00000560132  LP6005500-DNA_B01  \n",
       "87            0.7  EGAF00000525477  LP6005690-DNA_E01  \n",
       "88            0.5  EGAF00000501978          LP6007597  \n",
       "89            0.6  EGAF00000525468  LP6005690-DNA_H01  \n",
       "90            0.3  EGAF00000560133  LP6005500-DNA_D01  \n",
       "91            0.7  EGAF00000525476  LP6005690-DNA_D01  \n",
       "93            0.7  EGAF00000525478  LP6005690-DNA_A02  \n",
       "94            0.2  EGAF00000560171  LP6005500-DNA_C01  \n",
       "95            0.5  EGAF00000501977          LP6007594  \n",
       "96            0.9  EGAF00000525467  LP6005690-DNA_G01  \n",
       "99            0.6  EGAF00000525188  LP6007404-DNA_A01  \n",
       "\n",
       "[23 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>...</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "      <th>EGA_ID</th>\n",
       "      <th>EGAIdentifier</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "      <td>EGAF00000530893</td>\n",
       "      <td>LP6007520-DNA_A01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>EGAF00000586547</td>\n",
       "      <td>LP6005334-DNA_A03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "      <td>EGAF00000525525</td>\n",
       "      <td>LP6007409-DNA_A01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  ...  % aligned pairs  \\\n",
       "0    IB             Blood                     4.0  ...            93.72   \n",
       "1  IIIC  Normal esophagus                     3.0  ...            93.70   \n",
       "2   IIB             Blood                     NaN  ...            93.33   \n",
       "\n",
       "  median insert size  % mapQ=0  error rate HiQ  genome coverage mapQ>3  \\\n",
       "0              315.0      4.57           0.229                   58.65   \n",
       "1              306.0      5.03           0.224                   58.86   \n",
       "2              299.0      4.91           0.235                   64.32   \n",
       "\n",
       "   Callable genome  totalNumberSnps  Sample purity           EGA_ID  \\\n",
       "0            0.975              NaN            0.5  EGAF00000530893   \n",
       "1            0.972              NaN            0.7  EGAF00000586547   \n",
       "2            0.976              NaN            0.4  EGAF00000525525   \n",
       "\n",
       "       EGAIdentifier  \n",
       "0  LP6007520-DNA_A01  \n",
       "1  LP6005334-DNA_A03  \n",
       "2  LP6007409-DNA_A01  \n",
       "\n",
       "[3 rows x 22 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_ginny_species_WOL_scrubbed  =  23\n",
      "<artifact: FeatureTable[Frequency] uuid: 69577db9-21bc-4666-945f-ff410ed88bba>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:127: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL - scrubbed\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired_scrubbed/feature-table.biom',\n",
    "                         'EAC_Ross_ginny_species_WOL_scrubbed',\n",
    "                        Ginny_list=True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c69b83f-d2d7-4579-9693-d6bd239f7d98",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "###### Not in paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "637f66a1-644b-454f-9edf-05726fb49beb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_paired_species_WOL_scrubbed  =  30\n",
      "<artifact: FeatureTable[Frequency] uuid: 47764dbc-610f-41f9-87ff-09493f70a123>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:103: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL - scrubbed\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired_scrubbed/feature-table.biom',\n",
    "                         'EAC_Ross_paired_species_WOL_scrubbed')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "adb1ea91-0f36-4fe6-b170-09a07027ae72",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_paired_genome_WOL_scrubbed  =  30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_paired_species_WOL  =  30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_paired_genome_WOL  =  30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.7</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                       79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                       67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                       65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue BE maximal length (cm) Tumor length (mm)  \\\n",
       "0    IB             Blood                    4.0              15.0   \n",
       "1  IIIC  Normal esophagus                    3.0              70.0   \n",
       "2   IIB             Blood                    NaN              29.0   \n",
       "\n",
       "  purity-filtered pairs % aligned pairs median insert size % mapQ=0  \\\n",
       "0           950,588,085           93.72              315.0     4.57   \n",
       "1           995,657,497            93.7              306.0     5.03   \n",
       "2         1,068,844,258           93.33              299.0     4.91   \n",
       "\n",
       "  error rate HiQ genome coverage mapQ>3 Callable genome totalNumberSnps  \\\n",
       "0          0.229                  58.65           0.975             NaN   \n",
       "1          0.224                  58.86           0.972             NaN   \n",
       "2          0.235                  64.32           0.976             NaN   \n",
       "\n",
       "  Sample purity  \n",
       "0           0.5  \n",
       "1           0.7  \n",
       "2           0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_exact_Ross_paired_species_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_zebra_Ross_paired_genome_WOL  =  30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.7</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                       79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                       67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                       65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue BE maximal length (cm) Tumor length (mm)  \\\n",
       "0    IB             Blood                    4.0              15.0   \n",
       "1  IIIC  Normal esophagus                    3.0              70.0   \n",
       "2   IIB             Blood                    NaN              29.0   \n",
       "\n",
       "  purity-filtered pairs % aligned pairs median insert size % mapQ=0  \\\n",
       "0           950,588,085           93.72              315.0     4.57   \n",
       "1           995,657,497            93.7              306.0     5.03   \n",
       "2         1,068,844,258           93.33              299.0     4.91   \n",
       "\n",
       "  error rate HiQ genome coverage mapQ>3 Callable genome totalNumberSnps  \\\n",
       "0          0.229                  58.65           0.975             NaN   \n",
       "1          0.224                  58.86           0.972             NaN   \n",
       "2          0.235                  64.32           0.976             NaN   \n",
       "\n",
       "  Sample purity  \n",
       "0           0.5  \n",
       "1           0.7  \n",
       "2           0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_zebra_exact_Ross_paired_genome_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_paired_species_rep200  =  30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.7</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                       79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                       67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                       65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue BE maximal length (cm) Tumor length (mm)  \\\n",
       "0    IB             Blood                    4.0              15.0   \n",
       "1  IIIC  Normal esophagus                    3.0              70.0   \n",
       "2   IIB             Blood                    NaN              29.0   \n",
       "\n",
       "  purity-filtered pairs % aligned pairs median insert size % mapQ=0  \\\n",
       "0           950,588,085           93.72              315.0     4.57   \n",
       "1           995,657,497            93.7              306.0     5.03   \n",
       "2         1,068,844,258           93.33              299.0     4.91   \n",
       "\n",
       "  error rate HiQ genome coverage mapQ>3 Callable genome totalNumberSnps  \\\n",
       "0          0.229                  58.65           0.975             NaN   \n",
       "1          0.224                  58.86           0.972             NaN   \n",
       "2          0.235                  64.32           0.976             NaN   \n",
       "\n",
       "  Sample purity  \n",
       "0           0.5  \n",
       "1           0.7  \n",
       "2           0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_exact_Ross_paired_species_rep200  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#WOL - scrubbed\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired_scrubbed/feature-table.biom',\n",
    "                         'EAC_Ross_paired_genome_WOL_scrubbed')\n",
    "\n",
    "#WOL\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'EAC_Ross_paired_species_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'EAC_Ross_paired_genome_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'EAC_exact_Ross_paired_species_WOL',\n",
    "                         exact_pairs=True)\n",
    "\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'EAC_zebra_Ross_paired_genome_WOL',\n",
    "                        zebra = 'Ross_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'EAC_zebra_exact_Ross_paired_genome_WOL',\n",
    "                         exact_pairs=True,\n",
    "                        zebra = 'Ross_WOL')\n",
    "\n",
    "#Rep200\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_rep200_R1_R2_paired.biom',\n",
    "                         'EAC_Ross_paired_species_rep200')\n",
    "\n",
    "normal_be_eac_data_split(['EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_rep200_R1_R2_paired.biom',\n",
    "                         'EAC_exact_Ross_paired_species_rep200',\n",
    "                         exact_pairs = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a4f79443-a5d0-4304-b20a-b1ed2d80d424",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor_10_EGAF00000530893</td>\n",
       "      <td>10</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>950,588,085</td>\n",
       "      <td>93.72</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.57</td>\n",
       "      <td>0.229</td>\n",
       "      <td>58.65</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Tumor_11_EGAF00000586547</td>\n",
       "      <td>11</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>995,657,497</td>\n",
       "      <td>93.70</td>\n",
       "      <td>306.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.224</td>\n",
       "      <td>58.86</td>\n",
       "      <td>0.972</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Tumor_12_EGAF00000525525</td>\n",
       "      <td>12</td>\n",
       "      <td>EAC</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>1,068,844,258</td>\n",
       "      <td>93.33</td>\n",
       "      <td>299.0</td>\n",
       "      <td>4.91</td>\n",
       "      <td>0.235</td>\n",
       "      <td>64.32</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample_name Patient Sample_type  \\\n",
       "0  Tumor_10_EGAF00000530893      10         EAC   \n",
       "1  Tumor_11_EGAF00000586547      11         EAC   \n",
       "2  Tumor_12_EGAF00000525525      12         EAC   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           950,588,085            93.72               315.0      4.57   \n",
       "1           995,657,497            93.70               306.0      5.03   \n",
       "2         1,068,844,258            93.33               299.0      4.91   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.229                   58.65            0.975              NaN   \n",
       "1           0.224                   58.86            0.972              NaN   \n",
       "2           0.235                   64.32            0.976              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.5  \n",
       "1            0.7  \n",
       "2            0.4  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_Ross_combo_species_WOL  =  30\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split('EAC',\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_combo_R1_R2_paired_unpaired.biom',\n",
    "                         'EAC_Ross_combo_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c22a144-c7d5-4f8d-9739-de7b229fe583",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### (Not used in paper) Normal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c578f6fe-b9d1-4762-8bde-831c1f79985c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Normal_10_EGAF00000530908</td>\n",
       "      <td>10</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>807,792,398</td>\n",
       "      <td>92.49</td>\n",
       "      <td>296.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.245</td>\n",
       "      <td>48.64</td>\n",
       "      <td>0.968</td>\n",
       "      <td>3523708.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Normal_11_EGAF00000586533</td>\n",
       "      <td>11</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>639,793,060</td>\n",
       "      <td>92.22</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.88</td>\n",
       "      <td>0.241</td>\n",
       "      <td>37.77</td>\n",
       "      <td>0.969</td>\n",
       "      <td>3541117.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Normal_12_EGAF00000525507</td>\n",
       "      <td>12</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>559,984,669</td>\n",
       "      <td>88.64</td>\n",
       "      <td>299.0</td>\n",
       "      <td>5.09</td>\n",
       "      <td>0.257</td>\n",
       "      <td>32.15</td>\n",
       "      <td>0.964</td>\n",
       "      <td>3491591.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample_name Patient Sample_type  \\\n",
       "0  Normal_10_EGAF00000530908      10      Normal   \n",
       "1  Normal_11_EGAF00000586533      11      Normal   \n",
       "2  Normal_12_EGAF00000525507      12      Normal   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           807,792,398            92.49               296.0      5.03   \n",
       "1           639,793,060            92.22               315.0      4.88   \n",
       "2           559,984,669            88.64               299.0      5.09   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   48.64            0.968        3523708.0   \n",
       "1           0.241                   37.77            0.969        3541117.0   \n",
       "2           0.257                   32.15            0.964        3491591.0   \n",
       "\n",
       "   Sample purity  \n",
       "0            NaN  \n",
       "1            NaN  \n",
       "2            NaN  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in Norm_Ross_combo_species_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split('Normal',\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_combo_R1_R2_paired_unpaired.biom',\n",
    "                         'Norm_Ross_combo_species_WOL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "594e7bef-0d76-4f1c-a598-f8288733cbde",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Normal_10_EGAF00000530908</td>\n",
       "      <td>10</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>807,792,398</td>\n",
       "      <td>92.49</td>\n",
       "      <td>296.0</td>\n",
       "      <td>5.03</td>\n",
       "      <td>0.245</td>\n",
       "      <td>48.64</td>\n",
       "      <td>0.968</td>\n",
       "      <td>3523708.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Normal_11_EGAF00000586533</td>\n",
       "      <td>11</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>639,793,060</td>\n",
       "      <td>92.22</td>\n",
       "      <td>315.0</td>\n",
       "      <td>4.88</td>\n",
       "      <td>0.241</td>\n",
       "      <td>37.77</td>\n",
       "      <td>0.969</td>\n",
       "      <td>3541117.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Normal_12_EGAF00000525507</td>\n",
       "      <td>12</td>\n",
       "      <td>Normal</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>559,984,669</td>\n",
       "      <td>88.64</td>\n",
       "      <td>299.0</td>\n",
       "      <td>5.09</td>\n",
       "      <td>0.257</td>\n",
       "      <td>32.15</td>\n",
       "      <td>0.964</td>\n",
       "      <td>3491591.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample_name Patient Sample_type  \\\n",
       "0  Normal_10_EGAF00000530908      10      Normal   \n",
       "1  Normal_11_EGAF00000586533      11      Normal   \n",
       "2  Normal_12_EGAF00000525507      12      Normal   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           807,792,398            92.49               296.0      5.03   \n",
       "1           639,793,060            92.22               315.0      4.88   \n",
       "2           559,984,669            88.64               299.0      5.09   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   48.64            0.968        3523708.0   \n",
       "1           0.241                   37.77            0.969        3541117.0   \n",
       "2           0.257                   32.15            0.964        3491591.0   \n",
       "\n",
       "   Sample purity  \n",
       "0            NaN  \n",
       "1            NaN  \n",
       "2            NaN  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in Norm_Ross_paired_species_WOL  =  23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Normal'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'Norm_Ross_paired_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92f1b507-aa33-4f7b-95bc-919637cfc18e",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### (Not used in paper) Normal + BE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "69b549fb-dad1-4ffb-be59-e72314be1416",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in Norm_BE_Ross_paired_species_WOL  =  58\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Spring23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:32: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Normal', 'Barretts'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'Norm_BE_Ross_paired_species_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e347629-9e73-4147-a68f-fb019dda798c",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### (Not used in paper) BE + EAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c0f7dd8e-1592-4f3b-bbe9-008a23c0f65b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_EAC_Ross_paired_species_WOL  =  65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:54: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.30</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.50</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.20</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender  Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                        79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                        67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                        65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue  BE maximal length (cm)  Tumor length (mm)  \\\n",
       "0    IB             Blood                     4.0               15.0   \n",
       "1  IIIC  Normal esophagus                     3.0               70.0   \n",
       "2   IIB             Blood                     NaN               29.0   \n",
       "\n",
       "  purity-filtered pairs  % aligned pairs  median insert size  % mapQ=0  \\\n",
       "0           935,820,463            92.57               339.0      5.30   \n",
       "1         1,135,954,265            93.50               300.0      4.77   \n",
       "2           955,544,208            93.01               310.0      5.04   \n",
       "\n",
       "   error rate HiQ  genome coverage mapQ>3  Callable genome  totalNumberSnps  \\\n",
       "0           0.245                   55.76            0.975              NaN   \n",
       "1           0.231                   67.45            0.976              NaN   \n",
       "2           0.243                   57.20            0.975              NaN   \n",
       "\n",
       "   Sample purity  \n",
       "0            0.8  \n",
       "1            0.6  \n",
       "2            0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_EAC_Ross_paired_genome_WOL  =  65\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:54: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'BE_EAC_Ross_paired_species_WOL')\n",
    "\n",
    "normal_be_eac_data_split(['Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'BE_EAC_Ross_paired_genome_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dbf437a0-93e1-4427-a5ab-5cda1ed8d659",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### (Not used in paper) Exact Pairs of BE/EAC\n",
    "Note: Still unclear why some have multiple BE samples in the paper, so chose random BE for each patient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e30c85dd-9814-4d81-a9c4-d6d2593be37b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.3</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.5</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.2</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                       79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                       67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                       65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue BE maximal length (cm) Tumor length (mm)  \\\n",
       "0    IB             Blood                    4.0              15.0   \n",
       "1  IIIC  Normal esophagus                    3.0              70.0   \n",
       "2   IIB             Blood                    NaN              29.0   \n",
       "\n",
       "  purity-filtered pairs % aligned pairs median insert size % mapQ=0  \\\n",
       "0           935,820,463           92.57              339.0      5.3   \n",
       "1         1,135,954,265            93.5              300.0     4.77   \n",
       "2           955,544,208           93.01              310.0     5.04   \n",
       "\n",
       "  error rate HiQ genome coverage mapQ>3 Callable genome totalNumberSnps  \\\n",
       "0          0.245                  55.76           0.975             NaN   \n",
       "1          0.231                  67.45           0.976             NaN   \n",
       "2          0.243                   57.2           0.975             NaN   \n",
       "\n",
       "  Sample purity  \n",
       "0           0.8  \n",
       "1           0.6  \n",
       "2           0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_EAC_exact_Ross_paired_species_WOL  =  46\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:54: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>Patient</th>\n",
       "      <th>Sample_type</th>\n",
       "      <th>Centre</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age_at_time_of_diagnosis</th>\n",
       "      <th>TNM7</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Germline tissue</th>\n",
       "      <th>BE maximal length (cm)</th>\n",
       "      <th>Tumor length (mm)</th>\n",
       "      <th>purity-filtered pairs</th>\n",
       "      <th>% aligned pairs</th>\n",
       "      <th>median insert size</th>\n",
       "      <th>% mapQ=0</th>\n",
       "      <th>error rate HiQ</th>\n",
       "      <th>genome coverage mapQ&gt;3</th>\n",
       "      <th>Callable genome</th>\n",
       "      <th>totalNumberSnps</th>\n",
       "      <th>Sample purity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Barretts_10_EGAF00000530875</td>\n",
       "      <td>10</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>F</td>\n",
       "      <td>79</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>IB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>4.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>935,820,463</td>\n",
       "      <td>92.57</td>\n",
       "      <td>339.0</td>\n",
       "      <td>5.3</td>\n",
       "      <td>0.245</td>\n",
       "      <td>55.76</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Barretts_11_EGAF00000586502</td>\n",
       "      <td>11</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>67</td>\n",
       "      <td>T4N2M0</td>\n",
       "      <td>IIIC</td>\n",
       "      <td>Normal esophagus</td>\n",
       "      <td>3.0</td>\n",
       "      <td>70.0</td>\n",
       "      <td>1,135,954,265</td>\n",
       "      <td>93.5</td>\n",
       "      <td>300.0</td>\n",
       "      <td>4.77</td>\n",
       "      <td>0.231</td>\n",
       "      <td>67.45</td>\n",
       "      <td>0.976</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Barretts_12_EGAF00000525202</td>\n",
       "      <td>12</td>\n",
       "      <td>Barretts</td>\n",
       "      <td>Addenbrookes Hospital, Cambridge</td>\n",
       "      <td>M</td>\n",
       "      <td>65</td>\n",
       "      <td>T2N1M0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>Blood</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.0</td>\n",
       "      <td>955,544,208</td>\n",
       "      <td>93.01</td>\n",
       "      <td>310.0</td>\n",
       "      <td>5.04</td>\n",
       "      <td>0.243</td>\n",
       "      <td>57.2</td>\n",
       "      <td>0.975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_name Patient Sample_type  \\\n",
       "0  Barretts_10_EGAF00000530875      10    Barretts   \n",
       "1  Barretts_11_EGAF00000586502      11    Barretts   \n",
       "2  Barretts_12_EGAF00000525202      12    Barretts   \n",
       "\n",
       "                             Centre Gender Age_at_time_of_diagnosis    TNM7  \\\n",
       "0  Addenbrookes Hospital, Cambridge      F                       79  T2N0M0   \n",
       "1  Addenbrookes Hospital, Cambridge      M                       67  T4N2M0   \n",
       "2  Addenbrookes Hospital, Cambridge      M                       65  T2N1M0   \n",
       "\n",
       "  Stage   Germline tissue BE maximal length (cm) Tumor length (mm)  \\\n",
       "0    IB             Blood                    4.0              15.0   \n",
       "1  IIIC  Normal esophagus                    3.0              70.0   \n",
       "2   IIB             Blood                    NaN              29.0   \n",
       "\n",
       "  purity-filtered pairs % aligned pairs median insert size % mapQ=0  \\\n",
       "0           935,820,463           92.57              339.0      5.3   \n",
       "1         1,135,954,265            93.5              300.0     4.77   \n",
       "2           955,544,208           93.01              310.0     5.04   \n",
       "\n",
       "  error rate HiQ genome coverage mapQ>3 Callable genome totalNumberSnps  \\\n",
       "0          0.245                  55.76           0.975             NaN   \n",
       "1          0.231                  67.45           0.976             NaN   \n",
       "2          0.243                   57.2           0.975             NaN   \n",
       "\n",
       "  Sample purity  \n",
       "0           0.8  \n",
       "1           0.6  \n",
       "2           0.3  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_EAC_exact_Ross_paired_genome_WOL  =  46\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:54: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_be_eac_data_split(['Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/species_WOL_R1_R2_paired.biom',\n",
    "                         'BE_EAC_exact_Ross_paired_species_WOL',\n",
    "                         exact_pairs=True)\n",
    "\n",
    "normal_be_eac_data_split(['Barretts', 'EAC'],\n",
    "                         os.getcwd() + '/qiita_downloads/qiitaTBD_Norm_BE_EAC/genome_WOL_R1_R2_paired.biom',\n",
    "                         'BE_EAC_exact_Ross_paired_genome_WOL',\n",
    "                        exact_pairs=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e6d5b2c-3936-4d51-94c3-c60ce4ab38ce",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### (Supp?) From Sam: Barrett's Esophgous Progressors/Non Progressors: using MetaPhlan4\n",
    "Paulson, T. G. et al. Somatic whole genome dynamics of precancer in Barrettâ€™s esophagus reveals features associated with disease progression. Nature Communications vol. 13 (2022) https://doi.org/10.1038/s41467-022-29767-7"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90118ef0-c827-48f1-b4a0-8467e99759df",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a8a9880a-bf07-43ec-8caa-19ec988fa8ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               87.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_species_MP  =  202\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               87.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_species_MP_counts  =  202\n"
     ]
    }
   ],
   "source": [
    "#Relative Abudnace\n",
    "Sam_BE_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['NP', 'Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "72fe3d28-81a2-474b-880e-400e9aba02a9",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "58ff07c9-4c06-416e-9afc-979fcc299299",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               87.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_species_MP  =  111\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name               race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01  White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02  White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1064.BE.03  White_nonHispanic      M         17         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  17-24987N-105W-BLD                4             23567   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               87.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_species_MP_counts  =  111\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['NP',], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_NP_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['NP'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_NP_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dfeb2eb4-eaf7-4448-8175-a994626ff617",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "247770e3-12fe-4889-9ebd-2f7f4232a498",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               53.23           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_species_MP  =  53\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.01</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23561</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.02</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>7</td>\n",
       "      <td>23564</td>\n",
       "      <td>NP</td>\n",
       "      <td>78.83</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.01</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>3</td>\n",
       "      <td>23611</td>\n",
       "      <td>NP</td>\n",
       "      <td>53.23</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.01      White_nonHispanic      M         17         T1   \n",
       "1  14598.PCGA.1064.BE.02      White_nonHispanic      M         17         T1   \n",
       "2  14598.PCGA.1067.BE.01  Asian_PacificIslander      M         55         T1   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                3             23561   \n",
       "1            1  17-24987N-105W-BLD                7             23564   \n",
       "2            1  55-25056N-204U-BLD                3             23611   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               78.83           14598  \n",
       "1                NP               78.83           14598  \n",
       "2                NP               53.23           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T1_species_MP_counts  =  53\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['NP',], ['T1'], \n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_NP_T1_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['NP'], ['T1'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_NP_T1_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cc98a2e-80d4-4c39-82ff-03fc036daa8f",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Non Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d02eac75-3ef0-409d-90ae-df96522e04e5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               87.07           14598  \n",
       "1                NP               87.07           14598  \n",
       "2                NP               57.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_species_MP  =  58\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1064.BE.03</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>4</td>\n",
       "      <td>23567</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1064.BE.04</td>\n",
       "      <td>White_nonHispanic</td>\n",
       "      <td>M</td>\n",
       "      <td>17</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>17-24987N-105W-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>24933</td>\n",
       "      <td>NP</td>\n",
       "      <td>87.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1067.BE.02</td>\n",
       "      <td>Asian_PacificIslander</td>\n",
       "      <td>M</td>\n",
       "      <td>55</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>55-25056N-204U-BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>23614</td>\n",
       "      <td>NP</td>\n",
       "      <td>57.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name                   race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1064.BE.03      White_nonHispanic      M         17         T2   \n",
       "1  14598.PCGA.1064.BE.04      White_nonHispanic      M         17         T2   \n",
       "2  14598.PCGA.1067.BE.02  Asian_PacificIslander      M         55         T2   \n",
       "\n",
       "   ismainstudy        pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  17-24987N-105W-BLD                4             23567   \n",
       "1            1  17-24987N-105W-BLD                1             24933   \n",
       "2            1  55-25056N-204U-BLD                1             23614   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0                NP               87.07           14598  \n",
       "1                NP               87.07           14598  \n",
       "2                NP               57.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_NP_T2_species_MP_counts  =  58\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['NP',], ['T2'], \n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_NP_T2_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['NP'], ['T2'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_NP_T2_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae337c1f-f4aa-4686-b3fc-ecab1010e916",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1356a3c6-ad57-4886-8295-215590e0046b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               56.25           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_species_MP  =  91\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  42-24954N-53L-GST                5             24289   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               56.25           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_species_MP_counts  =  91\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['Prog',], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_Prog_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['Prog'], ['T1', 'T2'], #, 'LongFollowUp'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_Prog_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a53bda85-177b-4514-ac06-5a8fa06927bd",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c131cff3-8586-4f4c-a4f8-7c48d5fc7c65",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               56.25           14598  \n",
       "1              Prog               56.25           14598  \n",
       "2              Prog               64.11           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_species_MP  =  40\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.03</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24289</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.04</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>3</td>\n",
       "      <td>24292</td>\n",
       "      <td>Prog</td>\n",
       "      <td>56.25</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T1</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>23585</td>\n",
       "      <td>Prog</td>\n",
       "      <td>64.11</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.03  Other_Unknown      M         42         T1   \n",
       "1  14598.PCGA.1065.BE.04  Other_Unknown      M         42         T1   \n",
       "2  14598.PCGA.1066.BE.01  Other_Unknown      F         43         T1   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24289   \n",
       "1            1  42-24954N-53L-GST                3             24292   \n",
       "2            1  43-24956N-29P-GST                1             23585   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               56.25           14598  \n",
       "1              Prog               56.25           14598  \n",
       "2              Prog               64.11           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T1_species_MP_counts  =  40\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['Prog',], ['T1'], \n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_Prog_T1_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['Prog'], ['T1'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_Prog_T1_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "063ea34d-285c-45a4-8816-6f24039051fa",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "##### Barrett's Esophgous Progressors T2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5f3932df-afb6-4fa1-af69-ddcd164ec99e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               65.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_species_MP  =  51\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>gender</th>\n",
       "      <th>patientid</th>\n",
       "      <th>sampletype</th>\n",
       "      <th>ismainstudy</th>\n",
       "      <th>pairednormal</th>\n",
       "      <th>distancefromgej</th>\n",
       "      <th>orginal_sampleid</th>\n",
       "      <th>progressionstatus</th>\n",
       "      <th>patientageatsample</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14598.PCGA.1065.BE.01</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>5</td>\n",
       "      <td>24283</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14598.PCGA.1065.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>42</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>42-24954N-53L-GST</td>\n",
       "      <td>1</td>\n",
       "      <td>24286</td>\n",
       "      <td>Prog</td>\n",
       "      <td>57.65</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14598.PCGA.1066.BE.02</td>\n",
       "      <td>Other_Unknown</td>\n",
       "      <td>F</td>\n",
       "      <td>43</td>\n",
       "      <td>T2</td>\n",
       "      <td>1</td>\n",
       "      <td>43-24956N-29P-GST</td>\n",
       "      <td>4</td>\n",
       "      <td>23593</td>\n",
       "      <td>Prog</td>\n",
       "      <td>65.07</td>\n",
       "      <td>14598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             sample_name           race gender  patientid sampletype  \\\n",
       "0  14598.PCGA.1065.BE.01  Other_Unknown      M         42         T2   \n",
       "1  14598.PCGA.1065.BE.02  Other_Unknown      M         42         T2   \n",
       "2  14598.PCGA.1066.BE.02  Other_Unknown      F         43         T2   \n",
       "\n",
       "   ismainstudy       pairednormal  distancefromgej  orginal_sampleid  \\\n",
       "0            1  42-24954N-53L-GST                5             24283   \n",
       "1            1  42-24954N-53L-GST                1             24286   \n",
       "2            1  43-24956N-29P-GST                4             23593   \n",
       "\n",
       "  progressionstatus  patientageatsample  qiita_study_id  \n",
       "0              Prog               57.65           14598  \n",
       "1              Prog               57.65           14598  \n",
       "2              Prog               65.07           14598  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in BE_Prog_T2_species_MP_counts  =  51\n"
     ]
    }
   ],
   "source": [
    "Sam_BE_data_split(['Prog',], ['T2'], \n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/metaphlan.wide.species.csv',\n",
    "              'BE_Prog_T2_species_MP')\n",
    "\n",
    "#Counts\n",
    "Sam_BE_data_split(['Prog'], ['T2'],\n",
    "              '/Users/cguccion/Dropbox/Storage/RawData/23_06_23_Hutch_BE_Sam_Data/2023-06-23-metaphlan-analysis-bundle/convert_counts/metaphlan.wide.counts.species.csv',\n",
    "              'BE_Prog_T2_species_MP_counts', True)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a495ec7-65d8-49fa-9399-c62c3a733d2c",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Not used in paper: TCGA EAC: BAD\n",
    "BAD: From TCGA, not exact paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8c526fca-583c-4a50-9d13-91a7ec13b53b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>size</th>\n",
       "      <th>gender</th>\n",
       "      <th>case_id</th>\n",
       "      <th>project</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>aliquot_id</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>sample_uuid</th>\n",
       "      <th>aliquot_uuid</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>primary_site</th>\n",
       "      <th>vital_status</th>\n",
       "      <th>investigation</th>\n",
       "      <th>host_subject_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14990.64226f74c281925c8815b407</td>\n",
       "      <td>white</td>\n",
       "      <td>192795648</td>\n",
       "      <td>female</td>\n",
       "      <td>TCGA-L5-A4OG</td>\n",
       "      <td>cguccione/tcga-eac</td>\n",
       "      <td>not reported</td>\n",
       "      <td>TCGA-L5-A4OG-01A-11D-A267-26</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>e8e44c65-b4e4-4688-a217-206905e68a7f</td>\n",
       "      <td>04c91c91-e55a-4075-b08a-f80568bb4a7b</td>\n",
       "      <td>Adenomas and Adenocarcinomas</td>\n",
       "      <td>Esophagus</td>\n",
       "      <td>Alive</td>\n",
       "      <td>TCGA-ESCA</td>\n",
       "      <td>eb4ad376-4cc1-4d2c-9b93-ad2b082c77c4</td>\n",
       "      <td>14990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14990.6422776e8c11cc752762f39b</td>\n",
       "      <td>white</td>\n",
       "      <td>143024713</td>\n",
       "      <td>male</td>\n",
       "      <td>TCGA-L5-A4OI</td>\n",
       "      <td>cguccione/tcga-eac</td>\n",
       "      <td>not reported</td>\n",
       "      <td>TCGA-L5-A4OI-01A-11D-A267-26</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>1b1cbb34-da71-4441-b1a1-c936e8f04c73</td>\n",
       "      <td>8d5a0d62-d9c1-46d1-9b81-685293b45e6b</td>\n",
       "      <td>Adenomas and Adenocarcinomas</td>\n",
       "      <td>Esophagus</td>\n",
       "      <td>Alive</td>\n",
       "      <td>TCGA-ESCA</td>\n",
       "      <td>cb2af13b-cbed-4657-8fd4-fed8db375cba</td>\n",
       "      <td>14990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14990.6422782f8c11cc752762f3a3</td>\n",
       "      <td>not reported</td>\n",
       "      <td>221282761</td>\n",
       "      <td>female</td>\n",
       "      <td>TCGA-L5-A43M</td>\n",
       "      <td>cguccione/tcga-eac</td>\n",
       "      <td>not reported</td>\n",
       "      <td>TCGA-L5-A43M-01A-11D-A248-26</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>aeec137a-e929-439c-b7e7-33943cffb642</td>\n",
       "      <td>97721561-b3ce-4b56-b0d2-b7c70e978526</td>\n",
       "      <td>Adenomas and Adenocarcinomas</td>\n",
       "      <td>Esophagus</td>\n",
       "      <td>Alive</td>\n",
       "      <td>TCGA-ESCA</td>\n",
       "      <td>29da08f4-c9e2-4e7a-9733-b4ec5b9c5f13</td>\n",
       "      <td>14990</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sample_name          race       size  gender  \\\n",
       "0  14990.64226f74c281925c8815b407         white  192795648  female   \n",
       "1  14990.6422776e8c11cc752762f39b         white  143024713    male   \n",
       "2  14990.6422782f8c11cc752762f3a3  not reported  221282761  female   \n",
       "\n",
       "        case_id             project     ethnicity  \\\n",
       "0  TCGA-L5-A4OG  cguccione/tcga-eac  not reported   \n",
       "1  TCGA-L5-A4OI  cguccione/tcga-eac  not reported   \n",
       "2  TCGA-L5-A43M  cguccione/tcga-eac  not reported   \n",
       "\n",
       "                     aliquot_id    sample_type  \\\n",
       "0  TCGA-L5-A4OG-01A-11D-A267-26  Primary Tumor   \n",
       "1  TCGA-L5-A4OI-01A-11D-A267-26  Primary Tumor   \n",
       "2  TCGA-L5-A43M-01A-11D-A248-26  Primary Tumor   \n",
       "\n",
       "                            sample_uuid                          aliquot_uuid  \\\n",
       "0  e8e44c65-b4e4-4688-a217-206905e68a7f  04c91c91-e55a-4075-b08a-f80568bb4a7b   \n",
       "1  1b1cbb34-da71-4441-b1a1-c936e8f04c73  8d5a0d62-d9c1-46d1-9b81-685293b45e6b   \n",
       "2  aeec137a-e929-439c-b7e7-33943cffb642  97721561-b3ce-4b56-b0d2-b7c70e978526   \n",
       "\n",
       "                   disease_type primary_site vital_status investigation  \\\n",
       "0  Adenomas and Adenocarcinomas    Esophagus        Alive     TCGA-ESCA   \n",
       "1  Adenomas and Adenocarcinomas    Esophagus        Alive     TCGA-ESCA   \n",
       "2  Adenomas and Adenocarcinomas    Esophagus        Alive     TCGA-ESCA   \n",
       "\n",
       "                        host_subject_id  qiita_study_id  \n",
       "0  eb4ad376-4cc1-4d2c-9b93-ad2b082c77c4           14990  \n",
       "1  cb2af13b-cbed-4657-8fd4-fed8db375cba           14990  \n",
       "2  29da08f4-c9e2-4e7a-9733-b4ec5b9c5f13           14990  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_TCGA_species_WOL  =  19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:54: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>race</th>\n",
       "      <th>size</th>\n",
       "      <th>gender</th>\n",
       "      <th>case_id</th>\n",
       "      <th>project</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>aliquot_id</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>sample_uuid</th>\n",
       "      <th>aliquot_uuid</th>\n",
       "      <th>disease_type</th>\n",
       "      <th>primary_site</th>\n",
       "      <th>vital_status</th>\n",
       "      <th>investigation</th>\n",
       "      <th>host_subject_id</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14990.64226f74c281925c8815b407</td>\n",
       "      <td>white</td>\n",
       "      <td>192795648</td>\n",
       "      <td>female</td>\n",
       "      <td>TCGA-L5-A4OG</td>\n",
       "      <td>cguccione/tcga-eac</td>\n",
       "      <td>not reported</td>\n",
       "      <td>TCGA-L5-A4OG-01A-11D-A267-26</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>e8e44c65-b4e4-4688-a217-206905e68a7f</td>\n",
       "      <td>04c91c91-e55a-4075-b08a-f80568bb4a7b</td>\n",
       "      <td>Adenomas and Adenocarcinomas</td>\n",
       "      <td>Esophagus</td>\n",
       "      <td>Alive</td>\n",
       "      <td>TCGA-ESCA</td>\n",
       "      <td>eb4ad376-4cc1-4d2c-9b93-ad2b082c77c4</td>\n",
       "      <td>14990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14990.6422776e8c11cc752762f39b</td>\n",
       "      <td>white</td>\n",
       "      <td>143024713</td>\n",
       "      <td>male</td>\n",
       "      <td>TCGA-L5-A4OI</td>\n",
       "      <td>cguccione/tcga-eac</td>\n",
       "      <td>not reported</td>\n",
       "      <td>TCGA-L5-A4OI-01A-11D-A267-26</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>1b1cbb34-da71-4441-b1a1-c936e8f04c73</td>\n",
       "      <td>8d5a0d62-d9c1-46d1-9b81-685293b45e6b</td>\n",
       "      <td>Adenomas and Adenocarcinomas</td>\n",
       "      <td>Esophagus</td>\n",
       "      <td>Alive</td>\n",
       "      <td>TCGA-ESCA</td>\n",
       "      <td>cb2af13b-cbed-4657-8fd4-fed8db375cba</td>\n",
       "      <td>14990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14990.6422782f8c11cc752762f3a3</td>\n",
       "      <td>not reported</td>\n",
       "      <td>221282761</td>\n",
       "      <td>female</td>\n",
       "      <td>TCGA-L5-A43M</td>\n",
       "      <td>cguccione/tcga-eac</td>\n",
       "      <td>not reported</td>\n",
       "      <td>TCGA-L5-A43M-01A-11D-A248-26</td>\n",
       "      <td>Primary Tumor</td>\n",
       "      <td>aeec137a-e929-439c-b7e7-33943cffb642</td>\n",
       "      <td>97721561-b3ce-4b56-b0d2-b7c70e978526</td>\n",
       "      <td>Adenomas and Adenocarcinomas</td>\n",
       "      <td>Esophagus</td>\n",
       "      <td>Alive</td>\n",
       "      <td>TCGA-ESCA</td>\n",
       "      <td>29da08f4-c9e2-4e7a-9733-b4ec5b9c5f13</td>\n",
       "      <td>14990</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sample_name          race       size  gender  \\\n",
       "0  14990.64226f74c281925c8815b407         white  192795648  female   \n",
       "1  14990.6422776e8c11cc752762f39b         white  143024713    male   \n",
       "2  14990.6422782f8c11cc752762f3a3  not reported  221282761  female   \n",
       "\n",
       "        case_id             project     ethnicity  \\\n",
       "0  TCGA-L5-A4OG  cguccione/tcga-eac  not reported   \n",
       "1  TCGA-L5-A4OI  cguccione/tcga-eac  not reported   \n",
       "2  TCGA-L5-A43M  cguccione/tcga-eac  not reported   \n",
       "\n",
       "                     aliquot_id    sample_type  \\\n",
       "0  TCGA-L5-A4OG-01A-11D-A267-26  Primary Tumor   \n",
       "1  TCGA-L5-A4OI-01A-11D-A267-26  Primary Tumor   \n",
       "2  TCGA-L5-A43M-01A-11D-A248-26  Primary Tumor   \n",
       "\n",
       "                            sample_uuid                          aliquot_uuid  \\\n",
       "0  e8e44c65-b4e4-4688-a217-206905e68a7f  04c91c91-e55a-4075-b08a-f80568bb4a7b   \n",
       "1  1b1cbb34-da71-4441-b1a1-c936e8f04c73  8d5a0d62-d9c1-46d1-9b81-685293b45e6b   \n",
       "2  aeec137a-e929-439c-b7e7-33943cffb642  97721561-b3ce-4b56-b0d2-b7c70e978526   \n",
       "\n",
       "                   disease_type primary_site vital_status investigation  \\\n",
       "0  Adenomas and Adenocarcinomas    Esophagus        Alive     TCGA-ESCA   \n",
       "1  Adenomas and Adenocarcinomas    Esophagus        Alive     TCGA-ESCA   \n",
       "2  Adenomas and Adenocarcinomas    Esophagus        Alive     TCGA-ESCA   \n",
       "\n",
       "                        host_subject_id  qiita_study_id  \n",
       "0  eb4ad376-4cc1-4d2c-9b93-ad2b082c77c4           14990  \n",
       "1  cb2af13b-cbed-4657-8fd4-fed8db375cba           14990  \n",
       "2  29da08f4-c9e2-4e7a-9733-b4ec5b9c5f13           14990  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_TCGA_zebra_genome_WOL  =  19\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:54: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eac_tcga_data_split(os.getcwd() +'/qiita_downloads/qiita14990_EAC_TCGA/species_WOL/feature-table.biom',\n",
    "                'EAC_TCGA_species_WOL')\n",
    "\n",
    "eac_tcga_data_split(os.getcwd() +'/qiita_downloads/qiita14990_EAC_TCGA/171534_none_WOL.biom',\n",
    "                'EAC_TCGA_zebra_genome_WOL',\n",
    "                zebra = 'EAC_TCGA_WOL')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6a673970-f958-4626-9406-8294826e5c88",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Not used in paper: EAC : OLD ICGC, 14521\n",
    "From ICGC , not exact paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "db6ce605-5d5e-41a0-8f41-cf496e36b1e7",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>filename</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>project</th>\n",
       "      <th>project_code</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>submitted_donor_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14521.006cdcd42d55e07a2fd1ad2d31dd1a87</td>\n",
       "      <td>unknown</td>\n",
       "      <td>relapse</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>299.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>distant recurrence/metastasis</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam</td>\n",
       "      <td>DO50381</td>\n",
       "      <td>SA528854</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-RS-031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>no</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>65.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>712.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-SH-071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>unknown</td>\n",
       "      <td>stable</td>\n",
       "      <td>67.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>401.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>alive</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-ZZ-019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              sample_name  \\\n",
       "0  14521.006cdcd42d55e07a2fd1ad2d31dd1a87   \n",
       "1  14521.0337efea9b59b88f804ea384c356377c   \n",
       "2  14521.04a446c6a660694dafe43ccaf1db7162   \n",
       "\n",
       "  cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "0                              unknown                      relapse   \n",
       "1                                   no       no evidence of disease   \n",
       "2                              unknown                       stable   \n",
       "\n",
       "   donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "0                    58.0                     58.0   \n",
       "1                    65.0                     65.0   \n",
       "2                    67.0                      NaN   \n",
       "\n",
       "   donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "0                        58.0                 C15.5   \n",
       "1                        66.0                 C15.5   \n",
       "2                        68.0                 C15.5   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_relapse_interval  \\\n",
       "0                            299.0                   123.0   \n",
       "1                            712.0                     NaN   \n",
       "2                            401.0                     NaN   \n",
       "\n",
       "              donor_relapse_type  ... donor_vital_status  \\\n",
       "0  distant recurrence/metastasis  ...           deceased   \n",
       "1                            NaN  ...           deceased   \n",
       "2                            NaN  ...              alive   \n",
       "\n",
       "                                        filename icgc_donor_id icgc_sample_id  \\\n",
       "0  006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam       DO50381       SA528854   \n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam       DO50389       SA528947   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam       DO50412       SA528884   \n",
       "\n",
       "  prior_malignancy  project project_code qiita_study_id  \\\n",
       "0          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "1          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "2          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "\n",
       "  study_donor_involved_in submitted_donor_id  \n",
       "0                   PCAWG      OCCAMS-RS-031  \n",
       "1                   PCAWG      OCCAMS-SH-071  \n",
       "2                   PCAWG      OCCAMS-ZZ-019  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_species_WOL  =  98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>filename</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>project</th>\n",
       "      <th>project_code</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>submitted_donor_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14521.006cdcd42d55e07a2fd1ad2d31dd1a87</td>\n",
       "      <td>unknown</td>\n",
       "      <td>relapse</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>299.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>distant recurrence/metastasis</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam</td>\n",
       "      <td>DO50381</td>\n",
       "      <td>SA528854</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-RS-031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>no</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>65.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>712.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-SH-071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>unknown</td>\n",
       "      <td>stable</td>\n",
       "      <td>67.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>401.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>alive</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-ZZ-019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              sample_name  \\\n",
       "0  14521.006cdcd42d55e07a2fd1ad2d31dd1a87   \n",
       "1  14521.0337efea9b59b88f804ea384c356377c   \n",
       "2  14521.04a446c6a660694dafe43ccaf1db7162   \n",
       "\n",
       "  cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "0                              unknown                      relapse   \n",
       "1                                   no       no evidence of disease   \n",
       "2                              unknown                       stable   \n",
       "\n",
       "   donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "0                    58.0                     58.0   \n",
       "1                    65.0                     65.0   \n",
       "2                    67.0                      NaN   \n",
       "\n",
       "   donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "0                        58.0                 C15.5   \n",
       "1                        66.0                 C15.5   \n",
       "2                        68.0                 C15.5   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_relapse_interval  \\\n",
       "0                            299.0                   123.0   \n",
       "1                            712.0                     NaN   \n",
       "2                            401.0                     NaN   \n",
       "\n",
       "              donor_relapse_type  ... donor_vital_status  \\\n",
       "0  distant recurrence/metastasis  ...           deceased   \n",
       "1                            NaN  ...           deceased   \n",
       "2                            NaN  ...              alive   \n",
       "\n",
       "                                        filename icgc_donor_id icgc_sample_id  \\\n",
       "0  006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam       DO50381       SA528854   \n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam       DO50389       SA528947   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam       DO50412       SA528884   \n",
       "\n",
       "  prior_malignancy  project project_code qiita_study_id  \\\n",
       "0          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "1          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "2          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "\n",
       "  study_donor_involved_in submitted_donor_id  \n",
       "0                   PCAWG      OCCAMS-RS-031  \n",
       "1                   PCAWG      OCCAMS-SH-071  \n",
       "2                   PCAWG      OCCAMS-ZZ-019  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_genome_WOL  =  98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>filename</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>project</th>\n",
       "      <th>project_code</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>submitted_donor_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14521.006cdcd42d55e07a2fd1ad2d31dd1a87</td>\n",
       "      <td>unknown</td>\n",
       "      <td>relapse</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>299.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>distant recurrence/metastasis</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam</td>\n",
       "      <td>DO50381</td>\n",
       "      <td>SA528854</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-RS-031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>no</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>65.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>712.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>0337efea9b59b88f804ea384c356377c.filtered.bam</td>\n",
       "      <td>DO50389</td>\n",
       "      <td>SA528947</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-SH-071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14521.04a446c6a660694dafe43ccaf1db7162</td>\n",
       "      <td>unknown</td>\n",
       "      <td>stable</td>\n",
       "      <td>67.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>401.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>alive</td>\n",
       "      <td>04a446c6a660694dafe43ccaf1db7162.filtered.bam</td>\n",
       "      <td>DO50412</td>\n",
       "      <td>SA528884</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-ZZ-019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              sample_name  \\\n",
       "0  14521.006cdcd42d55e07a2fd1ad2d31dd1a87   \n",
       "1  14521.0337efea9b59b88f804ea384c356377c   \n",
       "2  14521.04a446c6a660694dafe43ccaf1db7162   \n",
       "\n",
       "  cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "0                              unknown                      relapse   \n",
       "1                                   no       no evidence of disease   \n",
       "2                              unknown                       stable   \n",
       "\n",
       "   donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "0                    58.0                     58.0   \n",
       "1                    65.0                     65.0   \n",
       "2                    67.0                      NaN   \n",
       "\n",
       "   donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "0                        58.0                 C15.5   \n",
       "1                        66.0                 C15.5   \n",
       "2                        68.0                 C15.5   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_relapse_interval  \\\n",
       "0                            299.0                   123.0   \n",
       "1                            712.0                     NaN   \n",
       "2                            401.0                     NaN   \n",
       "\n",
       "              donor_relapse_type  ... donor_vital_status  \\\n",
       "0  distant recurrence/metastasis  ...           deceased   \n",
       "1                            NaN  ...           deceased   \n",
       "2                            NaN  ...              alive   \n",
       "\n",
       "                                        filename icgc_donor_id icgc_sample_id  \\\n",
       "0  006cdcd42d55e07a2fd1ad2d31dd1a87.filtered.bam       DO50381       SA528854   \n",
       "1  0337efea9b59b88f804ea384c356377c.filtered.bam       DO50389       SA528947   \n",
       "2  04a446c6a660694dafe43ccaf1db7162.filtered.bam       DO50412       SA528884   \n",
       "\n",
       "  prior_malignancy  project project_code qiita_study_id  \\\n",
       "0          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "1          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "2          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "\n",
       "  study_donor_involved_in submitted_donor_id  \n",
       "0                   PCAWG      OCCAMS-RS-031  \n",
       "1                   PCAWG      OCCAMS-SH-071  \n",
       "2                   PCAWG      OCCAMS-ZZ-019  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_zebra_genome_WOL  =  98\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>filename</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>project</th>\n",
       "      <th>project_code</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>submitted_donor_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>14521.3756a5bb0047504a87838271674b6424</td>\n",
       "      <td>yes</td>\n",
       "      <td>stable</td>\n",
       "      <td>60.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>1262.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>3756a5bb0047504a87838271674b6424.filtered.bam</td>\n",
       "      <td>DO50441</td>\n",
       "      <td>SA528993</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-AH-174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>14521.18b320f63f6c5dd2dd4e0c30c87a1ce9</td>\n",
       "      <td>unknown</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>75.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>1003.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>alive</td>\n",
       "      <td>18b320f63f6c5dd2dd4e0c30c87a1ce9.filtered.bam</td>\n",
       "      <td>DO50434</td>\n",
       "      <td>SA528916</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-AH-061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>14521.5d003d442dbaf291b70d2d8a85db9670</td>\n",
       "      <td>no</td>\n",
       "      <td>stable</td>\n",
       "      <td>50.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>239.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>deceased</td>\n",
       "      <td>5d003d442dbaf291b70d2d8a85db9670.filtered.bam</td>\n",
       "      <td>DO50402</td>\n",
       "      <td>SA528872</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-WG-002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               sample_name  \\\n",
       "48  14521.3756a5bb0047504a87838271674b6424   \n",
       "27  14521.18b320f63f6c5dd2dd4e0c30c87a1ce9   \n",
       "79  14521.5d003d442dbaf291b70d2d8a85db9670   \n",
       "\n",
       "   cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "48                                  yes                       stable   \n",
       "27                              unknown       no evidence of disease   \n",
       "79                                   no                       stable   \n",
       "\n",
       "    donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "48                    60.0                     60.0   \n",
       "27                    75.0                     75.0   \n",
       "79                    50.0                     50.0   \n",
       "\n",
       "    donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "48                        63.0                 C15.5   \n",
       "27                        77.0                 C15.5   \n",
       "79                        50.0                 C15.5   \n",
       "\n",
       "    donor_interval_of_last_followup  donor_relapse_interval  \\\n",
       "48                           1262.0                     NaN   \n",
       "27                           1003.0                     NaN   \n",
       "79                            239.0                    79.0   \n",
       "\n",
       "   donor_relapse_type  ... donor_vital_status  \\\n",
       "48                NaN  ...           deceased   \n",
       "27                NaN  ...              alive   \n",
       "79                NaN  ...           deceased   \n",
       "\n",
       "                                         filename icgc_donor_id  \\\n",
       "48  3756a5bb0047504a87838271674b6424.filtered.bam       DO50441   \n",
       "27  18b320f63f6c5dd2dd4e0c30c87a1ce9.filtered.bam       DO50434   \n",
       "79  5d003d442dbaf291b70d2d8a85db9670.filtered.bam       DO50402   \n",
       "\n",
       "   icgc_sample_id prior_malignancy  project project_code qiita_study_id  \\\n",
       "48       SA528993          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "27       SA528916          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "79       SA528872          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "\n",
       "   study_donor_involved_in submitted_donor_id  \n",
       "48                   PCAWG      OCCAMS-AH-174  \n",
       "27                   PCAWG      OCCAMS-AH-061  \n",
       "79                   PCAWG      OCCAMS-WG-002  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_species_23_WOL  =  24\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>filename</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>project</th>\n",
       "      <th>project_code</th>\n",
       "      <th>qiita_study_id</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>submitted_donor_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>14521.706e8a8516284214f76ba9a63b61697d</td>\n",
       "      <td>unknown</td>\n",
       "      <td>stable</td>\n",
       "      <td>57.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>1059.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>alive</td>\n",
       "      <td>706e8a8516284214f76ba9a63b61697d.filtered.bam</td>\n",
       "      <td>DO50408</td>\n",
       "      <td>SA528878</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-WG-019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>14521.18b320f63f6c5dd2dd4e0c30c87a1ce9</td>\n",
       "      <td>unknown</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>75.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>77.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>1003.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>alive</td>\n",
       "      <td>18b320f63f6c5dd2dd4e0c30c87a1ce9.filtered.bam</td>\n",
       "      <td>DO50434</td>\n",
       "      <td>SA528916</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-AH-061</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>14521.d073e37c97d855da5783148477977ec5</td>\n",
       "      <td>yes</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>79.0</td>\n",
       "      <td>79.0</td>\n",
       "      <td>81.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>860.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>alive</td>\n",
       "      <td>d073e37c97d855da5783148477977ec5.filtered.bam</td>\n",
       "      <td>DO50445</td>\n",
       "      <td>SA528928</td>\n",
       "      <td>unknown</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>ESAD-UK</td>\n",
       "      <td>14521</td>\n",
       "      <td>PCAWG</td>\n",
       "      <td>OCCAMS-ED-036</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                sample_name  \\\n",
       "95   14521.706e8a8516284214f76ba9a63b61697d   \n",
       "27   14521.18b320f63f6c5dd2dd4e0c30c87a1ce9   \n",
       "165  14521.d073e37c97d855da5783148477977ec5   \n",
       "\n",
       "    cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "95                               unknown                       stable   \n",
       "27                               unknown       no evidence of disease   \n",
       "165                                  yes       no evidence of disease   \n",
       "\n",
       "     donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "95                     57.0                     57.0   \n",
       "27                     75.0                     75.0   \n",
       "165                    79.0                     79.0   \n",
       "\n",
       "     donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "95                         59.0                 C15.5   \n",
       "27                         77.0                 C15.5   \n",
       "165                        81.0                 C15.5   \n",
       "\n",
       "     donor_interval_of_last_followup  donor_relapse_interval  \\\n",
       "95                            1059.0                     NaN   \n",
       "27                            1003.0                     NaN   \n",
       "165                            860.0                     NaN   \n",
       "\n",
       "    donor_relapse_type  ... donor_vital_status  \\\n",
       "95                 NaN  ...              alive   \n",
       "27                 NaN  ...              alive   \n",
       "165                NaN  ...              alive   \n",
       "\n",
       "                                          filename icgc_donor_id  \\\n",
       "95   706e8a8516284214f76ba9a63b61697d.filtered.bam       DO50408   \n",
       "27   18b320f63f6c5dd2dd4e0c30c87a1ce9.filtered.bam       DO50434   \n",
       "165  d073e37c97d855da5783148477977ec5.filtered.bam       DO50445   \n",
       "\n",
       "    icgc_sample_id prior_malignancy  project project_code qiita_study_id  \\\n",
       "95        SA528878          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "27        SA528916          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "165       SA528928          unknown  ESAD-UK      ESAD-UK          14521   \n",
       "\n",
       "    study_donor_involved_in submitted_donor_id  \n",
       "95                    PCAWG      OCCAMS-WG-019  \n",
       "27                    PCAWG      OCCAMS-AH-061  \n",
       "165                   PCAWG      OCCAMS-ED-036  \n",
       "\n",
       "[3 rows x 25 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_genome_zebra_23_WOL  =  26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>tumour_histological_type</th>\n",
       "      <th>tumour_grading_system</th>\n",
       "      <th>tumour_grade</th>\n",
       "      <th>tumour_grade_supplemental</th>\n",
       "      <th>tumour_stage_system</th>\n",
       "      <th>tumour_stage</th>\n",
       "      <th>tumour_stage_supplemental</th>\n",
       "      <th>digital_image_of_stained_section</th>\n",
       "      <th>percentage_cellularity</th>\n",
       "      <th>level_of_cellularity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14521.04ceb5f141916f7fc091a5e0aa51ada3</td>\n",
       "      <td>no</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>74.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>767.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14521.08f6c8561dcbc9c4e311cc79d87c73fa</td>\n",
       "      <td>yes</td>\n",
       "      <td>stable</td>\n",
       "      <td>53.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>1771.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T4aN0M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14521.0b7983f4d85911b7351f5136819c96f1</td>\n",
       "      <td>yes</td>\n",
       "      <td>stable</td>\n",
       "      <td>56.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>1473.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              sample_name  \\\n",
       "0  14521.04ceb5f141916f7fc091a5e0aa51ada3   \n",
       "1  14521.08f6c8561dcbc9c4e311cc79d87c73fa   \n",
       "2  14521.0b7983f4d85911b7351f5136819c96f1   \n",
       "\n",
       "  cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "0                                   no       no evidence of disease   \n",
       "1                                  yes                       stable   \n",
       "2                                  yes                       stable   \n",
       "\n",
       "   donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "0                    74.0                     74.0   \n",
       "1                    53.0                     53.0   \n",
       "2                    56.0                     56.0   \n",
       "\n",
       "   donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "0                        76.0                 C15.5   \n",
       "1                        57.0                 C15.5   \n",
       "2                        60.0                 C15.5   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_relapse_interval donor_relapse_type  \\\n",
       "0                            767.0                     NaN                NaN   \n",
       "1                           1771.0                     NaN                NaN   \n",
       "2                           1473.0                     NaN                NaN   \n",
       "\n",
       "   ... tumour_histological_type  tumour_grading_system tumour_grade  \\\n",
       "0  ...                   8140/3               standard          2.0   \n",
       "1  ...                   8140/3               standard          2.0   \n",
       "2  ...                   8140/3               standard          2.0   \n",
       "\n",
       "  tumour_grade_supplemental tumour_stage_system tumour_stage  \\\n",
       "0                       NaN                TNM7       T2N0M0   \n",
       "1                       NaN                TNM7       T4aN0M   \n",
       "2                       NaN                TNM7       T3N0M0   \n",
       "\n",
       "  tumour_stage_supplemental digital_image_of_stained_section  \\\n",
       "0                       NaN                              NaN   \n",
       "1                       NaN                              NaN   \n",
       "2                       NaN                              NaN   \n",
       "\n",
       "  percentage_cellularity level_of_cellularity  \n",
       "0                    NaN                  NaN  \n",
       "1                    NaN                  NaN  \n",
       "2                    NaN                  NaN  \n",
       "\n",
       "[3 rows x 54 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_genome_WOL_lowStage  =  16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>tumour_histological_type</th>\n",
       "      <th>tumour_grading_system</th>\n",
       "      <th>tumour_grade</th>\n",
       "      <th>tumour_grade_supplemental</th>\n",
       "      <th>tumour_stage_system</th>\n",
       "      <th>tumour_stage</th>\n",
       "      <th>tumour_stage_supplemental</th>\n",
       "      <th>digital_image_of_stained_section</th>\n",
       "      <th>percentage_cellularity</th>\n",
       "      <th>level_of_cellularity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>no</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>65.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>712.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T3N2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14521.066f8f871be1147c121d60aa552efad0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>stable</td>\n",
       "      <td>67.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>2162.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14521.0a250b03103541ef9137514aa0ac5af0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>73.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>1127.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T3N1M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              sample_name  \\\n",
       "0  14521.0337efea9b59b88f804ea384c356377c   \n",
       "1  14521.066f8f871be1147c121d60aa552efad0   \n",
       "2  14521.0a250b03103541ef9137514aa0ac5af0   \n",
       "\n",
       "  cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "0                                   no       no evidence of disease   \n",
       "1                              unknown                       stable   \n",
       "2                              unknown       no evidence of disease   \n",
       "\n",
       "   donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "0                    65.0                     65.0   \n",
       "1                    67.0                     67.0   \n",
       "2                    73.0                     73.0   \n",
       "\n",
       "   donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "0                        66.0                 C15.5   \n",
       "1                        72.0                 C15.5   \n",
       "2                        76.0                 C15.5   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_relapse_interval donor_relapse_type  \\\n",
       "0                            712.0                     NaN                NaN   \n",
       "1                           2162.0                     NaN                NaN   \n",
       "2                           1127.0                     NaN                NaN   \n",
       "\n",
       "   ... tumour_histological_type  tumour_grading_system tumour_grade  \\\n",
       "0  ...                   8140/3               standard          3.0   \n",
       "1  ...                   8140/3               standard          3.0   \n",
       "2  ...                   8140/3               standard          3.0   \n",
       "\n",
       "  tumour_grade_supplemental tumour_stage_system tumour_stage  \\\n",
       "0                       NaN                TNM7        T3N2M   \n",
       "1                       NaN                TNM7       T3N0M0   \n",
       "2                       NaN                TNM7        T3N1M   \n",
       "\n",
       "  tumour_stage_supplemental digital_image_of_stained_section  \\\n",
       "0                       NaN                              NaN   \n",
       "1                       NaN                              NaN   \n",
       "2                       NaN                              NaN   \n",
       "\n",
       "  percentage_cellularity level_of_cellularity  \n",
       "0                    NaN                  NaN  \n",
       "1                    NaN                  NaN  \n",
       "2                    NaN                  NaN  \n",
       "\n",
       "[3 rows x 54 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_genome_WOL_highStage  =  16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>tumour_histological_type</th>\n",
       "      <th>tumour_grading_system</th>\n",
       "      <th>tumour_grade</th>\n",
       "      <th>tumour_grade_supplemental</th>\n",
       "      <th>tumour_stage_system</th>\n",
       "      <th>tumour_stage</th>\n",
       "      <th>tumour_stage_supplemental</th>\n",
       "      <th>digital_image_of_stained_section</th>\n",
       "      <th>percentage_cellularity</th>\n",
       "      <th>level_of_cellularity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14521.04ceb5f141916f7fc091a5e0aa51ada3</td>\n",
       "      <td>no</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>74.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>767.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T2N0M0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14521.08f6c8561dcbc9c4e311cc79d87c73fa</td>\n",
       "      <td>yes</td>\n",
       "      <td>stable</td>\n",
       "      <td>53.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>57.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>1771.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T4aN0M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14521.0b7983f4d85911b7351f5136819c96f1</td>\n",
       "      <td>yes</td>\n",
       "      <td>stable</td>\n",
       "      <td>56.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>1473.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              sample_name  \\\n",
       "0  14521.04ceb5f141916f7fc091a5e0aa51ada3   \n",
       "1  14521.08f6c8561dcbc9c4e311cc79d87c73fa   \n",
       "2  14521.0b7983f4d85911b7351f5136819c96f1   \n",
       "\n",
       "  cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "0                                   no       no evidence of disease   \n",
       "1                                  yes                       stable   \n",
       "2                                  yes                       stable   \n",
       "\n",
       "   donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "0                    74.0                     74.0   \n",
       "1                    53.0                     53.0   \n",
       "2                    56.0                     56.0   \n",
       "\n",
       "   donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "0                        76.0                 C15.5   \n",
       "1                        57.0                 C15.5   \n",
       "2                        60.0                 C15.5   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_relapse_interval donor_relapse_type  \\\n",
       "0                            767.0                     NaN                NaN   \n",
       "1                           1771.0                     NaN                NaN   \n",
       "2                           1473.0                     NaN                NaN   \n",
       "\n",
       "   ... tumour_histological_type  tumour_grading_system tumour_grade  \\\n",
       "0  ...                   8140/3               standard          2.0   \n",
       "1  ...                   8140/3               standard          2.0   \n",
       "2  ...                   8140/3               standard          2.0   \n",
       "\n",
       "  tumour_grade_supplemental tumour_stage_system tumour_stage  \\\n",
       "0                       NaN                TNM7       T2N0M0   \n",
       "1                       NaN                TNM7       T4aN0M   \n",
       "2                       NaN                TNM7       T3N0M0   \n",
       "\n",
       "  tumour_stage_supplemental digital_image_of_stained_section  \\\n",
       "0                       NaN                              NaN   \n",
       "1                       NaN                              NaN   \n",
       "2                       NaN                              NaN   \n",
       "\n",
       "  percentage_cellularity level_of_cellularity  \n",
       "0                    NaN                  NaN  \n",
       "1                    NaN                  NaN  \n",
       "2                    NaN                  NaN  \n",
       "\n",
       "[3 rows x 54 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_species_WOL_lowStage  =  16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>...</th>\n",
       "      <th>tumour_histological_type</th>\n",
       "      <th>tumour_grading_system</th>\n",
       "      <th>tumour_grade</th>\n",
       "      <th>tumour_grade_supplemental</th>\n",
       "      <th>tumour_stage_system</th>\n",
       "      <th>tumour_stage</th>\n",
       "      <th>tumour_stage_supplemental</th>\n",
       "      <th>digital_image_of_stained_section</th>\n",
       "      <th>percentage_cellularity</th>\n",
       "      <th>level_of_cellularity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14521.0337efea9b59b88f804ea384c356377c</td>\n",
       "      <td>no</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>65.0</td>\n",
       "      <td>65.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>712.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T3N2M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14521.066f8f871be1147c121d60aa552efad0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>stable</td>\n",
       "      <td>67.0</td>\n",
       "      <td>67.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>2162.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T3N0M0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14521.0a250b03103541ef9137514aa0ac5af0</td>\n",
       "      <td>unknown</td>\n",
       "      <td>no evidence of disease</td>\n",
       "      <td>73.0</td>\n",
       "      <td>73.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>C15.5</td>\n",
       "      <td>1127.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>8140/3</td>\n",
       "      <td>standard</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNM7</td>\n",
       "      <td>T3N1M</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              sample_name  \\\n",
       "0  14521.0337efea9b59b88f804ea384c356377c   \n",
       "1  14521.066f8f871be1147c121d60aa552efad0   \n",
       "2  14521.0a250b03103541ef9137514aa0ac5af0   \n",
       "\n",
       "  cancer_history_first_degree_relative disease_status_last_followup  \\\n",
       "0                                   no       no evidence of disease   \n",
       "1                              unknown                       stable   \n",
       "2                              unknown       no evidence of disease   \n",
       "\n",
       "   donor_age_at_diagnosis  donor_age_at_enrollment  \\\n",
       "0                    65.0                     65.0   \n",
       "1                    67.0                     67.0   \n",
       "2                    73.0                     73.0   \n",
       "\n",
       "   donor_age_at_last_followup donor_diagnosis_icd10  \\\n",
       "0                        66.0                 C15.5   \n",
       "1                        72.0                 C15.5   \n",
       "2                        76.0                 C15.5   \n",
       "\n",
       "   donor_interval_of_last_followup  donor_relapse_interval donor_relapse_type  \\\n",
       "0                            712.0                     NaN                NaN   \n",
       "1                           2162.0                     NaN                NaN   \n",
       "2                           1127.0                     NaN                NaN   \n",
       "\n",
       "   ... tumour_histological_type  tumour_grading_system tumour_grade  \\\n",
       "0  ...                   8140/3               standard          3.0   \n",
       "1  ...                   8140/3               standard          3.0   \n",
       "2  ...                   8140/3               standard          3.0   \n",
       "\n",
       "  tumour_grade_supplemental tumour_stage_system tumour_stage  \\\n",
       "0                       NaN                TNM7        T3N2M   \n",
       "1                       NaN                TNM7       T3N0M0   \n",
       "2                       NaN                TNM7        T3N1M   \n",
       "\n",
       "  tumour_stage_supplemental digital_image_of_stained_section  \\\n",
       "0                       NaN                              NaN   \n",
       "1                       NaN                              NaN   \n",
       "2                       NaN                              NaN   \n",
       "\n",
       "  percentage_cellularity level_of_cellularity  \n",
       "0                    NaN                  NaN  \n",
       "1                    NaN                  NaN  \n",
       "2                    NaN                  NaN  \n",
       "\n",
       "[3 rows x 54 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of samples in EAC_species_WOL_highStage  =  16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Summer23/ISMEJ_EsophModeling_Paper/Figures/comad_of_EAC_progression/python_scripts/import_split_biom.py:90: FutureWarning: The behavior of .astype from SparseDtype to a non-sparse dtype is deprecated. In a future version, this will return a non-sparse array with the requested dtype. To retain the old behavior, use `obj.astype(SparseDtype(dtype))`\n",
      "  df.to_csv(filename, sep = '\\t')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147030_species_WOL.biom',\n",
    "                'EAC_species_WOL')\n",
    "\n",
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147032_genome_WOL.biom',\n",
    "                'EAC_genome_WOL')\n",
    "\n",
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147032_genome_WOL.biom',\n",
    "                'EAC_zebra_genome_WOL',\n",
    "              zebra= 'EAC_WOL')\n",
    "\n",
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147030_species_WOL.biom',\n",
    "                'EAC_species_23_WOL',\n",
    "              trim = 48)\n",
    "\n",
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147032_genome_WOL.biom',\n",
    "                'EAC_genome_zebra_23_WOL',\n",
    "               zebra = 'EAC_WOL',\n",
    "              trim = 48)\n",
    "\n",
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147032_genome_WOL.biom',\n",
    "                'EAC_genome_WOL_lowStage',\n",
    "              stage='low')\n",
    "\n",
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147032_genome_WOL.biom',\n",
    "                'EAC_genome_WOL_highStage',\n",
    "              stage='high')\n",
    "\n",
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147030_species_WOL.biom',\n",
    "                'EAC_species_WOL_lowStage',\n",
    "              stage = 'low')\n",
    "\n",
    "eac_data_split(os.getcwd() +'/qiita_downloads/qiita14521_EAC_ICGC/147030_species_WOL.biom',\n",
    "                'EAC_species_WOL_highStage',\n",
    "              stage = 'high')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "comad_EAC_figures",
   "language": "python",
   "name": "comad_eac_figures"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
